<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Create DNAm data sets</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BBMRIomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="data.html">Data</a>
</li>
<li>
  <a href="usage.html">Basic Usage</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Use cases
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">DNA methylation</li>
    <li>
      <a href="EWAS.html">EWAS</a>
    </li>
    <li class="dropdown-header">RNA seq</li>
    <li>
      <a href="TWAS.html">TWAS</a>
    </li>
    <li class="dropdown-header">Metabolites</li>
    <li>
      <a href="MWAS.html">MWAS</a>
    </li>
    <li class="dropdown-header">Genotypes</li>
    <li>
      <a href="HRC.html">imputed genotypes</a>
    </li>
    <li class="dropdown-header">Integrated analyses</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Developers
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="metadatabase.html">Accessing the metadatabase</a>
    </li>
    <li>
      <a href="contribute.html">Contributing</a>
    </li>
    <li class="dropdown-header">Creating Datasets</li>
    <li>
      <a href="create_DNAm_Freeze2_unrelated_datasets.html">DNAm Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_RNAseq_Freeze2_unrelated_datasets.html">RNAseq Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_RNAseq_Freeze2_GoNL_dataset.html">RNAseq Freeze2 GoNL</a>
    </li>
    <li>
      <a href="create_GRCh38_RNAseq_Freeze2_unrelated_datasets.html">GRCh38 RNAseq Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_GRCh38_RNAseq_Freeze2_GoNL_dataset.html">GRCh38 RNAseq Freeze2 GoNL</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Developers (legacy)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Quality Control</li>
    <li>
      <a href="DNAm_qc.html">DNAm</a>
    </li>
    <li class="dropdown-header">RNAseq</li>
    <li class="dropdown-header">Creating Data Sets</li>
    <li>
      <a href="create_RNA_datasets.html">RNAseq</a>
    </li>
    <li>
      <a href="create_DNAm_datasets.html">DNAm</a>
    </li>
    <li>
      <a href="linkingRP4_RP3.html">Linking RP4 with RP3</a>
    </li>
    <li>
      <a href="freeze2.html">Determining freeze 2</a>
    </li>
    <li class="dropdown-header">Miscellaneous</li>
    <li>
      <a href="metadatabase_legacy.html">Accessing the metadatabase</a>
    </li>
    <li class="dropdown-header">Phenotyping</li>
    <li>
      <a href="phenotyping_rnaseq.html">Using RNAseq</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Create DNAm data sets</h1>

</div>


<div id="prepare-sample-metadata" class="section level1">
<h1>Prepare sample metadata</h1>
<p>Preprocess metadata including phenotype harmonization.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(BBMRIomics)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3">samplesheets &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;methylationSamplesheet&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">samplesheets &lt;-<span class="st"> </span>samplesheets[<span class="op">!</span><span class="kw">duplicated</span>(samplesheets<span class="op">$</span>run_id),]</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="co">##for unrelated sets else skip</span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7">colData &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="dt">file=</span><span class="kw">file.path</span>(VM_BASE_ANALYSIS, <span class="st">&quot;BBMRIomics/BBMRIomics/data&quot;</span>, <span class="st">&quot;freeze2_dnam_identifiers.csv&quot;</span>), <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>, <span class="dt">as.is=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb1-8" data-line-number="8">colData &lt;-<span class="st"> </span><span class="kw">merge</span>(colData, samplesheets, <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;run_id&quot;</span>, <span class="st">&quot;ids&quot;</span>))</a>
<a class="sourceLine" id="cb1-9" data-line-number="9"></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="co">##qc</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11"></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"></a>
<a class="sourceLine" id="cb1-13" data-line-number="13"></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="co">##harmonize phenotypes</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="kw">source</span>(<span class="kw">file.path</span>(<span class="kw">path.package</span>(<span class="st">&quot;BBMRIomics&quot;</span>), <span class="st">&quot;scripts/Phenotype_Helpers.R&quot;</span>), <span class="dt">verbose=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb1-16" data-line-number="16">phenotypes &lt;-<span class="st"> </span><span class="kw">cleanPhenotypes</span>()</a>
<a class="sourceLine" id="cb1-17" data-line-number="17"></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="co">##add phenotypes</span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19">colData &lt;-<span class="st"> </span><span class="kw">merge</span>(colData, phenotypes, <span class="dt">by=</span><span class="st">&quot;ids&quot;</span>, <span class="dt">all.x=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb1-20" data-line-number="20"></a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="co">##add genotype ids</span></a>
<a class="sourceLine" id="cb1-22" data-line-number="22">imputation &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;getImputations&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB)</a>
<a class="sourceLine" id="cb1-23" data-line-number="23">hrc &lt;-<span class="st"> </span><span class="kw">subset</span>(imputation, imputation_reference <span class="op">==</span><span class="st"> &quot;HRC&quot;</span>)</a>
<a class="sourceLine" id="cb1-24" data-line-number="24">colData &lt;-<span class="st"> </span><span class="kw">merge</span>(colData, hrc[, <span class="kw">c</span>(<span class="st">&quot;ids&quot;</span>, <span class="st">&quot;imputation_id&quot;</span>)], <span class="dt">by=</span><span class="st">&quot;ids&quot;</span>, <span class="dt">all.x=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb1-25" data-line-number="25"></a>
<a class="sourceLine" id="cb1-26" data-line-number="26"><span class="co">##add uuid and gonl_id</span></a>
<a class="sourceLine" id="cb1-27" data-line-number="27">ids &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;getIds&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB)</a>
<a class="sourceLine" id="cb1-28" data-line-number="28">colData &lt;-<span class="st"> </span><span class="kw">merge</span>(colData, ids[, <span class="kw">c</span>(<span class="st">&quot;ids&quot;</span>, <span class="st">&quot;gonl_id&quot;</span>)], <span class="dt">by=</span><span class="st">&quot;ids&quot;</span>, <span class="dt">all.x=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb1-29" data-line-number="29"></a>
<a class="sourceLine" id="cb1-30" data-line-number="30"><span class="co">##ordering</span></a>
<a class="sourceLine" id="cb1-31" data-line-number="31">first &lt;-<span class="st"> </span><span class="kw">match</span>(<span class="kw">c</span>(<span class="st">&quot;ids&quot;</span>, <span class="st">&quot;uuid&quot;</span>, <span class="st">&quot;biobank_id&quot;</span>, <span class="st">&quot;gonl_id&quot;</span>, <span class="st">&quot;run_id&quot;</span>), <span class="kw">colnames</span>(colData))</a>
<a class="sourceLine" id="cb1-32" data-line-number="32"></a>
<a class="sourceLine" id="cb1-33" data-line-number="33">colData &lt;-<span class="st"> </span><span class="kw">cbind</span>(colData[, first], colData[, <span class="op">-</span>first])</a>
<a class="sourceLine" id="cb1-34" data-line-number="34"><span class="kw">rownames</span>(colData) &lt;-<span class="st"> </span>colData<span class="op">$</span>uuid</a>
<a class="sourceLine" id="cb1-35" data-line-number="35"></a>
<a class="sourceLine" id="cb1-36" data-line-number="36"><span class="kw">dim</span>(colData)</a>
<a class="sourceLine" id="cb1-37" data-line-number="37"><span class="kw">head</span>(colData)</a>
<a class="sourceLine" id="cb1-38" data-line-number="38"></a>
<a class="sourceLine" id="cb1-39" data-line-number="39"><span class="kw">table</span>(<span class="op">!</span><span class="kw">is.na</span>(colData<span class="op">$</span>gonl_id))</a>
<a class="sourceLine" id="cb1-40" data-line-number="40"><span class="kw">table</span>(<span class="op">!</span><span class="kw">is.na</span>(colData<span class="op">$</span>imputation_id))</a>
<a class="sourceLine" id="cb1-41" data-line-number="41"><span class="kw">table</span>(<span class="op">!</span>(<span class="kw">is.na</span>(colData<span class="op">$</span>imputation_id) <span class="op">&amp;</span><span class="st"> </span><span class="kw">is.na</span>(colData<span class="op">$</span>gonl_id)))</a></code></pre></div>
</div>
<div id="creating-the-summarizedexperiments" class="section level1">
<h1>Creating the <code>SummarizedExperiments</code></h1>
<p>Here we follow the approach described at <a href="https://molepi.github.io/DNAmArray_workflow/">DNAmArray</a> for preprocessing and quality control.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">library</span>(BiocParallel)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">library</span>(DNAmArray)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw">library</span>(impute)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="kw">register</span>(<span class="kw">MulticoreParam</span>(<span class="dv">16</span>, <span class="dt">log=</span><span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">path450k &lt;-<span class="st"> </span><span class="kw">file.path</span>(VM_BASE_DATA, <span class="st">&quot;IlluminaHumanMethylation450k&quot;</span>)</a>
<a class="sourceLine" id="cb2-6" data-line-number="6"></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="cf">for</span>(biobank <span class="cf">in</span> <span class="kw">unique</span>(colData<span class="op">$</span>biobank_id)[<span class="op">-</span><span class="dv">1</span>]) {</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">    biobank &lt;-<span class="st"> &quot;RS&quot;</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9">    targets &lt;-<span class="st"> </span><span class="kw">subset</span>(colData, biobank_id <span class="op">==</span><span class="st"> </span>biobank)</a>
<a class="sourceLine" id="cb2-10" data-line-number="10">    targets<span class="op">$</span>Basename &lt;-<span class="st"> </span><span class="kw">with</span>(targets,</a>
<a class="sourceLine" id="cb2-11" data-line-number="11">                             <span class="kw">file.path</span>(path450k, <span class="st">&quot;raw&quot;</span>, Sentrix_Barcode,</a>
<a class="sourceLine" id="cb2-12" data-line-number="12">                                       <span class="kw">paste</span>(Sentrix_Barcode, Sentrix_Position, <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)))</a>
<a class="sourceLine" id="cb2-13" data-line-number="13">    <span class="co">##reading the data</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14">    RGset &lt;-<span class="st"> </span><span class="kw">read.metharray.exp.par</span>(targets, <span class="dt">verbose=</span><span class="ot">FALSE</span>, <span class="dt">extended=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-15" data-line-number="15">    <span class="co">##normalize</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16">    pc &lt;-<span class="st"> </span><span class="kw">screeplot</span>(RGset)</a>
<a class="sourceLine" id="cb2-17" data-line-number="17">    GRset &lt;-<span class="st"> </span><span class="kw">preprocessFunnorm.DNAmArray</span>(RGset, <span class="dt">nPCs=</span><span class="dv">3</span>, <span class="dt">keepCN=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb2-18" data-line-number="18">    <span class="kw">rm</span>(pc)</a>
<a class="sourceLine" id="cb2-19" data-line-number="19">    <span class="kw">gc</span>() <span class="co">#remove and clean</span></a>
<a class="sourceLine" id="cb2-20" data-line-number="20">    <span class="co">##filter</span></a>
<a class="sourceLine" id="cb2-21" data-line-number="21">    RGset &lt;-<span class="st"> </span><span class="kw">probeFiltering</span>(RGset)</a>
<a class="sourceLine" id="cb2-22" data-line-number="22">    <span class="co">##store M-values</span></a>
<a class="sourceLine" id="cb2-23" data-line-number="23">    mvalues &lt;-<span class="st"> </span><span class="kw">reduce</span>(GRset, RGset, <span class="dt">what=</span><span class="st">&quot;M&quot;</span>)</a>
<a class="sourceLine" id="cb2-24" data-line-number="24">    <span class="kw">sink</span>(<span class="st">&quot;/dev/null&quot;</span>) <span class="co">##to ignore impute.knn output</span></a>
<a class="sourceLine" id="cb2-25" data-line-number="25">    mvalues &lt;-<span class="st"> </span><span class="kw">impute.knn</span>(<span class="kw">as.matrix</span>(mvalues))<span class="op">$</span>data</a>
<a class="sourceLine" id="cb2-26" data-line-number="26">    <span class="kw">sink</span>()</a>
<a class="sourceLine" id="cb2-27" data-line-number="27"></a>
<a class="sourceLine" id="cb2-28" data-line-number="28">    mid &lt;-<span class="st"> </span><span class="kw">match</span>(<span class="kw">colnames</span>(mvalues), colData<span class="op">$</span>run_id)</a>
<a class="sourceLine" id="cb2-29" data-line-number="29">    <span class="kw">colnames</span>(mvalues) &lt;-<span class="st"> </span>colData<span class="op">$</span>uuid[mid]</a>
<a class="sourceLine" id="cb2-30" data-line-number="30">    rowData &lt;-<span class="st"> </span><span class="kw">getPlatform</span>(<span class="dt">platform=</span><span class="st">&quot;HM450&quot;</span>, <span class="dt">genome=</span><span class="st">&quot;hg19&quot;</span>)</a>
<a class="sourceLine" id="cb2-31" data-line-number="31">    mvalues &lt;-<span class="st"> </span><span class="kw">makeSE</span>(mvalues, colData[,<span class="op">-</span><span class="dv">1</span>], rowData, <span class="dt">note=</span><span class="st">&quot;No chen probe filtering, Functional normalization, knn imputed&quot;</span>)</a>
<a class="sourceLine" id="cb2-32" data-line-number="32">    <span class="kw">save</span>(mvalues, <span class="dt">file=</span><span class="kw">file.path</span>(path450k, <span class="st">&quot;450k&quot;</span>, biobank, <span class="kw">paste0</span>(<span class="st">&quot;methData_Mvalues_&quot;</span>, biobank, <span class="st">&quot;_F2_cleaned.RData&quot;</span>)))</a>
<a class="sourceLine" id="cb2-33" data-line-number="33">    <span class="kw">rm</span>(mvalues)</a>
<a class="sourceLine" id="cb2-34" data-line-number="34">    <span class="kw">gc</span>() <span class="co">#remove and clean</span></a>
<a class="sourceLine" id="cb2-35" data-line-number="35">    <span class="co">##store beta-values</span></a>
<a class="sourceLine" id="cb2-36" data-line-number="36">    betas &lt;-<span class="st"> </span><span class="kw">reduce</span>(GRset, RGset, <span class="dt">what=</span><span class="st">&quot;beta&quot;</span>)</a>
<a class="sourceLine" id="cb2-37" data-line-number="37">    <span class="kw">sink</span>(<span class="st">&quot;/dev/null&quot;</span>) <span class="co">##to ignore impute.knn output</span></a>
<a class="sourceLine" id="cb2-38" data-line-number="38">    betas &lt;-<span class="st"> </span><span class="kw">impute.knn</span>(<span class="kw">as.matrix</span>(betas))<span class="op">$</span>data</a>
<a class="sourceLine" id="cb2-39" data-line-number="39">    <span class="kw">sink</span>()</a>
<a class="sourceLine" id="cb2-40" data-line-number="40">    mid &lt;-<span class="st"> </span><span class="kw">match</span>(<span class="kw">colnames</span>(betas), colData<span class="op">$</span>run_id)</a>
<a class="sourceLine" id="cb2-41" data-line-number="41">    <span class="kw">colnames</span>(betas) &lt;-<span class="st"> </span>colData<span class="op">$</span>uuid[mid]</a>
<a class="sourceLine" id="cb2-42" data-line-number="42">    betas &lt;-<span class="st"> </span><span class="kw">makeSE</span>(betas, colData[,<span class="op">-</span><span class="dv">1</span>], rowData, <span class="dt">note=</span><span class="st">&quot;No chen probe filtering, Functional normalization, knn imputed&quot;</span>)</a>
<a class="sourceLine" id="cb2-43" data-line-number="43">    <span class="kw">save</span>(betas, <span class="dt">file=</span><span class="kw">file.path</span>(path450k, <span class="st">&quot;450k&quot;</span>, biobank, <span class="kw">paste0</span>(<span class="st">&quot;methData_Betas_&quot;</span>, biobank, <span class="st">&quot;_F2_cleaned.RData&quot;</span>)))</a>
<a class="sourceLine" id="cb2-44" data-line-number="44">    <span class="kw">rm</span>(betas)</a>
<a class="sourceLine" id="cb2-45" data-line-number="45">    <span class="kw">rm</span>(GRset)</a>
<a class="sourceLine" id="cb2-46" data-line-number="46">    <span class="kw">rm</span>(RGset)</a>
<a class="sourceLine" id="cb2-47" data-line-number="47">    <span class="kw">gc</span>() <span class="co">##remove and clean</span></a>
<a class="sourceLine" id="cb2-48" data-line-number="48">}</a></code></pre></div>
</div>
<div id="create-one-summarizedexperiment-containg-all-cohorts" class="section level1">
<h1>Create one <code>SummarizedExperiment</code> containg all cohorts</h1>
<p>Here we merge the <code>SummarizedExperiment</code> of the separate cohorts in one single <code>SummarizedExperiment</code>.</p>
<blockquote>
<p>NOTE: direct use of <code>cbind</code> currently doesn’t work with <code>GRanges</code> containing <code>DNAStringSets</code> problem has been reported to Bioconductor-developers who proposed to fix it!</p>
</blockquote>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">path450k &lt;-<span class="st"> </span><span class="kw">file.path</span>(VM_BASE_DATA, <span class="st">&quot;IlluminaHumanMethylation450k&quot;</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">biobank &lt;-<span class="st"> &quot;PAN&quot;</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="kw">load</span>(<span class="kw">file.path</span>(path450k, <span class="st">&quot;450k&quot;</span>, biobank, <span class="kw">paste0</span>(<span class="st">&quot;methData_Betas_&quot;</span>, biobank, <span class="st">&quot;_F2_cleaned.RData&quot;</span>)))</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">b &lt;-<span class="st"> </span>betas</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="kw">library</span>(GenomicRanges)</a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="cf">for</span>(biobank <span class="cf">in</span> <span class="kw">c</span>(<span class="st">&quot;CODAM&quot;</span>, <span class="st">&quot;LLS&quot;</span>, <span class="st">&quot;LL&quot;</span>, <span class="st">&quot;RS&quot;</span>, <span class="st">&quot;NTR&quot;</span>)) {</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">    <span class="kw">load</span>(<span class="kw">file.path</span>(path450k, <span class="st">&quot;450k&quot;</span>, biobank, <span class="kw">paste0</span>(<span class="st">&quot;methData_Betas_&quot;</span>, biobank, <span class="st">&quot;_F2_cleaned.RData&quot;</span>)))</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">    betas &lt;-<span class="st"> </span><span class="kw">subsetByOverlaps</span>(betas, b)</a>
<a class="sourceLine" id="cb3-9" data-line-number="9">    b &lt;-<span class="st"> </span><span class="kw">subsetByOverlaps</span>(b, betas)</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">    <span class="kw">rowData</span>(betas) &lt;-<span class="st"> </span><span class="ot">NULL</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11">    b &lt;-<span class="st"> </span>SummarizedExperiment<span class="op">::</span><span class="kw">cbind</span>(b, betas)</a>
<a class="sourceLine" id="cb3-12" data-line-number="12">    <span class="kw">gc</span>()</a>
<a class="sourceLine" id="cb3-13" data-line-number="13">    <span class="kw">print</span>(b)</a>
<a class="sourceLine" id="cb3-14" data-line-number="14">}</a>
<a class="sourceLine" id="cb3-15" data-line-number="15">betas &lt;-<span class="st"> </span>b</a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="kw">save</span>(betas, <span class="dt">file=</span><span class="kw">file.path</span>(path450k, <span class="st">&quot;450k&quot;</span>, <span class="kw">paste0</span>(<span class="st">&quot;methData_Betas_BIOS_F2_cleaned.RData&quot;</span>)))</a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">##can we reduce the size?</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18">tools<span class="op">::</span><span class="kw">checkRdaFiles</span>(<span class="kw">file.path</span>(path450k, <span class="st">&quot;450k&quot;</span>, <span class="kw">paste0</span>(<span class="st">&quot;methData_Betas_BIOS_F2_cleaned.RData&quot;</span>)))</a>
<a class="sourceLine" id="cb3-19" data-line-number="19"></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">##Now mvalues-values</span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22">path450k &lt;-<span class="st"> </span><span class="kw">file.path</span>(VM_BASE_DATA, <span class="st">&quot;IlluminaHumanMethylation450k&quot;</span>)</a>
<a class="sourceLine" id="cb3-23" data-line-number="23">biobank &lt;-<span class="st"> &quot;PAN&quot;</span></a>
<a class="sourceLine" id="cb3-24" data-line-number="24"><span class="kw">load</span>(<span class="kw">file.path</span>(path450k, <span class="st">&quot;450k&quot;</span>, biobank, <span class="kw">paste0</span>(<span class="st">&quot;methData_Mvalues_&quot;</span>, biobank, <span class="st">&quot;_F2_cleaned.RData&quot;</span>)))</a>
<a class="sourceLine" id="cb3-25" data-line-number="25">b &lt;-<span class="st"> </span>mvalues</a>
<a class="sourceLine" id="cb3-26" data-line-number="26"><span class="kw">library</span>(GenomicRanges)</a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="cf">for</span>(biobank <span class="cf">in</span> <span class="kw">c</span>(<span class="st">&quot;CODAM&quot;</span>, <span class="st">&quot;LLS&quot;</span>, <span class="st">&quot;LL&quot;</span>, <span class="st">&quot;RS&quot;</span>, <span class="st">&quot;NTR&quot;</span>)) {</a>
<a class="sourceLine" id="cb3-28" data-line-number="28">    <span class="kw">load</span>(<span class="kw">file.path</span>(path450k, <span class="st">&quot;450k&quot;</span>, biobank, <span class="kw">paste0</span>(<span class="st">&quot;methData_Mvalues_&quot;</span>, biobank, <span class="st">&quot;_F2_cleaned.RData&quot;</span>)))</a>
<a class="sourceLine" id="cb3-29" data-line-number="29">    mvalues &lt;-<span class="st"> </span><span class="kw">subsetByOverlaps</span>(mvalues, b)</a>
<a class="sourceLine" id="cb3-30" data-line-number="30">    b &lt;-<span class="st"> </span><span class="kw">subsetByOverlaps</span>(b, mvalues)</a>
<a class="sourceLine" id="cb3-31" data-line-number="31">    <span class="kw">rowData</span>(mvalues) &lt;-<span class="st"> </span><span class="ot">NULL</span></a>
<a class="sourceLine" id="cb3-32" data-line-number="32">    b &lt;-<span class="st"> </span>SummarizedExperiment<span class="op">::</span><span class="kw">cbind</span>(b, mvalues)</a>
<a class="sourceLine" id="cb3-33" data-line-number="33">    <span class="kw">gc</span>()</a>
<a class="sourceLine" id="cb3-34" data-line-number="34">    <span class="kw">print</span>(b)</a>
<a class="sourceLine" id="cb3-35" data-line-number="35">}</a>
<a class="sourceLine" id="cb3-36" data-line-number="36">mvalues &lt;-<span class="st"> </span>b</a>
<a class="sourceLine" id="cb3-37" data-line-number="37"><span class="kw">save</span>(mvalues, <span class="dt">file=</span><span class="kw">file.path</span>(path450k, <span class="st">&quot;450k&quot;</span>, <span class="kw">paste0</span>(<span class="st">&quot;methData_Mvalues_BIOS_F2_cleaned.RData&quot;</span>)))</a>
<a class="sourceLine" id="cb3-38" data-line-number="38"><span class="co">##can we reduce the size?</span></a>
<a class="sourceLine" id="cb3-39" data-line-number="39">tools<span class="op">::</span><span class="kw">checkRdaFiles</span>(<span class="kw">file.path</span>(path450k, <span class="st">&quot;450k&quot;</span>, <span class="kw">paste0</span>(<span class="st">&quot;methData_Mvalues_BIOS_F2_cleaned.RData&quot;</span>)))</a></code></pre></div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
