<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>450k quality control</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BBMRIomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="data.html">Data</a>
</li>
<li>
  <a href="usage.html">Basic Usage</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Use cases
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">DNA methylation</li>
    <li>
      <a href="EWAS.html">EWAS</a>
    </li>
    <li class="dropdown-header">RNA seq</li>
    <li>
      <a href="TWAS.html">TWAS</a>
    </li>
    <li class="dropdown-header">Metabolites</li>
    <li>
      <a href="MWAS.html">MWAS</a>
    </li>
    <li class="dropdown-header">Genotypes</li>
    <li>
      <a href="HRC.html">imputed genotypes</a>
    </li>
    <li class="dropdown-header">Integrated analyses</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Developers
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="metadatabase.html">Accessing the metadatabase</a>
    </li>
    <li>
      <a href="contribute.html">Contributing</a>
    </li>
    <li class="dropdown-header">Creating Datasets</li>
    <li>
      <a href="create_DNAm_Freeze2_unrelated_datasets.html">DNAm Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_RNAseq_Freeze2_unrelated_datasets.html">RNAseq Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_RNAseq_Freeze2_GoNL_dataset.html">RNAseq Freeze2 GoNL</a>
    </li>
    <li>
      <a href="create_GRCh38_RNAseq_Freeze2_unrelated_datasets.html">GRCh38 RNAseq Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_GRCh38_RNAseq_Freeze2_GoNL_dataset.html">GRCh38 RNAseq Freeze2 GoNL</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Developers (legacy)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Quality Control</li>
    <li>
      <a href="DNAm_qc.html">DNAm</a>
    </li>
    <li class="dropdown-header">RNAseq</li>
    <li class="dropdown-header">Creating Data Sets</li>
    <li>
      <a href="create_RNA_datasets.html">RNAseq</a>
    </li>
    <li>
      <a href="create_DNAm_datasets.html">DNAm</a>
    </li>
    <li>
      <a href="linkingRP4_RP3.html">Linking RP4 with RP3</a>
    </li>
    <li>
      <a href="freeze2.html">Determining freeze 2</a>
    </li>
    <li class="dropdown-header">Miscellaneous</li>
    <li>
      <a href="metadatabase_legacy.html">Accessing the metadatabase</a>
    </li>
    <li class="dropdown-header">Phenotyping</li>
    <li>
      <a href="phenotyping_rnaseq.html">Using RNAseq</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">450k quality control</h1>

</div>


<div id="sample-level-quality-control" class="section level2">
<h2>Sample-level quality control</h2>
<p>The package <a href="http://bioconductor.org/packages/MethylAid/"><strong>MethylAid</strong></a> is used for sample-level quality control. <a href="http://bioconductor.org/packages/MethylAid/"><strong>MethylAid</strong></a> is specially designed for quality control of large sets of Illumina Human DNA methylation array data sets e.g., epigenomewide association studies (EWAS) using the 450k or 850k (EPIC) arrays. Extracting intensities from IDAT files can be done in batches and/or in parallel to reduce memory load and/or overcome long run-times. It requires two function calls in going from IDAT files to launch the interactive web application; <code>summarize</code> and <code>visualize</code>. For more information see <span class="citation">(Iterson et al. 2014)</span>.</p>
<div id="creating-targets-files" class="section level3">
<h3>Creating targets-files</h3>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">path450k &lt;-<span class="st"> </span><span class="kw">file.path</span>(VM_BASE_DATA, <span class="st">&quot;IlluminaHumanMethylation450k&quot;</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">samplesheets &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;methylationSamplesheet&quot;</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">samplesheets &lt;-<span class="st"> </span>samplesheets[<span class="op">!</span><span class="kw">duplicated</span>(samplesheets<span class="op">$</span>run_id),]</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">samplesheets &lt;-<span class="st"> </span>samplesheets[<span class="kw">order</span>(samplesheets<span class="op">$</span>ids),]</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">head</span>(samplesheets)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">dim</span>(samplesheets)</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="cf">for</span>(biobank <span class="cf">in</span> <span class="kw">c</span>(<span class="st">&quot;CODAM&quot;</span>, <span class="st">&quot;LL&quot;</span>, <span class="st">&quot;LLS&quot;</span>, <span class="st">&quot;NTR&quot;</span>, <span class="st">&quot;PAN&quot;</span>, <span class="st">&quot;RS&quot;</span>))</a>
<a class="sourceLine" id="cb1-8" data-line-number="8">    <span class="kw">write.table</span>(<span class="kw">subset</span>(samplesheets, biobank_id <span class="op">==</span><span class="st"> </span>biobank),</a>
<a class="sourceLine" id="cb1-9" data-line-number="9">                <span class="dt">file =</span> <span class="kw">file.path</span>(path450k, <span class="st">&quot;450k&quot;</span>, biobank, <span class="st">&quot;sample_sheet.txt&quot;</span>),</a>
<a class="sourceLine" id="cb1-10" data-line-number="10">                <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>, <span class="dt">row.names=</span><span class="ot">FALSE</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb1-11" data-line-number="11">samplesheets<span class="op">$</span>Basename &lt;-<span class="st"> </span><span class="kw">with</span>(samplesheets,</a>
<a class="sourceLine" id="cb1-12" data-line-number="12">                              <span class="kw">file.path</span>(path450k, <span class="st">&quot;raw&quot;</span>, Sentrix_Barcode,</a>
<a class="sourceLine" id="cb1-13" data-line-number="13">                                        <span class="kw">paste</span>(Sentrix_Barcode, Sentrix_Position, <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)))</a></code></pre></div>
<p>Now a check is performed to see if all idat-files exists or if we missed one on the VM.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">idats &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">paste0</span>(samplesheets<span class="op">$</span>Basename, <span class="st">&quot;_Red.idat&quot;</span>),</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">           <span class="kw">paste0</span>(samplesheets<span class="op">$</span>Basename, <span class="st">&quot;_Grn.idat&quot;</span>))</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">exists &lt;-<span class="st"> </span><span class="kw">file.exists</span>(idats)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="kw">table</span>(exists)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">files &lt;-<span class="st"> </span><span class="kw">dir</span>(<span class="kw">file.path</span>(path450k, <span class="st">&quot;raw&quot;</span>), <span class="dt">recursive=</span><span class="ot">TRUE</span>, <span class="dt">full.names=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="kw">setdiff</span>(files, idats)</a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="kw">setdiff</span>(idats, files)</a></code></pre></div>
</div>
<div id="summarize" class="section level3">
<h3>Summarize</h3>
<p>Now we are ready to run <strong>MethylAid</strong> summarization.</p>
<p>For each biobank we extract all quality control probe intensities using the <strong>MethylAid</strong> function <code>summarize</code>. Since, we have a VM with multiple cores we can do the summarization in parallel. For each biobank we generate an <code>RData</code>-file called <strong>MethylAid</strong> and store it in the biobanks data directory.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">register</span>(<span class="kw">MulticoreParam</span>(<span class="dv">10</span>, <span class="dt">log=</span><span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">tmp &lt;-<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">c</span>(<span class="st">&quot;CODAM&quot;</span>, <span class="st">&quot;LL&quot;</span>, <span class="st">&quot;LLS&quot;</span>, <span class="st">&quot;NTR&quot;</span>, <span class="st">&quot;PAN&quot;</span>, <span class="st">&quot;RS&quot;</span>), <span class="cf">function</span>(biobank) {</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">    samplesheet &lt;-<span class="st"> </span>samplesheets[samplesheets<span class="op">$</span>biobank_id <span class="op">==</span><span class="st"> </span>biobank,]</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">    <span class="kw">summarize</span>(samplesheet, <span class="dt">batchSize =</span> <span class="dv">100</span>, <span class="dt">BPPARAM =</span> <span class="kw">bpparam</span>(),</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">              <span class="dt">file=</span><span class="kw">file.path</span>(path450k, <span class="st">&quot;450k&quot;</span>, biobank, <span class="st">&quot;MethylAid&quot;</span>))</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">})</a></code></pre></div>
<p>See Apps for the interactive sample-level quality control.</p>
<div id="refs" class="references">
<div id="ref-Iterson2014">
<p>Iterson, M. van, E. W. Tobi, R. C. Slieker, W. den Hollander, R. Luijk, P. E. Slagboom, and B. T. Heijmans. 2014. “MethylAid: visual and interactive quality control of large Illumina 450k datasets.” <em>Bioinformatics</em> 30 (23): 3435–7.</p>
</div>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
