<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Determining Freeze 2</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BBMRIomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="data.html">Data</a>
</li>
<li>
  <a href="usage.html">Basic Usage</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Use cases
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">DNA methylation</li>
    <li>
      <a href="EWAS.html">EWAS</a>
    </li>
    <li class="dropdown-header">RNA seq</li>
    <li>
      <a href="TWAS.html">TWAS</a>
    </li>
    <li class="dropdown-header">Metabolites</li>
    <li>
      <a href="MWAS.html">MWAS</a>
    </li>
    <li class="dropdown-header">Genotypes</li>
    <li>
      <a href="HRC.html">imputed genotypes</a>
    </li>
    <li class="dropdown-header">Integrated analyses</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Developers
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="metadatabase.html">Accessing the metadatabase</a>
    </li>
    <li>
      <a href="contribute.html">Contributing</a>
    </li>
    <li class="dropdown-header">Creating Datasets</li>
    <li>
      <a href="create_DNAm_Freeze2_unrelated_datasets.html">DNAm Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_RNAseq_Freeze2_unrelated_datasets.html">RNAseq Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_RNAseq_Freeze2_GoNL_dataset.html">RNAseq Freeze2 GoNL</a>
    </li>
    <li>
      <a href="create_GRCh38_RNAseq_Freeze2_unrelated_datasets.html">GRCh38 RNAseq Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_GRCh38_RNAseq_Freeze2_GoNL_dataset.html">GRCh38 RNAseq Freeze2 GoNL</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Developers (legacy)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Quality Control</li>
    <li>
      <a href="DNAm_qc.html">DNAm</a>
    </li>
    <li class="dropdown-header">RNAseq</li>
    <li class="dropdown-header">Creating Data Sets</li>
    <li>
      <a href="create_RNA_datasets.html">RNAseq</a>
    </li>
    <li>
      <a href="create_DNAm_datasets.html">DNAm</a>
    </li>
    <li>
      <a href="linkingRP4_RP3.html">Linking RP4 with RP3</a>
    </li>
    <li>
      <a href="freeze2.html">Determining freeze 2</a>
    </li>
    <li class="dropdown-header">Miscellaneous</li>
    <li>
      <a href="metadatabase_legacy.html">Accessing the metadatabase</a>
    </li>
    <li class="dropdown-header">Phenotyping</li>
    <li>
      <a href="phenotyping_rnaseq.html">Using RNAseq</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Determining Freeze 2</h1>

</div>


<p>The aim of the BIOS consortium is to generate RNA sequencing and DNA methylation data for 4000 individuals that have been selected for having array-based genotypes or whole sequencing data (GoNL) already available. Furthermore, the generated data should pass a set of quality control metrics.</p>
<p>The following code shows how to extract the RNA sequencing and DNA methylation samples that pass quality control.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">ids &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;getIds&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB)</a></code></pre></div>
<pre><code>## No username and password provided for the MDB use stored views!</code></pre>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">rna &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;getRNASeqRuns&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB)</a></code></pre></div>
<pre><code>## No username and password provided for the MDB use stored views!</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">rna &lt;-<span class="st"> </span>rna[<span class="op">!</span><span class="kw">duplicated</span>(rna<span class="op">$</span>run_id),] <span class="co">##drop overlapping freezes</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">rna &lt;-<span class="st"> </span><span class="kw">subset</span>(rna, qc <span class="op">==</span><span class="st"> &quot;passed&quot;</span>) <span class="co">##keep those passing qc</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3">rna &lt;-<span class="st"> </span><span class="kw">subset</span>(rna, type <span class="op">!=</span><span class="st"> &quot;replicate&quot;</span>) <span class="co">##drop replicated</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">##now we have still original, reruns and merged runs</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5">merged &lt;-<span class="st"> </span><span class="kw">subset</span>(rna, type <span class="op">==</span><span class="st"> &quot;merged&quot;</span> <span class="op">&amp;</span><span class="st"> </span>freeze <span class="op">==</span><span class="st"> </span><span class="dv">2</span>) <span class="co">##select</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6">original &lt;-<span class="st"> </span><span class="kw">subset</span>(rna, <span class="op">!</span>(ids <span class="op">%in%</span><span class="st"> </span>merged<span class="op">$</span>ids) <span class="op">&amp;</span><span class="st"> </span>type <span class="op">==</span><span class="st"> &quot;original&quot;</span>)</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">rna &lt;-<span class="st"> </span><span class="kw">rbind</span>(merged, original)</a>
<a class="sourceLine" id="cb3-8" data-line-number="8"></a>
<a class="sourceLine" id="cb3-9" data-line-number="9">dnam &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;getMethylationRuns&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB)</a></code></pre></div>
<pre><code>## No username and password provided for the MDB use stored views!</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">dnam &lt;-<span class="st"> </span>dnam[<span class="op">!</span><span class="kw">duplicated</span>(dnam<span class="op">$</span>run_id),] <span class="co">##drop overlapping freezes</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2">dnam &lt;-<span class="st"> </span><span class="kw">subset</span>(dnam, qc <span class="op">==</span><span class="st"> &quot;passed&quot;</span>) <span class="co">##keep those passing qc</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3">dnam &lt;-<span class="st"> </span><span class="kw">subset</span>(dnam, type <span class="op">!=</span><span class="st"> &quot;replicate&quot;</span>) <span class="co">##drop replicated</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">##small checks</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="kw">dim</span>(ids)</a></code></pre></div>
<pre><code>## [1] 6379   14</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">dim</span>(rna)</a></code></pre></div>
<pre><code>## [1] 4427    9</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">dim</span>(dnam)</a></code></pre></div>
<pre><code>## [1] 6072    7</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">sum</span>(<span class="kw">duplicated</span>(ids<span class="op">$</span>ids))</a></code></pre></div>
<pre><code>## [1] 0</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">sum</span>(<span class="kw">duplicated</span>(rna<span class="op">$</span>ids))</a></code></pre></div>
<pre><code>## [1] 0</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">sum</span>(<span class="kw">duplicated</span>(dnam<span class="op">$</span>ids))</a></code></pre></div>
<pre><code>## [1] 0</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw">library</span>(limma)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"></a>
<a class="sourceLine" id="cb10-3" data-line-number="3">df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">rna =</span> ids<span class="op">$</span>ids <span class="op">%in%</span><span class="st"> </span>rna<span class="op">$</span>ids,</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">                 <span class="dt">dnam =</span> ids<span class="op">$</span>ids <span class="op">%in%</span><span class="st"> </span>dnam<span class="op">$</span>ids)</a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="kw">colSums</span>(df)</a></code></pre></div>
<pre><code>##  rna dnam 
## 4427 6072</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">vc &lt;-<span class="st"> </span><span class="kw">vennCounts</span>(df)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">vc</a></code></pre></div>
<pre><code>##   rna dnam Counts
## 1   0    0     92
## 2   0    1   1860
## 3   1    0    215
## 4   1    1   4212
## attr(,"class")
## [1] "VennCounts"</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">vennDiagram</span>(vc)</a></code></pre></div>
<p>Another requirement for Freeze 2 is to have a maximal set of unrelated individuals for which both RNA sequencing and dna methylation data could be generated.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">relations &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;getRelations&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB)</a></code></pre></div>
<pre><code>## No username and password provided for the MDB use stored views!</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">relations &lt;-<span class="st"> </span><span class="kw">subset</span>(relations, <span class="op">!</span><span class="kw">is.na</span>(relation_id))</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="kw">all</span>(relations<span class="op">$</span>ids <span class="op">%in%</span><span class="st"> </span>relations<span class="op">$</span>relation_id) <span class="co">##relations consistent</span></a></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw">length</span>(relations<span class="op">$</span>ids)</a></code></pre></div>
<pre><code>## [1] 4262</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw">length</span>(<span class="kw">unique</span>(relations<span class="op">$</span>ids))</a></code></pre></div>
<pre><code>## [1] 3558</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw">table</span>(relations<span class="op">$</span>relation_type)</a></code></pre></div>
<pre><code>## 
##           2nd degree family genetical 1st degree family 
##                           6                           8 
##                   has child          has dizygotic twin 
##                         534                        1028 
##        has monozygotic twin                  has parent 
##                        1881                         548 
##   has repeated measurements                     has sib 
##                          88                          99 
##  inferred 1st degree family 
##                          70</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">getRelation &lt;-<span class="st"> </span><span class="cf">function</span>(id, relations) <span class="kw">as.character</span>(relations<span class="op">$</span>id[relations<span class="op">$</span>relation_id <span class="op">%in%</span><span class="st"> </span>id])</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"></a>
<a class="sourceLine" id="cb18-3" data-line-number="3">getFamily &lt;-<span class="st"> </span><span class="cf">function</span>(id, relations) {</a>
<a class="sourceLine" id="cb18-4" data-line-number="4">    fam &lt;-<span class="st"> </span>id</a>
<a class="sourceLine" id="cb18-5" data-line-number="5">    id &lt;-<span class="st"> </span><span class="kw">getRelation</span>(id, relations)</a>
<a class="sourceLine" id="cb18-6" data-line-number="6">    <span class="cf">while</span>( <span class="kw">length</span>(<span class="kw">setdiff</span>(id, fam)) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span> ) {</a>
<a class="sourceLine" id="cb18-7" data-line-number="7">        id &lt;-<span class="st"> </span><span class="kw">setdiff</span>(id, fam)</a>
<a class="sourceLine" id="cb18-8" data-line-number="8">        fam &lt;-<span class="st"> </span><span class="kw">c</span>(fam, id)</a>
<a class="sourceLine" id="cb18-9" data-line-number="9">        id &lt;-<span class="st"> </span><span class="kw">getRelation</span>(id, relations)</a>
<a class="sourceLine" id="cb18-10" data-line-number="10">    }</a>
<a class="sourceLine" id="cb18-11" data-line-number="11">    fam</a>
<a class="sourceLine" id="cb18-12" data-line-number="12">}</a>
<a class="sourceLine" id="cb18-13" data-line-number="13"></a>
<a class="sourceLine" id="cb18-14" data-line-number="14">relations<span class="op">$</span>family_id &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb18-15" data-line-number="15">famId &lt;-<span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb18-16" data-line-number="16"><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(relations)) {</a>
<a class="sourceLine" id="cb18-17" data-line-number="17">    id &lt;-<span class="st"> </span><span class="kw">as.character</span>(relations<span class="op">$</span>ids)[i]</a>
<a class="sourceLine" id="cb18-18" data-line-number="18">    family &lt;-<span class="st"> </span><span class="kw">getFamily</span>(id, relations)</a>
<a class="sourceLine" id="cb18-19" data-line-number="19"></a>
<a class="sourceLine" id="cb18-20" data-line-number="20">    <span class="cf">if</span>(<span class="kw">all</span>(<span class="kw">is.na</span>(relations<span class="op">$</span>family_id[relations<span class="op">$</span>ids <span class="op">%in%</span><span class="st"> </span>family]))) {</a>
<a class="sourceLine" id="cb18-21" data-line-number="21">        <span class="co">##message(&quot;adding new family...&quot;)</span></a>
<a class="sourceLine" id="cb18-22" data-line-number="22">        famId &lt;-<span class="st"> </span>famId <span class="op">+</span><span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb18-23" data-line-number="23">        relations<span class="op">$</span>family_id[relations<span class="op">$</span>ids <span class="op">%in%</span><span class="st"> </span>family] &lt;-<span class="st"> </span>famId</a>
<a class="sourceLine" id="cb18-24" data-line-number="24">    }</a>
<a class="sourceLine" id="cb18-25" data-line-number="25">}</a>
<a class="sourceLine" id="cb18-26" data-line-number="26"></a>
<a class="sourceLine" id="cb18-27" data-line-number="27"><span class="kw">max</span>(relations<span class="op">$</span>family_id)</a></code></pre></div>
<pre><code>## [1] 1616</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">famSizes &lt;-<span class="st"> </span><span class="kw">table</span>(relations<span class="op">$</span>family_id)</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="kw">table</span>(famSizes)</a></code></pre></div>
<pre><code>## famSizes
##    2    4    6    8   10   12   17   18   21   22   26   28 
## 1375  171   13    3   22   19    1    6    1    1    1    3</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw">head</span>(<span class="kw">names</span>(famSizes[famSizes<span class="op">==</span><span class="dv">2</span>]))</a></code></pre></div>
<pre><code>## [1] "1" "2" "3" "4" "6" "7"</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw">subset</span>(relations, family_id <span class="op">==</span><span class="st"> </span><span class="dv">1</span>)</a></code></pre></div>
<pre><code>##           ids         uuid biobank_id gonl_id gwas_id
## 17 CODAM-2037 BIOS71A89511      CODAM    <NA>    2037
## 73 CODAM-2240 BIOS6601F3E0      CODAM    <NA>    2240
##                 relation_type relation_id family_id
## 17 inferred 1st degree family  CODAM-2240         1
## 73 inferred 1st degree family  CODAM-2037         1</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="kw">head</span>(<span class="kw">names</span>(famSizes[famSizes<span class="op">==</span><span class="dv">4</span>]))</a></code></pre></div>
<pre><code>## [1] "9"  "10" "11" "12" "13" "14"</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="kw">subset</span>(relations, family_id <span class="op">==</span><span class="st"> </span><span class="dv">9</span>)</a></code></pre></div>
<pre><code>##                ids        uuid biobank_id  gonl_id gwas_id relation_type
## 956 LL-LLDeep_1600 BIOS82666E4         LL gonl-56a    <NA>     has child
## 960 LL-LLDeep_1603 BIOS855C858         LL gonl-56b    <NA>     has child
## 981 LL-LLDeep_1619 BIOS9542DD0         LL gonl-56c    <NA>    has parent
## 982 LL-LLDeep_1619 BIOS9542DD0         LL gonl-56c    <NA>    has parent
##        relation_id family_id
## 956 LL-LLDeep_1619         9
## 960 LL-LLDeep_1619         9
## 981 LL-LLDeep_1600         9
## 982 LL-LLDeep_1603         9</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw">head</span>(<span class="kw">names</span>(famSizes[famSizes<span class="op">==</span><span class="dv">12</span>]))</a></code></pre></div>
<pre><code>## [1] "165" "194" "308" "325" "383" "387"</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw">subset</span>(relations, family_id <span class="op">==</span><span class="st"> </span><span class="dv">165</span>)</a></code></pre></div>
<pre><code>##                           ids         uuid biobank_id gonl_id gwas_id
## 2148         NTR-A1083C-10146 BIOS7E1BFC2D        NTR    <NA>   10175
## 2149         NTR-A1083C-10146 BIOS7E1BFC2D        NTR    <NA>   10175
## 2150         NTR-A1083C-10146 BIOS7E1BFC2D        NTR    <NA>   10175
## 2151 NTR-A1083C-NTR15215-8666 BIOS59AC7812        NTR    <NA>   10175
## 2152 NTR-A1083C-NTR15215-8666 BIOS59AC7812        NTR    <NA>   10175
## 2153 NTR-A1083C-NTR15215-8666 BIOS59AC7812        NTR    <NA>   10175
## 2154         NTR-A1083D-10175 BIOS0AC3A8CD        NTR    <NA>   10175
## 2155         NTR-A1083D-10175 BIOS0AC3A8CD        NTR    <NA>   10175
## 2156         NTR-A1083D-10175 BIOS0AC3A8CD        NTR    <NA>   10175
## 2157 NTR-A1083D-NTR15589-8860 BIOS063C99A7        NTR    <NA>   10175
## 2158 NTR-A1083D-NTR15589-8860 BIOS063C99A7        NTR    <NA>   10175
## 2159 NTR-A1083D-NTR15589-8860 BIOS063C99A7        NTR    <NA>   10175
##                  relation_type              relation_id family_id
## 2148      has monozygotic twin         NTR-A1083D-10175       165
## 2149      has monozygotic twin NTR-A1083D-NTR15589-8860       165
## 2150 has repeated measurements NTR-A1083C-NTR15215-8666       165
## 2151      has monozygotic twin         NTR-A1083D-10175       165
## 2152      has monozygotic twin NTR-A1083D-NTR15589-8860       165
## 2153 has repeated measurements         NTR-A1083C-10146       165
## 2154      has monozygotic twin         NTR-A1083C-10146       165
## 2155      has monozygotic twin NTR-A1083C-NTR15215-8666       165
## 2156 has repeated measurements NTR-A1083D-NTR15589-8860       165
## 2157      has monozygotic twin         NTR-A1083C-10146       165
## 2158      has monozygotic twin NTR-A1083C-NTR15215-8666       165
## 2159 has repeated measurements         NTR-A1083D-10175       165</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="kw">head</span>(<span class="kw">names</span>(famSizes[famSizes<span class="op">==</span><span class="dv">28</span>]))</a></code></pre></div>
<pre><code>## [1] "261" "692" "738"</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1"><span class="kw">subset</span>(relations, family_id <span class="op">==</span><span class="st"> </span><span class="dv">261</span>)</a></code></pre></div>
<pre><code>##                              ids         uuid biobank_id  gonl_id  gwas_id
## 2368 NTR-A118A-NTR01371-06D07229 BIOS29A0CC56        NTR gonl-96a 06D07229
## 2369 NTR-A118A-NTR01371-06D07229 BIOS29A0CC56        NTR gonl-96a 06D07229
## 2370 NTR-A118A-NTR01371-06D07229 BIOS29A0CC56        NTR gonl-96a 06D07229
## 2371 NTR-A118A-NTR01371-06D07229 BIOS29A0CC56        NTR gonl-96a 06D07229
## 2372 NTR-A118B-NTR01373-06D07230 BIOS13D06DE0        NTR gonl-96b 06D07230
## 2373 NTR-A118B-NTR01373-06D07230 BIOS13D06DE0        NTR gonl-96b 06D07230
## 2374 NTR-A118B-NTR01373-06D07230 BIOS13D06DE0        NTR gonl-96b 06D07230
## 2375 NTR-A118B-NTR01373-06D07230 BIOS13D06DE0        NTR gonl-96b 06D07230
## 2376               NTR-A118C-829 BIOS34C8B8A5        NTR gonl-96c    10375
## 2377               NTR-A118C-829 BIOS34C8B8A5        NTR gonl-96c    10375
## 2378               NTR-A118C-829 BIOS34C8B8A5        NTR gonl-96c    10375
## 2379               NTR-A118C-829 BIOS34C8B8A5        NTR gonl-96c    10375
## 2380               NTR-A118C-829 BIOS34C8B8A5        NTR gonl-96c    10375
## 2381   NTR-A118C-NT0027644-10375 BIOS125DBF1F        NTR gonl-96c    10375
## 2382   NTR-A118C-NT0027644-10375 BIOS125DBF1F        NTR gonl-96c    10375
## 2383   NTR-A118C-NT0027644-10375 BIOS125DBF1F        NTR gonl-96c    10375
## 2384   NTR-A118C-NT0027644-10375 BIOS125DBF1F        NTR gonl-96c    10375
## 2385   NTR-A118C-NT0027644-10375 BIOS125DBF1F        NTR gonl-96c    10375
## 2386   NTR-A118D-NT0027645-10376 BIOSD15C4FC7        NTR     <NA>    10375
## 2387   NTR-A118D-NT0027645-10376 BIOSD15C4FC7        NTR     <NA>    10375
## 2388   NTR-A118D-NT0027645-10376 BIOSD15C4FC7        NTR     <NA>    10375
## 2389   NTR-A118D-NT0027645-10376 BIOSD15C4FC7        NTR     <NA>    10375
## 2390   NTR-A118D-NT0027645-10376 BIOSD15C4FC7        NTR     <NA>    10375
## 2391      NTR-A118D-NTR00927-607 BIOSA249C90B        NTR     <NA>    10375
## 2392      NTR-A118D-NTR00927-607 BIOSA249C90B        NTR     <NA>    10375
## 2393      NTR-A118D-NTR00927-607 BIOSA249C90B        NTR     <NA>    10375
## 2394      NTR-A118D-NTR00927-607 BIOSA249C90B        NTR     <NA>    10375
## 2395      NTR-A118D-NTR00927-607 BIOSA249C90B        NTR     <NA>    10375
##                  relation_type                 relation_id family_id
## 2368                 has child               NTR-A118C-829       261
## 2369                 has child   NTR-A118C-NT0027644-10375       261
## 2370                 has child   NTR-A118D-NT0027645-10376       261
## 2371                 has child      NTR-A118D-NTR00927-607       261
## 2372                 has child               NTR-A118C-829       261
## 2373                 has child   NTR-A118C-NT0027644-10375       261
## 2374                 has child   NTR-A118D-NT0027645-10376       261
## 2375                 has child      NTR-A118D-NTR00927-607       261
## 2376      has monozygotic twin   NTR-A118D-NT0027645-10376       261
## 2377      has monozygotic twin      NTR-A118D-NTR00927-607       261
## 2378                has parent NTR-A118A-NTR01371-06D07229       261
## 2379                has parent NTR-A118B-NTR01373-06D07230       261
## 2380 has repeated measurements   NTR-A118C-NT0027644-10375       261
## 2381      has monozygotic twin   NTR-A118D-NT0027645-10376       261
## 2382      has monozygotic twin      NTR-A118D-NTR00927-607       261
## 2383                has parent NTR-A118A-NTR01371-06D07229       261
## 2384                has parent NTR-A118B-NTR01373-06D07230       261
## 2385 has repeated measurements               NTR-A118C-829       261
## 2386      has monozygotic twin               NTR-A118C-829       261
## 2387      has monozygotic twin   NTR-A118C-NT0027644-10375       261
## 2388                has parent NTR-A118A-NTR01371-06D07229       261
## 2389                has parent NTR-A118B-NTR01373-06D07230       261
## 2390 has repeated measurements      NTR-A118D-NTR00927-607       261
## 2391      has monozygotic twin               NTR-A118C-829       261
## 2392      has monozygotic twin   NTR-A118C-NT0027644-10375       261
## 2393                has parent NTR-A118A-NTR01371-06D07229       261
## 2394                has parent NTR-A118B-NTR01373-06D07230       261
## 2395 has repeated measurements   NTR-A118D-NT0027645-10376       261</code></pre>
<p>Now we can selected the maximal unrelated individuals e.g. in case of GoNL trio’s if all have dnam and rna chose the parents; to maximize the number of individuals.</p>
<p>The rna or dnam freeze 2 is extended with unrelated individuals having only rna or dnam.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1">reduceRelations &lt;-<span class="st"> </span><span class="cf">function</span>(ids, relations) {</a>
<a class="sourceLine" id="cb28-2" data-line-number="2"></a>
<a class="sourceLine" id="cb28-3" data-line-number="3">    rels &lt;-<span class="st"> </span>relations[relations<span class="op">$</span>ids <span class="op">%in%</span><span class="st"> </span>ids,]</a>
<a class="sourceLine" id="cb28-4" data-line-number="4"></a>
<a class="sourceLine" id="cb28-5" data-line-number="5">    selection &lt;-<span class="st"> </span><span class="cf">function</span>(fam) {</a>
<a class="sourceLine" id="cb28-6" data-line-number="6">        <span class="cf">if</span>(<span class="kw">sum</span>(fam<span class="op">$</span>relation_type <span class="op">==</span><span class="st"> &quot;has child&quot;</span>) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) { <span class="co">##both parents have rna/dnam</span></a>
<a class="sourceLine" id="cb28-7" data-line-number="7">            <span class="co">##message(&quot;family having both parents with both rna/dnam!&quot;)</span></a>
<a class="sourceLine" id="cb28-8" data-line-number="8">            <span class="kw">return</span>( <span class="kw">as.character</span>(fam<span class="op">$</span>ids[fam<span class="op">$</span>relation_type <span class="op">==</span><span class="st"> &quot;has child&quot;</span>]) )</a>
<a class="sourceLine" id="cb28-9" data-line-number="9">        } <span class="cf">else</span> <span class="cf">if</span>( <span class="kw">sum</span>(<span class="kw">grepl</span>(<span class="st">&quot;twin&quot;</span>, fam<span class="op">$</span>relation_type)) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">1</span> ) { <span class="co">##return single twin</span></a>
<a class="sourceLine" id="cb28-10" data-line-number="10">            <span class="co">##message(&quot;family with at least one twin with both rna/dnam!&quot;)</span></a>
<a class="sourceLine" id="cb28-11" data-line-number="11">                <span class="kw">return</span>( <span class="kw">as.character</span>(fam<span class="op">$</span>ids[<span class="kw">which</span>(<span class="kw">grepl</span>(<span class="st">&quot;twin&quot;</span>, fam<span class="op">$</span>relation_type))[<span class="dv">1</span>]]) )</a>
<a class="sourceLine" id="cb28-12" data-line-number="12">        } <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb28-13" data-line-number="13">            <span class="co">##message(&quot;family with at least on member having both rna/dnam!&quot;)</span></a>
<a class="sourceLine" id="cb28-14" data-line-number="14">            <span class="kw">return</span>( <span class="kw">as.character</span>(fam<span class="op">$</span>ids[<span class="dv">1</span>] ) ) <span class="co">##return any other single family member with complete rna/dnam</span></a>
<a class="sourceLine" id="cb28-15" data-line-number="15">        }</a>
<a class="sourceLine" id="cb28-16" data-line-number="16">    }</a>
<a class="sourceLine" id="cb28-17" data-line-number="17"></a>
<a class="sourceLine" id="cb28-18" data-line-number="18">    selected &lt;-<span class="st"> </span><span class="kw">c</span>()</a>
<a class="sourceLine" id="cb28-19" data-line-number="19">    <span class="cf">for</span>(i <span class="cf">in</span> <span class="kw">unique</span>(rels<span class="op">$</span>family_id)) {</a>
<a class="sourceLine" id="cb28-20" data-line-number="20">        fam &lt;-<span class="st"> </span><span class="kw">subset</span>(rels, family_id <span class="op">==</span><span class="st"> </span>i)</a>
<a class="sourceLine" id="cb28-21" data-line-number="21">        <span class="co">##print(fam)</span></a>
<a class="sourceLine" id="cb28-22" data-line-number="22">        select &lt;-<span class="st"> </span><span class="kw">selection</span>(fam)</a>
<a class="sourceLine" id="cb28-23" data-line-number="23">        <span class="co">##print(select)</span></a>
<a class="sourceLine" id="cb28-24" data-line-number="24">        selected &lt;-<span class="st"> </span><span class="kw">c</span>(selected, select)        </a>
<a class="sourceLine" id="cb28-25" data-line-number="25">    }</a>
<a class="sourceLine" id="cb28-26" data-line-number="26"></a>
<a class="sourceLine" id="cb28-27" data-line-number="27">    selected &lt;-<span class="st"> </span><span class="kw">unique</span>(selected) <span class="co">##relations of parents having two children</span></a>
<a class="sourceLine" id="cb28-28" data-line-number="28">    <span class="kw">c</span>(<span class="kw">setdiff</span>(ids, rels<span class="op">$</span>ids), selected)</a>
<a class="sourceLine" id="cb28-29" data-line-number="29">}</a>
<a class="sourceLine" id="cb28-30" data-line-number="30"></a>
<a class="sourceLine" id="cb28-31" data-line-number="31"><span class="co">##unrelated having both rna and dnam</span></a>
<a class="sourceLine" id="cb28-32" data-line-number="32">ovl &lt;-<span class="st"> </span><span class="kw">intersect</span>(rna<span class="op">$</span>ids, dnam<span class="op">$</span>ids)</a>
<a class="sourceLine" id="cb28-33" data-line-number="33">ovlRed &lt;-<span class="st"> </span><span class="kw">reduceRelations</span>(ovl, relations)</a>
<a class="sourceLine" id="cb28-34" data-line-number="34"><span class="kw">length</span>(ovlRed)</a></code></pre></div>
<pre><code>## [1] 3405</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1"><span class="kw">length</span>(<span class="kw">reduceRelations</span>(ovlRed, relations)) <span class="co">##check</span></a></code></pre></div>
<pre><code>## [1] 3405</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1"><span class="co">##remove all family members in remaining rna/dnam</span></a>
<a class="sourceLine" id="cb30-2" data-line-number="2">ovlRel &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(ovlRed, getFamily, relations))</a>
<a class="sourceLine" id="cb30-3" data-line-number="3"></a>
<a class="sourceLine" id="cb30-4" data-line-number="4">srna &lt;-<span class="st"> </span><span class="kw">subset</span>(rna, <span class="op">!</span>(ids <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(ovlRed, ovlRel)))</a>
<a class="sourceLine" id="cb30-5" data-line-number="5"><span class="co">##and get the unrelated set</span></a>
<a class="sourceLine" id="cb30-6" data-line-number="6">rnaOnly &lt;-<span class="st"> </span><span class="kw">reduceRelations</span>(srna<span class="op">$</span>ids, relations)</a>
<a class="sourceLine" id="cb30-7" data-line-number="7">rnaF2 &lt;-<span class="st"> </span><span class="kw">subset</span>(rna, ids <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(ovlRed, rnaOnly))</a>
<a class="sourceLine" id="cb30-8" data-line-number="8"><span class="kw">dim</span>(rnaF2)</a></code></pre></div>
<pre><code>## [1] 3530    9</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1">sdnam &lt;-<span class="st"> </span><span class="kw">subset</span>(dnam, <span class="op">!</span>(ids <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(ovlRed, ovlRel)))</a>
<a class="sourceLine" id="cb31-2" data-line-number="2"><span class="co">##and get the unrelated set</span></a>
<a class="sourceLine" id="cb31-3" data-line-number="3">dnamOnly &lt;-<span class="st"> </span><span class="kw">reduceRelations</span>(sdnam<span class="op">$</span>ids, relations)</a>
<a class="sourceLine" id="cb31-4" data-line-number="4">dnamF2 &lt;-<span class="st"> </span><span class="kw">subset</span>(dnam, ids <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(ovlRed, dnamOnly))</a>
<a class="sourceLine" id="cb31-5" data-line-number="5"><span class="kw">dim</span>(dnamF2)</a></code></pre></div>
<pre><code>## [1] 4386    7</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1">freeze2 &lt;-<span class="st"> </span><span class="kw">merge</span>(rnaF2[,<span class="kw">c</span>(<span class="st">&quot;ids&quot;</span>, <span class="st">&quot;run_id&quot;</span>)], dnamF2[,<span class="kw">c</span>(<span class="st">&quot;ids&quot;</span>, <span class="st">&quot;run_id&quot;</span>)], <span class="dt">by=</span><span class="st">&quot;ids&quot;</span>, <span class="dt">suffixes=</span><span class="kw">c</span>(<span class="st">&quot;_rna&quot;</span>,<span class="st">&quot;_dnam&quot;</span>))</a>
<a class="sourceLine" id="cb32-2" data-line-number="2"><span class="kw">head</span>(freeze2)</a></code></pre></div>
<pre><code>##          ids     run_id_rna       run_id_dnam
## 1 CODAM-2001 BD1NYRACXX-5-1 8667053102_R05C02
## 2 CODAM-2002 AD10W1ACXX-4-1 8667053157_R01C02
## 3 CODAM-2013 AD10W1ACXX-4-2 8655685053_R04C02
## 4 CODAM-2016 BD1NYRACXX-5-3 8655685094_R01C01
## 5 CODAM-2017 AD10W1ACXX-4-3 8667053076_R02C01
## 6 CODAM-2020 BD1NYRACXX-5-4 8655685021_R04C02</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="kw">dim</span>(freeze2)</a></code></pre></div>
<pre><code>## [1] 3405    3</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1"><span class="co">##double check</span></a>
<a class="sourceLine" id="cb34-2" data-line-number="2">relations &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;getRelations&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB)</a>
<a class="sourceLine" id="cb34-3" data-line-number="3"><span class="kw">dim</span>(freeze2)</a></code></pre></div>
<pre><code>## No username and password provided for the MDB use stored views!</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1">relations &lt;-<span class="st"> </span><span class="kw">subset</span>(relations, <span class="op">!</span><span class="kw">is.na</span>(relation_id))</a>
<a class="sourceLine" id="cb35-2" data-line-number="2">relations &lt;-<span class="st"> </span><span class="kw">subset</span>(relations, ids <span class="op">%in%</span><span class="st"> </span>freeze2<span class="op">$</span>ids)</a>
<a class="sourceLine" id="cb35-3" data-line-number="3"><span class="kw">intersect</span>(relations<span class="op">$</span>ids, relations<span class="op">$</span>relation_id) <span class="co">##intersect is empty!</span></a></code></pre></div>
<pre><code>## character(0)</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1"><span class="kw">write.table</span>(freeze2, <span class="dt">file=</span><span class="kw">file.path</span>(VM_BASE_ANALYSIS, <span class="st">&quot;BBMRIomics/BBMRIomics/data&quot;</span>, <span class="st">&quot;freeze2_overlap_identifiers.csv&quot;</span>), <span class="dt">row.names=</span><span class="ot">FALSE</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</a>
<a class="sourceLine" id="cb36-2" data-line-number="2"><span class="kw">write.table</span>(rnaF2[, <span class="kw">c</span>(<span class="st">&quot;ids&quot;</span>, <span class="st">&quot;run_id&quot;</span>)], <span class="dt">file=</span><span class="kw">file.path</span>(VM_BASE_ANALYSIS, <span class="st">&quot;BBMRIomics/BBMRIomics/data&quot;</span>, <span class="st">&quot;freeze2_rna_identifiers.csv&quot;</span>), <span class="dt">row.names=</span><span class="ot">FALSE</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</a>
<a class="sourceLine" id="cb36-3" data-line-number="3"><span class="kw">write.table</span>(dnamF2[, <span class="kw">c</span>(<span class="st">&quot;ids&quot;</span>, <span class="st">&quot;run_id&quot;</span>)], <span class="dt">file=</span><span class="kw">file.path</span>(VM_BASE_ANALYSIS, <span class="st">&quot;BBMRIomics/BBMRIomics/data&quot;</span>, <span class="st">&quot;freeze2_dnam_identifiers.csv&quot;</span>), <span class="dt">row.names=</span><span class="ot">FALSE</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</a></code></pre></div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
