<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Accessing the BIOS metadatabase</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BBMRIomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="data.html">Data</a>
</li>
<li>
  <a href="usage.html">Basic Usage</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Use cases
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">DNA methylation</li>
    <li>
      <a href="EWAS.html">EWAS</a>
    </li>
    <li class="dropdown-header">RNA seq</li>
    <li>
      <a href="TWAS.html">TWAS</a>
    </li>
    <li class="dropdown-header">Metabolites</li>
    <li>
      <a href="MWAS.html">MWAS</a>
    </li>
    <li class="dropdown-header">Genotypes</li>
    <li>
      <a href="HRC.html">imputed genotypes</a>
    </li>
    <li class="dropdown-header">Integrated analyses</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Developers
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="metadatabase.html">Accessing the metadatabase</a>
    </li>
    <li>
      <a href="contribute.html">Contributing</a>
    </li>
    <li class="dropdown-header">Creating Datasets</li>
    <li>
      <a href="create_DNAm_Freeze2_unrelated_datasets.html">DNAm Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_RNAseq_Freeze2_unrelated_datasets.html">RNAseq Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_RNAseq_Freeze2_GoNL_dataset.html">RNAseq Freeze2 GoNL</a>
    </li>
    <li>
      <a href="create_GRCh38_RNAseq_Freeze2_unrelated_datasets.html">GRCh38 RNAseq Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_GRCh38_RNAseq_Freeze2_GoNL_dataset.html">GRCh38 RNAseq Freeze2 GoNL</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Developers (legacy)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Quality Control</li>
    <li>
      <a href="DNAm_qc.html">DNAm</a>
    </li>
    <li class="dropdown-header">RNAseq</li>
    <li class="dropdown-header">Creating Data Sets</li>
    <li>
      <a href="create_RNA_datasets.html">RNAseq</a>
    </li>
    <li>
      <a href="create_DNAm_datasets.html">DNAm</a>
    </li>
    <li>
      <a href="linkingRP4_RP3.html">Linking RP4 with RP3</a>
    </li>
    <li>
      <a href="freeze2.html">Determining freeze 2</a>
    </li>
    <li class="dropdown-header">Miscellaneous</li>
    <li>
      <a href="metadatabase_legacy.html">Accessing the metadatabase</a>
    </li>
    <li class="dropdown-header">Phenotyping</li>
    <li>
      <a href="phenotyping_rnaseq.html">Using RNAseq</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Accessing the BIOS metadatabase</h1>

</div>


<blockquote>
<p>This document uses deprecated functions.</p>
</blockquote>
<p>The BIOS project has generated for over 4000 individuals RNA-sequencing and DNA methylation data. A part from these data, GoNL imputed genotypes were generated from existing genotypes and several phenotypes/demographic variables were collected for the same set of samples. A highly flexible sample-oriented metadatabase (MDb) was created in order to manage the dynamic generation of this large-scale multiple-omic data set.</p>
<p>The MDb is a non-relation database (<a href="http://couchdb.apache.org/" class="uri">http://couchdb.apache.org/</a>) that uses JSON to store records and JavaScript for querying. Furthermore, it has an HTTP API suitable to programmatically access the database from the GRID, e.g, the bios alignment pipeline.</p>
<p>Each record or document is a sample (individual) within the BIOS project and has a unique identifier.</p>
<p>Each document has a predefined structure according to our database <a href="couchdbapp/schema/bios.json">schema</a>. Custom R scripts are used to update or modify the database (<a href="BBMRIomics/R/MDb_Helpers.R">see</a>).</p>
<p>Access to the metadatabase (MDb) is restricted; please contact (Leon Mei).</p>
<div id="description-of-mdb-content" class="section level2">
<h2>Description of MDb content</h2>
<p>The MDb contains as much as meta-information as possible from all samples and datatypes: location of (raw) data on srm, md5 checksum verification, quality control information, links between the different identifiers used (person_id, dna_id, etc) and phenotype information.</p>
<p>Every sample’s meta information is encoded in a CouchDB document. Each document has a unique identifier (the bios_id) which is biobankname (CODAM, LL, LLS, NTR, RS and PAN) concatenated with person_id separated by a “-”, e.g. CODAM-2001. This unique bios_id is not suitable for use in the public domain, e.g., EGA upload, therefore a unique non-identifiable identifier has been created for each individual; the uuid.</p>
<p>Every update of a sample in the database is recorded by increasing a revision number. Therefore it is always possible to undo wrong updates.</p>
</div>
<div id="description-available-views" class="section level2">
<h2>Description available views</h2>
<p>Views are the way to extract information from the couchDb. Views are organized into designs; each design contains a number of views related to a particular kind of information that can be extracted from the MDb. For example, there is a design <code>EGA</code> which contains currently two views 1) <code>freeze1RNASeq</code> to extract those samples for which RNAseq data has been uploaded to EGA and 2) <code>freeze1Methylation</code> for the DNA methylation data.</p>
<p>Other relevant views are:</p>
<table>
<thead>
<tr class="header">
<th align="left">design:</th>
<th align="left">view:</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Identifiers</td>
<td align="left">Ids, Relations</td>
</tr>
<tr class="even">
<td align="left">Phenotypes</td>
<td align="left">Phenotypes</td>
</tr>
<tr class="odd">
<td align="left">RNA</td>
<td align="left">Fastq, RNARuns, RNASamplesheet</td>
</tr>
<tr class="even">
<td align="left">DNAm</td>
<td align="left">Idat, DNAmRuns, DNAmSamplesheet</td>
</tr>
<tr class="odd">
<td align="left">DNA</td>
<td align="left">Imputations</td>
</tr>
<tr class="even">
<td align="left">EGA</td>
<td align="left">freeze1RNA, freeze1DNAm, freeze2RNA, freeze2DNAm</td>
</tr>
</tbody>
</table>
<blockquote>
<p>Note: We can always add views if necessary; please contact Leon Mei.</p>
</blockquote>
</div>
<div id="accessing-the-mdb" class="section level2">
<h2>Accessing the MDb</h2>
<p>Views can be downloaded as JSON documents by making a GET request. Most programming languages have utilities for making GET requests and to transform JSON documents. Some programming languages have an API for CouchDB e.g. JAVA and Python. There are several online tools available for transforming JSON documents to csv files.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">head</span>(<span class="kw">getView</span>(<span class="st">&quot;getIds&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">head</span>(<span class="kw">getView</span>(<span class="st">&quot;getFastq&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB))</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">head</span>(<span class="kw">getView</span>(<span class="st">&quot;getIdat&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB))</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">head</span>(<span class="kw">getView</span>(<span class="st">&quot;getRNASeqRuns&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB))</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">head</span>(<span class="kw">getView</span>(<span class="st">&quot;getMethylationRuns&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB))</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">head</span>(<span class="kw">getView</span>(<span class="st">&quot;getGenotypes&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB))</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">head</span>(<span class="kw">getView</span>(<span class="st">&quot;getImputations&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB))</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="kw">head</span>(<span class="kw">getView</span>(<span class="st">&quot;getRelations&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB))</a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="kw">head</span>(<span class="kw">getView</span>(<span class="st">&quot;allPhenotypes&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB))</a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="kw">head</span>(<span class="kw">getView</span>(<span class="st">&quot;rnaseqSamplesheet&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB))</a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="kw">head</span>(<span class="kw">getView</span>(<span class="st">&quot;methylationSamplesheet&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB))</a></code></pre></div>
<div id="access-the-metadatabase-using-r" class="section level3">
<h3>Access the metadatabase using R</h3>
<p>BBMRIomics uses a configuration file to read in your MDb username and password, so that you do not have to type it every time you use the MDb.</p>
<p>Create a file called <code>.bbmriomics</code> (<code>.biosrutils</code> is also a valid name) and stored it in your home directory on the VM (/home/username) and add e.g.:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="ex">usrpwdrp3</span>: <span class="st">&#39;rp3_username:password&#39;</span> </a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="ex">usrpwdrp4</span>: <span class="st">&#39;rp4_username:password&#39;</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="ex">proxy</span>:  /tmp/your_proxy</a></code></pre></div>
<p>The first line contains your username and password for the RP3 database. The latter two can be optionally for accessing the RP4 database and accessing data from SRM using the function <code>SRM2VM</code>.</p>
<p>On loading the <strong>BBMRIomics</strong> library your username and password will be set in the variable <code>RP3_MDB_USRPWD</code>. The <code>getView</code>-function can now be used like this:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">ids &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;getIds&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB)</a></code></pre></div>
<p>Also, access to the RNAseq run database (if you requested an account) is possible through the <code>getView</code>-function.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">stats &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;getStats&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_RDB)</a></code></pre></div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
