<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Phenotyping using RNAseq data (IN PROGRESS)</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BBMRIomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="data.html">Data</a>
</li>
<li>
  <a href="usage.html">Basic Usage</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Use cases
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">DNA methylation</li>
    <li>
      <a href="EWAS.html">EWAS</a>
    </li>
    <li class="dropdown-header">RNA seq</li>
    <li>
      <a href="TWAS.html">TWAS</a>
    </li>
    <li class="dropdown-header">Metabolites</li>
    <li>
      <a href="MWAS.html">MWAS</a>
    </li>
    <li class="dropdown-header">Genotypes</li>
    <li>
      <a href="HRC.html">imputed genotypes</a>
    </li>
    <li class="dropdown-header">Integrated analyses</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Developers
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="metadatabase.html">Accessing the metadatabase</a>
    </li>
    <li>
      <a href="contribute.html">Contributing</a>
    </li>
    <li class="dropdown-header">Creating Datasets</li>
    <li>
      <a href="create_DNAm_Freeze2_unrelated_datasets.html">DNAm Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_RNAseq_Freeze2_unrelated_datasets.html">RNAseq Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_RNAseq_Freeze2_GoNL_dataset.html">RNAseq Freeze2 GoNL</a>
    </li>
    <li>
      <a href="create_GRCh38_RNAseq_Freeze2_unrelated_datasets.html">GRCh38 RNAseq Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_GRCh38_RNAseq_Freeze2_GoNL_dataset.html">GRCh38 RNAseq Freeze2 GoNL</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Developers (legacy)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Quality Control</li>
    <li>
      <a href="DNAm_qc.html">DNAm</a>
    </li>
    <li class="dropdown-header">RNAseq</li>
    <li class="dropdown-header">Creating Data Sets</li>
    <li>
      <a href="create_RNA_datasets.html">RNAseq</a>
    </li>
    <li>
      <a href="create_DNAm_datasets.html">DNAm</a>
    </li>
    <li>
      <a href="linkingRP4_RP3.html">Linking RP4 with RP3</a>
    </li>
    <li>
      <a href="freeze2.html">Determining freeze 2</a>
    </li>
    <li class="dropdown-header">Miscellaneous</li>
    <li>
      <a href="metadatabase_legacy.html">Accessing the metadatabase</a>
    </li>
    <li class="dropdown-header">Phenotyping</li>
    <li>
      <a href="phenotyping_rnaseq.html">Using RNAseq</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Phenotyping using RNAseq data (IN PROGRESS)</h1>

</div>


<div id="intro" class="section level1">
<h1>Intro</h1>
<p>This document describes how we try to impute some missing phenotypes, specifically, sex, age, smoking status and white blood cell composition.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">preprocess &lt;-<span class="st"> </span><span class="cf">function</span>(sumExp) {</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">    <span class="kw">require</span>(SummarizedExperiment)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">    <span class="kw">require</span>(edgeR)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"></a>
<a class="sourceLine" id="cb1-5" data-line-number="5">    y &lt;-<span class="st"> </span><span class="kw">DGEList</span>(<span class="kw">assays</span>(sumExp)<span class="op">$</span>data)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"></a>
<a class="sourceLine" id="cb1-7" data-line-number="7">    <span class="kw">message</span>(<span class="st">&quot;[preprocess] drop features not expressed&quot;</span>)</a>
<a class="sourceLine" id="cb1-8" data-line-number="8">    <span class="co">##https://www.bioconductor.org/packages/devel/bioc/vignettes/limma/inst/doc/usersguide.pdf</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9">    A &lt;-<span class="st"> </span><span class="kw">rowSums</span>(y<span class="op">$</span>counts)</a>
<a class="sourceLine" id="cb1-10" data-line-number="10">    isexpr &lt;-<span class="st"> </span>A <span class="op">&gt;</span><span class="st"> </span><span class="dv">50</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11">    y &lt;-<span class="st"> </span>y[isexpr,]</a>
<a class="sourceLine" id="cb1-12" data-line-number="12"></a>
<a class="sourceLine" id="cb1-13" data-line-number="13">    <span class="kw">message</span>(<span class="st">&quot;[preprocess] normalize&quot;</span>)</a>
<a class="sourceLine" id="cb1-14" data-line-number="14">    y &lt;-<span class="st"> </span><span class="kw">calcNormFactors</span>(y) <span class="co">##normalize takes some time</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15"></a>
<a class="sourceLine" id="cb1-16" data-line-number="16">    <span class="kw">message</span>(<span class="st">&quot;[preprocess] calculate log counts per million&quot;</span>)</a>
<a class="sourceLine" id="cb1-17" data-line-number="17">    y &lt;-<span class="st"> </span><span class="kw">cpm</span>(y, <span class="dt">log=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb1-18" data-line-number="18"></a>
<a class="sourceLine" id="cb1-19" data-line-number="19">    sumExp &lt;-<span class="st"> </span>sumExp[isexpr,]</a>
<a class="sourceLine" id="cb1-20" data-line-number="20">    <span class="kw">assays</span>(sumExp)<span class="op">$</span>data &lt;-<span class="st"> </span>y <span class="co">##replace data with filtered subset</span></a>
<a class="sourceLine" id="cb1-21" data-line-number="21">    sumExp</a>
<a class="sourceLine" id="cb1-22" data-line-number="22">}</a>
<a class="sourceLine" id="cb1-23" data-line-number="23"></a>
<a class="sourceLine" id="cb1-24" data-line-number="24">validate &lt;-<span class="st"> </span><span class="cf">function</span>(phenotype, features, <span class="dt">train.frac =</span> <span class="dv">2</span><span class="op">/</span><span class="dv">3</span>, <span class="dt">methods =</span> <span class="kw">c</span>(<span class="st">&quot;ridge&quot;</span>, <span class="st">&quot;elastic-net&quot;</span>, <span class="st">&quot;pls&quot;</span>, <span class="st">&quot;lasso&quot;</span>, <span class="st">&quot;gbm&quot;</span>, <span class="st">&quot;mboost&quot;</span>), <span class="dt">ntop =</span> <span class="kw">c</span>(<span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">250</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">2500</span>, <span class="dv">5000</span>), <span class="dt">nrep =</span> <span class="dv">25</span>, <span class="dt">verbose=</span><span class="ot">TRUE</span>) {</a>
<a class="sourceLine" id="cb1-25" data-line-number="25">    <span class="kw">require</span>(omicsPrint)</a>
<a class="sourceLine" id="cb1-26" data-line-number="26">    <span class="kw">require</span>(reshape2)</a>
<a class="sourceLine" id="cb1-27" data-line-number="27">    <span class="kw">require</span>(BiocParallel)</a>
<a class="sourceLine" id="cb1-28" data-line-number="28">        </a>
<a class="sourceLine" id="cb1-29" data-line-number="29">    validations &lt;-<span class="st"> </span><span class="kw">c</span>()</a>
<a class="sourceLine" id="cb1-30" data-line-number="30">    <span class="cf">for</span>(ntopi <span class="cf">in</span> ntop) {</a>
<a class="sourceLine" id="cb1-31" data-line-number="31">        reps &lt;-<span class="st"> </span><span class="kw">bplapply</span>(<span class="dv">1</span><span class="op">:</span>nrep, <span class="cf">function</span>(x) <span class="kw">phenotyping</span>(phenotype, features, <span class="dt">train.frac=</span>train.frac, <span class="dt">methods =</span> methods, <span class="dt">ntop=</span>ntopi, <span class="dt">verbose=</span>verbose)<span class="op">$</span>validation)</a>
<a class="sourceLine" id="cb1-32" data-line-number="32">        reps &lt;-<span class="st"> </span><span class="kw">do.call</span>(<span class="st">&#39;rbind&#39;</span>, reps)</a>
<a class="sourceLine" id="cb1-33" data-line-number="33">        validation &lt;-<span class="st"> </span><span class="kw">melt</span>(reps, <span class="dt">varnames=</span><span class="kw">c</span>(<span class="st">&quot;method&quot;</span>, <span class="st">&quot;measure&quot;</span>))</a>
<a class="sourceLine" id="cb1-34" data-line-number="34">        validation<span class="op">$</span>ntop &lt;-<span class="st"> </span>ntopi</a>
<a class="sourceLine" id="cb1-35" data-line-number="35">        validations &lt;-<span class="st"> </span><span class="kw">rbind</span>(validations, validation)</a>
<a class="sourceLine" id="cb1-36" data-line-number="36">        <span class="kw">gc</span>()</a>
<a class="sourceLine" id="cb1-37" data-line-number="37">    }</a>
<a class="sourceLine" id="cb1-38" data-line-number="38">    validations</a>
<a class="sourceLine" id="cb1-39" data-line-number="39">}</a>
<a class="sourceLine" id="cb1-40" data-line-number="40"></a>
<a class="sourceLine" id="cb1-41" data-line-number="41">path &lt;-<span class="st"> &quot;/virdir/Scratch/RP3_analysis/imputed_phenotypes&quot;</span></a></code></pre></div>
</div>
<div id="inspect-missing-data" class="section level1">
<h1>Inspect missing data</h1>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">library</span>(BBMRIomics)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">library</span>(omicsPrint)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw">data</span>(rnaSeqData_ReadCounts_BIOS_cleaned)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">covariates &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(counts)[, <span class="kw">c</span>(<span class="st">&quot;Sex&quot;</span>, <span class="st">&quot;Sampling_Age&quot;</span>, <span class="st">&quot;Smoking&quot;</span>, <span class="st">&quot;Lymph_Perc&quot;</span>,<span class="st">&quot;Mono_Perc&quot;</span>, <span class="st">&quot;Eos_Perc&quot;</span>, <span class="st">&quot;Neut_Perc&quot;</span>)])</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">##missing covariate summary</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6">missingCov &lt;-<span class="st"> </span><span class="kw">apply</span>(covariates, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="kw">round</span>(<span class="dv">100</span><span class="op">*</span><span class="kw">table</span>(<span class="kw">is.na</span>(x))<span class="op">/</span><span class="kw">length</span>(x),<span class="dv">2</span>))</a>
<a class="sourceLine" id="cb2-7" data-line-number="7">missingCov</a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co">##missing sample summary</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9">missingSam &lt;-<span class="st"> </span><span class="kw">apply</span>(covariates, <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="kw">round</span>(<span class="dv">100</span><span class="op">*</span><span class="kw">sum</span>(<span class="kw">is.na</span>(x))<span class="op">/</span><span class="kw">length</span>(x),<span class="dv">2</span>))</a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="kw">table</span>(missingSam)</a>
<a class="sourceLine" id="cb2-11" data-line-number="11"></a>
<a class="sourceLine" id="cb2-12" data-line-number="12">tbl &lt;-<span class="st"> </span><span class="kw">table</span>(counts<span class="op">$</span>biobank_id, <span class="kw">is.na</span>(counts<span class="op">$</span>Sampling_Age))</a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="kw">round</span>(<span class="dv">100</span><span class="op">*</span>tbl[,<span class="dv">2</span>]<span class="op">/</span><span class="kw">rowSums</span>(tbl),<span class="dv">2</span>)</a>
<a class="sourceLine" id="cb2-14" data-line-number="14"></a>
<a class="sourceLine" id="cb2-15" data-line-number="15">biobank &lt;-<span class="st"> </span><span class="kw">factor</span>(<span class="kw">colData</span>(counts)<span class="op">$</span>biobank_id)</a>
<a class="sourceLine" id="cb2-16" data-line-number="16">missing &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">82</span>,<span class="dv">12</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">48</span>)</a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="kw">levels</span>(biobank) &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="kw">levels</span>(biobank), <span class="st">&quot; (&quot;</span>, missing, <span class="st">&quot;)&quot;</span>)</a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="kw">colData</span>(counts)<span class="op">$</span>biobank_id &lt;-<span class="st"> </span>biobank</a>
<a class="sourceLine" id="cb2-19" data-line-number="19"></a>
<a class="sourceLine" id="cb2-20" data-line-number="20"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb2-21" data-line-number="21">gp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">as.data.frame</span>(<span class="kw">colData</span>(counts)), <span class="kw">aes</span>(<span class="dt">x=</span>Sampling_Age))</a>
<a class="sourceLine" id="cb2-22" data-line-number="22">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">geom_histogram</span>()</a>
<a class="sourceLine" id="cb2-23" data-line-number="23">gp <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>biobank_id)</a>
<a class="sourceLine" id="cb2-24" data-line-number="24"><span class="kw">ggsave</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;Age_RNAseq_distr.eps&quot;</span>))</a></code></pre></div>
</div>
<div id="predicting-sex" class="section level1">
<h1>Predicting Sex</h1>
<div id="model-selection-and-validation" class="section level2">
<h2>Model Selection and Validation</h2>
<p>First we need to find out which algorithm and which number of features to use.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">data</span>(rnaSeqData_ReadCounts_BIOS_cleaned)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw">dim</span>(counts)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">cnts &lt;-<span class="st"> </span><span class="kw">preprocess</span>(counts)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">dim</span>(cnts)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="kw">metadata</span>(cnts)<span class="op">$</span>formula &lt;-<span class="st"> </span><span class="er">~</span><span class="dv">0</span><span class="op">+</span>Sex</a>
<a class="sourceLine" id="cb3-7" data-line-number="7"></a>
<a class="sourceLine" id="cb3-8" data-line-number="8">phenotype &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">get_all_vars</span>(<span class="kw">metadata</span>(cnts)<span class="op">$</span>formula, <span class="dt">data=</span><span class="kw">colData</span>(cnts)))</a>
<a class="sourceLine" id="cb3-9" data-line-number="9">features &lt;-<span class="st"> </span><span class="kw">assays</span>(cnts)<span class="op">$</span>data</a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="kw">table</span>(phenotype, <span class="dt">useNA=</span><span class="st">&quot;always&quot;</span>)</a>
<a class="sourceLine" id="cb3-11" data-line-number="11"></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="kw">library</span>(BiocParallel)</a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="kw">register</span>(<span class="kw">MulticoreParam</span>(<span class="dv">10</span>))</a>
<a class="sourceLine" id="cb3-14" data-line-number="14">validation &lt;-<span class="st"> </span><span class="kw">validate</span>(phenotype, features, <span class="dt">methods =</span> <span class="kw">c</span>(<span class="st">&quot;ridge&quot;</span>, <span class="st">&quot;elastic-net&quot;</span>, <span class="st">&quot;lasso&quot;</span>), <span class="dt">ntop =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">25</span>, <span class="dv">50</span>, <span class="dv">100</span>), <span class="dt">nrep =</span> <span class="dv">25</span>, <span class="dt">verbose=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb3-15" data-line-number="15"></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="kw">library</span>(ggsci)</a>
<a class="sourceLine" id="cb3-18" data-line-number="18">gp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">subset</span>(validation, measure <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;accuracy (Overall)&quot;</span>, <span class="st">&quot;f1 (Male)&quot;</span>, <span class="st">&quot;f1 (Female)&quot;</span>)),</a>
<a class="sourceLine" id="cb3-19" data-line-number="19">             <span class="kw">aes</span>(<span class="dt">x=</span><span class="kw">as.factor</span>(ntop), <span class="dt">y=</span>value, <span class="dt">col=</span>method))</a>
<a class="sourceLine" id="cb3-20" data-line-number="20">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>()</a>
<a class="sourceLine" id="cb3-21" data-line-number="21">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>measure, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>, <span class="dt">nrow=</span><span class="dv">3</span>, <span class="dt">ncol=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb3-22" data-line-number="22">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Number of features&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Validation of predictors for `Sex`&quot;</span>)</a>
<a class="sourceLine" id="cb3-23" data-line-number="23">gp <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_npg</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="dt">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>)</a>
<a class="sourceLine" id="cb3-24" data-line-number="24"></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="kw">ggsave</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;Sex_RNAseq_eval.eps&quot;</span>))</a></code></pre></div>
</div>
<div id="predict-missing" class="section level2">
<h2>Predict missing</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">phenotyped &lt;-<span class="st"> </span><span class="kw">phenotyping</span>(phenotype, features, <span class="dt">methods =</span> <span class="st">&quot;lasso&quot;</span>, <span class="dt">ntop=</span><span class="dv">25</span>, <span class="dt">verbose=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">phenotyped<span class="op">$</span>validation</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">predicted &lt;-<span class="st"> </span>phenotyped<span class="op">$</span>predicted</a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="kw">rownames</span>(predicted) &lt;-<span class="st"> </span><span class="kw">colnames</span>(features)</a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="kw">write.table</span>(predicted, <span class="dt">file=</span><span class="kw">file.path</span>(path, <span class="st">&quot;Sex_RNAseq.csv&quot;</span>), <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>)</a></code></pre></div>
<blockquote>
<p>TODO add diagnostic plots!</p>
</blockquote>
</div>
<div id="inspect-selected-features" class="section level2">
<h2>Inspect selected features</h2>
<p>It could be interesting to see which genes are the top genes selected for predicting Age.</p>
<blockquote>
<p>this does not include e.g., the lasso selection</p>
</blockquote>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">library</span>(org.Hs.eg.db)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">map &lt;-<span class="st"> </span><span class="kw">select</span>(org.Hs.eg.db, <span class="dt">keys =</span> <span class="kw">names</span>(phenotyped<span class="op">$</span>top), <span class="dt">keytype=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="dt">columns=</span><span class="kw">c</span>(<span class="st">&quot;SYMBOL&quot;</span>, <span class="st">&quot;CHR&quot;</span>))</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">top &lt;-<span class="st"> </span><span class="kw">merge</span>(map, phenotyped<span class="op">$</span>top, <span class="dt">by.x=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="dt">by.y=</span><span class="st">&quot;row.names&quot;</span>)</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">top &lt;-<span class="st"> </span>top[<span class="kw">order</span>(<span class="kw">abs</span>(top<span class="op">$</span>y), <span class="dt">decreasing=</span><span class="ot">TRUE</span>),]</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">top</a>
<a class="sourceLine" id="cb5-6" data-line-number="6"></a>
<a class="sourceLine" id="cb5-7" data-line-number="7">mid &lt;-<span class="st"> </span><span class="kw">match</span>(top<span class="op">$</span>ENSEMBL, <span class="kw">rownames</span>(features))</a>
<a class="sourceLine" id="cb5-8" data-line-number="8">y &lt;-<span class="st"> </span>features[mid,]</a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="kw">rownames</span>(y) &lt;-<span class="st"> </span>top<span class="op">$</span>SYMBOL</a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="kw">rownames</span>(y)[<span class="kw">is.na</span>(top<span class="op">$</span>SYMBOL)] &lt;-<span class="st"> </span>top<span class="op">$</span>ENSEMBL[<span class="kw">is.na</span>(top<span class="op">$</span>SYMBOL)]</a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="kw">rownames</span>(y) &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="kw">rownames</span>(y), <span class="st">&quot; ( chr&quot;</span>, top<span class="op">$</span>CHR, <span class="st">&quot;)&quot;</span>)</a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="kw">colnames</span>(y) &lt;-<span class="st"> </span><span class="ot">NULL</span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="kw">library</span>(ComplexHeatmap)</a>
<a class="sourceLine" id="cb5-15" data-line-number="15">sex &lt;-<span class="st"> </span><span class="kw">colData</span>(cnts)<span class="op">$</span>Sex</a>
<a class="sourceLine" id="cb5-16" data-line-number="16">sex[<span class="kw">is.na</span>(sex)] &lt;-<span class="st"> &quot;Unknown&quot;</span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17">ha &lt;-<span class="st"> </span><span class="kw">HeatmapAnnotation</span>(<span class="dt">df =</span> <span class="kw">data.frame</span>(<span class="dt">Sex =</span> <span class="kw">factor</span>(sex)), <span class="dt">col =</span> <span class="kw">list</span>(<span class="dt">Sex =</span> <span class="kw">c</span>(<span class="st">&quot;Male&quot;</span> =<span class="st">  &quot;blue&quot;</span>, <span class="st">&quot;Female&quot;</span> =<span class="st"> &quot;pink&quot;</span>, <span class="st">&quot;Unknown&quot;</span> =<span class="st"> &quot;grey&quot;</span>)))</a>
<a class="sourceLine" id="cb5-18" data-line-number="18"></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="kw">setEPS</span>()</a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="kw">postscript</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;Sex_RNAseq_feats.eps&quot;</span>))</a>
<a class="sourceLine" id="cb5-21" data-line-number="21"><span class="kw">Heatmap</span>(y, <span class="dt">top_annotation =</span> ha, <span class="dt">name=</span><span class="st">&quot;log CPM&quot;</span>, <span class="dt">row_names_gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span> <span class="dv">8</span>))</a>
<a class="sourceLine" id="cb5-22" data-line-number="22"><span class="kw">dev.off</span>()</a></code></pre></div>
</div>
<div id="detect-outliers" class="section level2">
<h2>Detect outliers</h2>
<p>Outlier or missclassified sample could potentially be wrongly labeled.</p>
<p>Since, we use a subset for training it could be accidentally be missclassified. Therefore, we run multiple models and collected the number and frequencies of missclassified samples.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">library</span>(BiocParallel)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw">register</span>(<span class="kw">MulticoreParam</span>(<span class="dv">10</span>))</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">nrep &lt;-<span class="st"> </span><span class="dv">25</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4">predicted &lt;-<span class="st"> </span><span class="kw">bplapply</span>(<span class="dv">1</span><span class="op">:</span>nrep, <span class="cf">function</span>(x) <span class="kw">phenotyping</span>(phenotype, features, <span class="dt">methods =</span> <span class="st">&quot;lasso&quot;</span>, <span class="dt">ntop=</span><span class="dv">25</span>, <span class="dt">verbose=</span><span class="ot">FALSE</span>)<span class="op">$</span>predicted)</a>
<a class="sourceLine" id="cb6-5" data-line-number="5">predicted &lt;-<span class="st"> </span><span class="kw">do.call</span>(<span class="st">&#39;cbind&#39;</span>, predicted)</a>
<a class="sourceLine" id="cb6-6" data-line-number="6"></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="kw">rownames</span>(predicted) &lt;-<span class="st"> </span><span class="kw">names</span>(phenotype) &lt;-<span class="st"> </span><span class="kw">colnames</span>(features)</a>
<a class="sourceLine" id="cb6-8" data-line-number="8"></a>
<a class="sourceLine" id="cb6-9" data-line-number="9">missclassified &lt;-<span class="st"> </span>(predicted <span class="op">!=</span><span class="st"> </span>phenotype)[<span class="op">!</span><span class="kw">is.na</span>(phenotype), ]</a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="kw">rownames</span>(missclassified) &lt;-<span class="st"> </span><span class="kw">colnames</span>(features)[<span class="op">!</span><span class="kw">is.na</span>(phenotype)]</a>
<a class="sourceLine" id="cb6-11" data-line-number="11">missclassified &lt;-<span class="st"> </span>missclassified[<span class="kw">rowSums</span>(missclassified) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>,]</a>
<a class="sourceLine" id="cb6-12" data-line-number="12">missclass &lt;-<span class="st"> </span><span class="dv">100</span><span class="op">*</span><span class="kw">rowSums</span>(missclassified)<span class="op">/</span>nrep</a>
<a class="sourceLine" id="cb6-13" data-line-number="13">missclass[missclass <span class="op">&gt;</span><span class="st"> </span><span class="dv">50</span>]</a>
<a class="sourceLine" id="cb6-14" data-line-number="14"></a>
<a class="sourceLine" id="cb6-15" data-line-number="15">x &lt;-<span class="st"> </span><span class="kw">cbind</span>(predicted[<span class="kw">rownames</span>(predicted) <span class="op">%in%</span><span class="st"> </span><span class="kw">names</span>(missclass[missclass <span class="op">&gt;</span><span class="st"> </span><span class="dv">50</span>]), <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>],</a>
<a class="sourceLine" id="cb6-16" data-line-number="16">      <span class="dt">reported=</span><span class="kw">as.character</span>(phenotype[<span class="kw">names</span>(phenotype) <span class="op">%in%</span><span class="st"> </span><span class="kw">names</span>(missclass[missclass <span class="op">&gt;</span><span class="st"> </span><span class="dv">50</span>])])</a>
<a class="sourceLine" id="cb6-17" data-line-number="17">)</a>
<a class="sourceLine" id="cb6-18" data-line-number="18">ids &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;getIds&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB, <span class="dt">verbose=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="kw">subset</span>(ids, uuid <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(x))</a></code></pre></div>
</div>
</div>
<div id="predicting-age" class="section level1">
<h1>Predicting Age</h1>
<div id="model-selection-and-validation-1" class="section level2">
<h2>Model selection and Validation</h2>
<p>First we need to find out which algorithm and which number of features to use:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">metadata</span>(cnts)<span class="op">$</span>formula &lt;-<span class="st"> </span><span class="er">~</span><span class="dv">0</span><span class="op">+</span>Sampling_Age</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">phenotype &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">get_all_vars</span>(<span class="kw">metadata</span>(cnts)<span class="op">$</span>formula, <span class="dt">data=</span><span class="kw">colData</span>(cnts)))</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">features &lt;-<span class="st"> </span><span class="kw">assays</span>(cnts)<span class="op">$</span>data</a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="kw">summary</span>(phenotype)</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">                       </a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="kw">library</span>(BiocParallel)</a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="kw">register</span>(<span class="kw">MulticoreParam</span>(<span class="dv">10</span>, <span class="dt">log=</span><span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb7-8" data-line-number="8">validation &lt;-<span class="st"> </span><span class="kw">validate</span>(phenotype, features, <span class="dt">methods =</span> <span class="kw">c</span>(<span class="st">&quot;ridge&quot;</span>, <span class="st">&quot;lasso&quot;</span>, <span class="st">&quot;elastic-net&quot;</span>, <span class="st">&quot;pls&quot;</span>),</a>
<a class="sourceLine" id="cb7-9" data-line-number="9">                       <span class="dt">ntop =</span> <span class="kw">c</span>(<span class="dv">100</span>, <span class="dv">250</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">2500</span>, <span class="dv">5000</span>, <span class="dv">10000</span>), <span class="dt">nrep =</span> <span class="dv">25</span>, <span class="dt">verbose=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb7-10" data-line-number="10"></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="kw">library</span>(ggsci)</a>
<a class="sourceLine" id="cb7-13" data-line-number="13">gp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(validation, <span class="kw">aes</span>(<span class="dt">x=</span><span class="kw">as.factor</span>(ntop), <span class="dt">y=</span>value, <span class="dt">col=</span>method))</a>
<a class="sourceLine" id="cb7-14" data-line-number="14">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>()</a>
<a class="sourceLine" id="cb7-15" data-line-number="15">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>measure, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>, <span class="dt">nrow=</span><span class="dv">3</span>, <span class="dt">ncol=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb7-16" data-line-number="16">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Number of features&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Validation of predictors for `Age`&quot;</span>)</a>
<a class="sourceLine" id="cb7-17" data-line-number="17">gp <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_npg</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="dt">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>)</a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="kw">ggsave</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;Age_RNAseq_eval.eps&quot;</span>))</a></code></pre></div>
<p>some diagnostic plots:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">phenotyped &lt;-<span class="st"> </span><span class="kw">phenotyping</span>(phenotype, features, <span class="dt">methods =</span> <span class="st">&quot;lasso&quot;</span>, <span class="dt">ntop=</span><span class="dv">5000</span>, <span class="dt">verbose=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">predicted &lt;-<span class="st"> </span>phenotyped<span class="op">$</span>predicted</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="kw">rownames</span>(predicted) &lt;-<span class="st"> </span><span class="kw">colnames</span>(features)</a>
<a class="sourceLine" id="cb8-4" data-line-number="4"></a>
<a class="sourceLine" id="cb8-5" data-line-number="5">type &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">&quot;train&quot;</span>, <span class="kw">length</span>(phenotype))</a>
<a class="sourceLine" id="cb8-6" data-line-number="6">type[<span class="kw">is.na</span>(phenotype)] &lt;-<span class="st"> &quot;unknown&quot;</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7">type[phenotyped<span class="op">$</span>testid] &lt;-<span class="st"> &quot;test&quot;</span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"></a>
<a class="sourceLine" id="cb8-9" data-line-number="9">phenotype[<span class="kw">is.na</span>(phenotype)] &lt;-<span class="st"> </span>predicted[<span class="kw">is.na</span>(phenotype)]</a>
<a class="sourceLine" id="cb8-10" data-line-number="10"></a>
<a class="sourceLine" id="cb8-11" data-line-number="11">d &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">predicted=</span>predicted[,<span class="dv">1</span>], <span class="dt">measured=</span>phenotype, <span class="dt">type=</span><span class="kw">factor</span>(type))</a>
<a class="sourceLine" id="cb8-12" data-line-number="12"></a>
<a class="sourceLine" id="cb8-13" data-line-number="13">gp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(d, <span class="kw">aes</span>(<span class="dt">x=</span>predicted, <span class="dt">y=</span>measured, <span class="dt">col=</span>type))</a>
<a class="sourceLine" id="cb8-14" data-line-number="14">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_abline</span>(<span class="dt">intercept=</span><span class="dv">0</span>, <span class="dt">slope=</span><span class="dv">1</span>, <span class="dt">col=</span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Age prediction&quot;</span>)</a>
<a class="sourceLine" id="cb8-15" data-line-number="15">gp <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_npg</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="dt">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>)</a>
<a class="sourceLine" id="cb8-16" data-line-number="16"><span class="kw">ggsave</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;Age_RNAseq_diagn.eps&quot;</span>))</a></code></pre></div>
</div>
<div id="predict-missing-1" class="section level2">
<h2>Predict missing</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">phenotyped &lt;-<span class="st"> </span><span class="kw">phenotyping</span>(phenotype, features, <span class="dt">methods =</span> <span class="st">&quot;lasso&quot;</span>, <span class="dt">ntop=</span><span class="dv">5000</span>, <span class="dt">verbose=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">predicted &lt;-<span class="st"> </span>phenotyped<span class="op">$</span>predicted</a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="kw">rownames</span>(predicted) &lt;-<span class="st"> </span><span class="kw">colnames</span>(features)</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="kw">write.table</span>(predicted, <span class="dt">file=</span><span class="kw">file.path</span>(path, <span class="st">&quot;Age_RNAseq.csv&quot;</span>), <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>)</a></code></pre></div>
</div>
<div id="inspect-selected-features-1" class="section level2">
<h2>Inspect selected features</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw">library</span>(org.Hs.eg.db)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">map &lt;-<span class="st"> </span><span class="kw">select</span>(org.Hs.eg.db, <span class="dt">keys =</span> <span class="kw">names</span>(phenotyped<span class="op">$</span>top), <span class="dt">keytype=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="dt">columns=</span><span class="st">&quot;SYMBOL&quot;</span>)</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">top &lt;-<span class="st"> </span><span class="kw">merge</span>(map, phenotyped<span class="op">$</span>top, <span class="dt">by.x=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="dt">by.y=</span><span class="st">&quot;row.names&quot;</span>)</a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="kw">head</span>(top[<span class="kw">order</span>(<span class="kw">abs</span>(top<span class="op">$</span>y), <span class="dt">decreasing=</span><span class="ot">TRUE</span>),])</a>
<a class="sourceLine" id="cb10-5" data-line-number="5"></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="kw">library</span>(irlba)</a>
<a class="sourceLine" id="cb10-7" data-line-number="7">y &lt;-<span class="st"> </span>features[<span class="kw">rownames</span>(features) <span class="op">%in%</span><span class="st"> </span><span class="kw">names</span>(phenotyped<span class="op">$</span>top),]</a>
<a class="sourceLine" id="cb10-8" data-line-number="8">pc &lt;-<span class="st"> </span><span class="kw">prcomp_irlba</span>(<span class="kw">t</span>(y))</a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="kw">summary</span>(pc)</a>
<a class="sourceLine" id="cb10-10" data-line-number="10"></a>
<a class="sourceLine" id="cb10-11" data-line-number="11">cols &lt;-<span class="st"> </span><span class="kw">rainbow</span>(<span class="dv">10</span>)[<span class="kw">cut</span>(phenotype, <span class="dt">breaks=</span><span class="dv">10</span>)]</a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="kw">plot</span>(pc<span class="op">$</span>x, <span class="dt">col=</span><span class="kw">adjustcolor</span>(cols, <span class="dt">alpha.f=</span><span class="fl">0.3</span>), <span class="dt">pch=</span><span class="dv">16</span>)</a></code></pre></div>
</div>
<div id="detect-outliers-1" class="section level2">
<h2>Detect outliers</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">library</span>(BiocParallel)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw">register</span>(<span class="kw">MulticoreParam</span>(<span class="dv">10</span>))</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">nrep &lt;-<span class="st"> </span><span class="dv">100</span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4">predicted &lt;-<span class="st"> </span><span class="kw">bplapply</span>(<span class="dv">1</span><span class="op">:</span>nrep, <span class="cf">function</span>(x) <span class="kw">phenotyping</span>(phenotype, features, <span class="dt">methods =</span> <span class="st">&quot;lasso&quot;</span>, <span class="dt">ntop=</span><span class="dv">5000</span>, <span class="dt">verbose=</span><span class="ot">FALSE</span>)<span class="op">$</span>predicted)</a>
<a class="sourceLine" id="cb11-5" data-line-number="5">predicted &lt;-<span class="st"> </span><span class="kw">do.call</span>(<span class="st">&#39;cbind&#39;</span>, predicted)</a>
<a class="sourceLine" id="cb11-6" data-line-number="6"></a>
<a class="sourceLine" id="cb11-7" data-line-number="7">residuals &lt;-<span class="st"> </span>predicted <span class="op">-</span><span class="st"> </span>phenotype</a>
<a class="sourceLine" id="cb11-8" data-line-number="8"></a>
<a class="sourceLine" id="cb11-9" data-line-number="9">popo &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rowSums</span>(<span class="kw">abs</span>(residuals) <span class="op">&gt;</span><span class="st"> </span><span class="kw">mean</span>(residuals, <span class="dt">na.rm=</span><span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span><span class="dv">3</span><span class="op">*</span><span class="kw">sd</span>(residuals, <span class="dt">na.rm=</span><span class="ot">TRUE</span>)) <span class="op">&gt;</span><span class="st"> </span><span class="dv">50</span>)</a>
<a class="sourceLine" id="cb11-10" data-line-number="10"></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="kw">names</span>(residuals) &lt;-<span class="st"> </span><span class="kw">colnames</span>(features)</a>
<a class="sourceLine" id="cb11-12" data-line-number="12"></a>
<a class="sourceLine" id="cb11-13" data-line-number="13"><span class="kw">names</span>(residuals)[popo]</a>
<a class="sourceLine" id="cb11-14" data-line-number="14"></a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="kw">library</span>(matrixStats)</a>
<a class="sourceLine" id="cb11-16" data-line-number="16">mns &lt;-<span class="st"> </span><span class="kw">rowMeans</span>(predicted)</a>
<a class="sourceLine" id="cb11-17" data-line-number="17">sds &lt;-<span class="st"> </span><span class="kw">rowSds</span>(predicted)</a>
<a class="sourceLine" id="cb11-18" data-line-number="18">col &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">length</span>(mns))</a>
<a class="sourceLine" id="cb11-19" data-line-number="19">col[popo] &lt;-<span class="st"> </span><span class="dv">2</span></a>
<a class="sourceLine" id="cb11-20" data-line-number="20"><span class="kw">plot</span>(phenotype, mns, <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">cex=</span>sds, <span class="dt">col=</span><span class="kw">adjustcolor</span>(col, <span class="dt">alpha.f=</span><span class="fl">0.3</span>), <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;reported age&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;average predicted age&quot;</span>)</a>
<a class="sourceLine" id="cb11-21" data-line-number="21"><span class="kw">abline</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">col=</span><span class="dv">4</span>, <span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">lty=</span><span class="dv">2</span>)</a>
<a class="sourceLine" id="cb11-22" data-line-number="22"><span class="kw">grid</span>(<span class="dt">col=</span><span class="dv">1</span>)</a></code></pre></div>
</div>
</div>
<div id="predict-white-blood-cell-composition" class="section level1">
<h1>Predict white blood cell composition</h1>
<div id="model-fitting-and-model-selection" class="section level2">
<h2>Model fitting and model selection</h2>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">metadata</span>(cnts)<span class="op">$</span>formula &lt;-<span class="st"> </span><span class="er">~</span><span class="dv">0</span> <span class="op">+</span><span class="st"> </span>Lymph_Perc <span class="op">+</span><span class="st"> </span>Mono_Perc <span class="op">+</span><span class="st"> </span>Eos_Perc <span class="op">+</span><span class="st"> </span>Neut_Perc</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"></a>
<a class="sourceLine" id="cb12-3" data-line-number="3">phenotypes &lt;-<span class="st"> </span><span class="kw">get_all_vars</span>(<span class="kw">metadata</span>(cnts)<span class="op">$</span>formula, <span class="dt">data=</span><span class="kw">colData</span>(cnts))</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">features &lt;-<span class="st"> </span><span class="kw">assays</span>(cnts)<span class="op">$</span>data</a>
<a class="sourceLine" id="cb12-5" data-line-number="5"></a>
<a class="sourceLine" id="cb12-6" data-line-number="6">complete &lt;-<span class="st"> </span><span class="kw">apply</span>(phenotypes, <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="kw">all</span>(<span class="op">!</span><span class="kw">is.na</span>(x)))</a>
<a class="sourceLine" id="cb12-7" data-line-number="7"><span class="kw">table</span>(complete)</a>
<a class="sourceLine" id="cb12-8" data-line-number="8"></a>
<a class="sourceLine" id="cb12-9" data-line-number="9">trainId &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="kw">which</span>(complete), <span class="dt">size =</span> (<span class="dv">2</span><span class="op">/</span><span class="dv">3</span>)<span class="op">*</span><span class="kw">sum</span>(complete))</a>
<a class="sourceLine" id="cb12-10" data-line-number="10">testId &lt;-<span class="st"> </span><span class="kw">setdiff</span>(<span class="kw">which</span>(complete), trainId)</a>
<a class="sourceLine" id="cb12-11" data-line-number="11"></a>
<a class="sourceLine" id="cb12-12" data-line-number="12"><span class="kw">length</span>(trainId)</a>
<a class="sourceLine" id="cb12-13" data-line-number="13"><span class="kw">length</span>(testId)</a>
<a class="sourceLine" id="cb12-14" data-line-number="14"></a>
<a class="sourceLine" id="cb12-15" data-line-number="15">flowcell &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;-.*$&quot;</span>, <span class="st">&quot;&quot;</span>, cnts<span class="op">$</span>run_id)</a>
<a class="sourceLine" id="cb12-16" data-line-number="16">batches &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(<span class="op">~</span>flowcell)</a>
<a class="sourceLine" id="cb12-17" data-line-number="17"></a>
<a class="sourceLine" id="cb12-18" data-line-number="18"><span class="kw">library</span>(pls)</a>
<a class="sourceLine" id="cb12-19" data-line-number="19"><span class="kw">pls.options</span>(<span class="dt">parallel =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb12-20" data-line-number="20">predictor &lt;-<span class="st"> </span><span class="kw">plsr</span>(<span class="kw">log10</span>(phenotypes<span class="op">+</span><span class="dv">1</span>)<span class="op">~</span>batches<span class="op">+</span>data, <span class="dt">ncomp =</span> <span class="dv">50</span>,</a>
<a class="sourceLine" id="cb12-21" data-line-number="21">                  <span class="dt">data =</span> <span class="kw">list</span>(<span class="dt">phenotypes =</span> <span class="kw">as.matrix</span>(phenotypes[trainId,]),</a>
<a class="sourceLine" id="cb12-22" data-line-number="22">                              <span class="dt">batches =</span> batches[trainId, ],</a>
<a class="sourceLine" id="cb12-23" data-line-number="23">                              <span class="dt">data=</span><span class="kw">t</span>(features[, trainId])), <span class="dt">validation =</span> <span class="st">&quot;CV&quot;</span>, <span class="dt">keep.model =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb12-24" data-line-number="24"></a>
<a class="sourceLine" id="cb12-25" data-line-number="25"><span class="kw">str</span>(predictor)</a>
<a class="sourceLine" id="cb12-26" data-line-number="26"><span class="kw">summary</span>(predictor)</a>
<a class="sourceLine" id="cb12-27" data-line-number="27"><span class="kw">cumsum</span>(<span class="kw">explvar</span>(predictor))</a>
<a class="sourceLine" id="cb12-28" data-line-number="28"><span class="kw">barplot</span>(<span class="kw">cumsum</span>(<span class="kw">explvar</span>(predictor)), <span class="dt">las =</span> <span class="dv">2</span>, <span class="dt">ylab =</span> <span class="st">&quot;Cumulative Variance Explained&quot;</span>)</a>
<a class="sourceLine" id="cb12-29" data-line-number="29"></a>
<a class="sourceLine" id="cb12-30" data-line-number="30"><span class="kw">validationplot</span>(predictor, <span class="dt">val.type =</span> <span class="st">&quot;R2&quot;</span>, <span class="dt">ncomp =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">50</span>)</a>
<a class="sourceLine" id="cb12-31" data-line-number="31"><span class="kw">setEPS</span>()</a>
<a class="sourceLine" id="cb12-32" data-line-number="32"><span class="kw">postscript</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;WBCC_eval.eps&quot;</span>))</a>
<a class="sourceLine" id="cb12-33" data-line-number="33"><span class="kw">validationplot</span>(predictor, <span class="dt">val.type =</span> <span class="st">&quot;RMSEP&quot;</span>, <span class="dt">ncomp =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">50</span>)</a>
<a class="sourceLine" id="cb12-34" data-line-number="34"><span class="kw">dev.off</span>()</a></code></pre></div>
</div>
<div id="evaluate-predictions" class="section level2">
<h2>Evaluate predictions</h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">predicted &lt;-<span class="st"> </span><span class="kw">predict</span>(predictor, <span class="dt">newdata =</span> <span class="kw">list</span>(<span class="dt">data=</span><span class="kw">t</span>(features[, testId]), <span class="dt">batches =</span> batches[testId,]) , <span class="dt">ncomp =</span> <span class="dv">20</span>)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"></a>
<a class="sourceLine" id="cb13-3" data-line-number="3">predicted &lt;-<span class="st"> </span><span class="dv">10</span><span class="op">^</span>predicted <span class="op">-</span><span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb13-4" data-line-number="4"></a>
<a class="sourceLine" id="cb13-5" data-line-number="5"><span class="kw">library</span>(reshape2)</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">data &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="kw">melt</span>(phenotypes[testId,], <span class="dt">value.name=</span><span class="st">&quot;measured&quot;</span>),</a>
<a class="sourceLine" id="cb13-7" data-line-number="7">              <span class="dt">predicted=</span><span class="kw">melt</span>(predicted[,,<span class="dv">1</span>])[,<span class="dv">3</span>])</a>
<a class="sourceLine" id="cb13-8" data-line-number="8"></a>
<a class="sourceLine" id="cb13-9" data-line-number="9"><span class="kw">by</span>(data, data<span class="op">$</span>variable, <span class="cf">function</span>(x) <span class="kw">round</span>(<span class="kw">cor</span>(x<span class="op">$</span>measured, x<span class="op">$</span>predicted),<span class="dv">2</span>))</a>
<a class="sourceLine" id="cb13-10" data-line-number="10"></a>
<a class="sourceLine" id="cb13-11" data-line-number="11"><span class="co">##hack to get plot area square</span></a>
<a class="sourceLine" id="cb13-12" data-line-number="12">dummy &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">simplify2array</span>(<span class="kw">by</span>(data, data<span class="op">$</span>variable, <span class="cf">function</span>(x) <span class="kw">range</span>(<span class="kw">c</span>(x<span class="op">$</span>measured, x<span class="op">$</span>predicted)))))</a>
<a class="sourceLine" id="cb13-13" data-line-number="13">dummy &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">variable=</span><span class="kw">rep</span>(<span class="kw">rownames</span>(dummy), <span class="dv">2</span>), <span class="dt">measured=</span><span class="kw">as.vector</span>(dummy), <span class="dt">predicted=</span><span class="kw">as.vector</span>(dummy))</a>
<a class="sourceLine" id="cb13-14" data-line-number="14"></a>
<a class="sourceLine" id="cb13-15" data-line-number="15"><span class="co">##correlation plot</span></a>
<a class="sourceLine" id="cb13-16" data-line-number="16"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb13-17" data-line-number="17"><span class="kw">library</span>(ggsci)</a>
<a class="sourceLine" id="cb13-18" data-line-number="18">gp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(data, <span class="kw">aes</span>(<span class="dt">x=</span>measured, <span class="dt">y=</span>predicted))</a>
<a class="sourceLine" id="cb13-19" data-line-number="19">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_abline</span>(<span class="dt">intercept=</span><span class="dv">0</span>, <span class="dt">slope=</span><span class="dv">1</span>, <span class="dt">col=</span><span class="dv">2</span>)</a>
<a class="sourceLine" id="cb13-20" data-line-number="20">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">geom_blank</span>(<span class="dt">data=</span>dummy)</a>
<a class="sourceLine" id="cb13-21" data-line-number="21">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>variable, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>)</a>
<a class="sourceLine" id="cb13-22" data-line-number="22">gp <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_npg</span>()</a>
<a class="sourceLine" id="cb13-23" data-line-number="23"><span class="kw">ggsave</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;WBCC_RNAseq_diagn1.eps&quot;</span>))</a>
<a class="sourceLine" id="cb13-24" data-line-number="24"></a>
<a class="sourceLine" id="cb13-25" data-line-number="25"><span class="co">##bland altman plot</span></a>
<a class="sourceLine" id="cb13-26" data-line-number="26">tdata &lt;-<span class="st"> </span>data</a>
<a class="sourceLine" id="cb13-27" data-line-number="27">tdata[,<span class="dv">2</span>] &lt;-<span class="st"> </span><span class="kw">rowMeans</span>(data[,<span class="op">-</span><span class="dv">1</span>])</a>
<a class="sourceLine" id="cb13-28" data-line-number="28">tdata[,<span class="dv">3</span>] &lt;-<span class="st"> </span>data[,<span class="dv">2</span>] <span class="op">-</span><span class="st"> </span>data[,<span class="dv">3</span>]</a>
<a class="sourceLine" id="cb13-29" data-line-number="29"><span class="kw">colnames</span>(tdata)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Average&quot;</span>, <span class="st">&quot;Difference&quot;</span>)</a>
<a class="sourceLine" id="cb13-30" data-line-number="30"></a>
<a class="sourceLine" id="cb13-31" data-line-number="31">gp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(tdata, <span class="kw">aes</span>(<span class="dt">x=</span>Average, <span class="dt">y=</span>Difference))</a>
<a class="sourceLine" id="cb13-32" data-line-number="32">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_hline</span>(<span class="dt">yintercept=</span><span class="dv">0</span>, <span class="dt">col=</span><span class="dv">2</span>)</a>
<a class="sourceLine" id="cb13-33" data-line-number="33">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>variable, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>)</a>
<a class="sourceLine" id="cb13-34" data-line-number="34">gp <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_npg</span>()</a>
<a class="sourceLine" id="cb13-35" data-line-number="35"><span class="kw">ggsave</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;WBCC_RNAseq_diagn2.eps&quot;</span>))</a></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">predicted &lt;-<span class="st"> </span><span class="kw">predict</span>(predictor, <span class="dt">newdata =</span> <span class="kw">list</span>(<span class="dt">data=</span><span class="kw">t</span>(features), <span class="dt">batches =</span> batches), <span class="dt">ncomp =</span> <span class="dv">20</span>)[,,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">predicted &lt;-<span class="st"> </span><span class="dv">10</span><span class="op">^</span>predicted<span class="dv">-1</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="kw">hist</span>(<span class="kw">rowSums</span>(predicted), <span class="dt">n=</span><span class="dv">50</span>)</a>
<a class="sourceLine" id="cb14-5" data-line-number="5"></a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="kw">dim</span>(predicted)</a>
<a class="sourceLine" id="cb14-7" data-line-number="7"><span class="kw">rownames</span>(predicted) &lt;-<span class="st"> </span><span class="kw">colnames</span>(features)</a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="kw">head</span>(predicted)</a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="kw">write.table</span>(predicted, <span class="dt">file=</span><span class="kw">file.path</span>(path, <span class="st">&quot;WBCC_RNAseq.csv&quot;</span>),</a>
<a class="sourceLine" id="cb14-10" data-line-number="10">            <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>)</a></code></pre></div>
</div>
</div>
<div id="predicting-smoking" class="section level1">
<h1>Predicting Smoking</h1>
<div id="model-selection-and-validation-2" class="section level2">
<h2>Model Selection and Validation</h2>
<p>First we need to find out which algorithm and which number of features to use.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw">data</span>(rnaSeqData_ReadCounts_BIOS_cleaned)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="kw">dim</span>(counts)</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">cnts &lt;-<span class="st"> </span><span class="kw">preprocess</span>(counts)</a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="kw">dim</span>(cnts)</a>
<a class="sourceLine" id="cb15-5" data-line-number="5"></a>
<a class="sourceLine" id="cb15-6" data-line-number="6"><span class="kw">metadata</span>(cnts)<span class="op">$</span>formula &lt;-<span class="st"> </span><span class="er">~</span><span class="dv">0</span><span class="op">+</span>Smoking</a>
<a class="sourceLine" id="cb15-7" data-line-number="7"></a>
<a class="sourceLine" id="cb15-8" data-line-number="8">phenotype &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">get_all_vars</span>(<span class="kw">metadata</span>(cnts)<span class="op">$</span>formula, <span class="dt">data=</span><span class="kw">colData</span>(cnts)))</a>
<a class="sourceLine" id="cb15-9" data-line-number="9">features &lt;-<span class="st"> </span><span class="kw">assays</span>(cnts)<span class="op">$</span>data</a>
<a class="sourceLine" id="cb15-10" data-line-number="10"></a>
<a class="sourceLine" id="cb15-11" data-line-number="11"><span class="kw">round</span>(<span class="dv">100</span><span class="op">*</span><span class="kw">table</span>(phenotype, <span class="dt">useNA=</span><span class="st">&quot;always&quot;</span>)<span class="op">/</span><span class="kw">length</span>(phenotype),<span class="dv">2</span>)</a>
<a class="sourceLine" id="cb15-12" data-line-number="12"><span class="kw">levels</span>(phenotype) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;current smoker&quot;</span>, <span class="st">&quot;non-smoker&quot;</span>, <span class="st">&quot;non-smoker&quot;</span>)</a>
<a class="sourceLine" id="cb15-13" data-line-number="13"><span class="kw">round</span>(<span class="dv">100</span><span class="op">*</span><span class="kw">table</span>(phenotype, <span class="dt">useNA=</span><span class="st">&quot;always&quot;</span>)<span class="op">/</span><span class="kw">length</span>(phenotype),<span class="dv">2</span>)</a>
<a class="sourceLine" id="cb15-14" data-line-number="14"></a>
<a class="sourceLine" id="cb15-15" data-line-number="15"><span class="kw">library</span>(BiocParallel)</a>
<a class="sourceLine" id="cb15-16" data-line-number="16"><span class="kw">register</span>(<span class="kw">MulticoreParam</span>(<span class="dv">5</span>))</a>
<a class="sourceLine" id="cb15-17" data-line-number="17">validation &lt;-<span class="st"> </span><span class="kw">validate</span>(phenotype, features, <span class="dt">methods =</span> <span class="kw">c</span>(<span class="st">&quot;ridge&quot;</span>, <span class="st">&quot;elastic-net&quot;</span>, <span class="st">&quot;lasso&quot;</span>), <span class="dt">ntop =</span> <span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">25</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">250</span>, <span class="dv">500</span>, <span class="dv">5000</span>, <span class="dv">10000</span>), <span class="dt">nrep =</span> <span class="dv">25</span>, <span class="dt">verbose=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb15-18" data-line-number="18"></a>
<a class="sourceLine" id="cb15-19" data-line-number="19"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb15-20" data-line-number="20"><span class="kw">library</span>(ggsci)</a>
<a class="sourceLine" id="cb15-21" data-line-number="21">gp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">subset</span>(validation, measure <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;accuracy (Overall)&quot;</span>, <span class="st">&quot;f1 (non-smoker)&quot;</span>, <span class="st">&quot;f1 (current smoker)&quot;</span>)),</a>
<a class="sourceLine" id="cb15-22" data-line-number="22">             <span class="kw">aes</span>(<span class="dt">x=</span><span class="kw">as.factor</span>(ntop), <span class="dt">y=</span>value, <span class="dt">col=</span>method))</a>
<a class="sourceLine" id="cb15-23" data-line-number="23">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>()</a>
<a class="sourceLine" id="cb15-24" data-line-number="24">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>measure, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>, <span class="dt">nrow=</span><span class="dv">3</span>, <span class="dt">ncol=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb15-25" data-line-number="25">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Number of features&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Validation of predictors for `Smoking`&quot;</span>)</a>
<a class="sourceLine" id="cb15-26" data-line-number="26">gp <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_npg</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="dt">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>)</a>
<a class="sourceLine" id="cb15-27" data-line-number="27"></a>
<a class="sourceLine" id="cb15-28" data-line-number="28"><span class="kw">ggsave</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;Smoking_RNAseq_eval.eps&quot;</span>))</a></code></pre></div>
</div>
<div id="predict-missing-2" class="section level2">
<h2>Predict missing</h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">phenotyped &lt;-<span class="st"> </span><span class="kw">phenotyping</span>(phenotype, features, <span class="dt">methods =</span> <span class="st">&quot;elastic-net&quot;</span>, <span class="dt">ntop=</span><span class="dv">50</span>, <span class="dt">verbose=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">phenotyped<span class="op">$</span>validation</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">predicted &lt;-<span class="st"> </span>phenotyped<span class="op">$</span>predicted</a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="kw">rownames</span>(predicted) &lt;-<span class="st"> </span><span class="kw">colnames</span>(features)</a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="kw">write.table</span>(predicted, <span class="dt">file=</span><span class="kw">file.path</span>(path, <span class="st">&quot;Smoking_RNAseq.csv&quot;</span>), <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>)</a></code></pre></div>
<blockquote>
<p>TODO add diagnostic plots!</p>
</blockquote>
</div>
<div id="inspect-selected-features-2" class="section level2">
<h2>Inspect selected features</h2>
<p>It could be interesting to see which genes are the top genes selected for predicting Age.</p>
<blockquote>
<p>this does not include e.g., the lasso selection</p>
</blockquote>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw">library</span>(org.Hs.eg.db)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">map &lt;-<span class="st"> </span><span class="kw">select</span>(org.Hs.eg.db, <span class="dt">keys =</span> <span class="kw">names</span>(phenotyped<span class="op">$</span>top), <span class="dt">keytype=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="dt">columns=</span><span class="kw">c</span>(<span class="st">&quot;SYMBOL&quot;</span>, <span class="st">&quot;CHR&quot;</span>))</a>
<a class="sourceLine" id="cb17-3" data-line-number="3">top &lt;-<span class="st"> </span><span class="kw">merge</span>(map, phenotyped<span class="op">$</span>top, <span class="dt">by.x=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="dt">by.y=</span><span class="st">&quot;row.names&quot;</span>)</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">top &lt;-<span class="st"> </span>top[<span class="kw">order</span>(<span class="kw">abs</span>(top<span class="op">$</span>y), <span class="dt">decreasing=</span><span class="ot">TRUE</span>),]</a>
<a class="sourceLine" id="cb17-5" data-line-number="5">top</a>
<a class="sourceLine" id="cb17-6" data-line-number="6"></a>
<a class="sourceLine" id="cb17-7" data-line-number="7">mid &lt;-<span class="st"> </span><span class="kw">match</span>(top<span class="op">$</span>ENSEMBL, <span class="kw">rownames</span>(features))</a>
<a class="sourceLine" id="cb17-8" data-line-number="8">y &lt;-<span class="st"> </span>features[mid,]</a>
<a class="sourceLine" id="cb17-9" data-line-number="9"><span class="kw">rownames</span>(y) &lt;-<span class="st"> </span>top<span class="op">$</span>SYMBOL</a>
<a class="sourceLine" id="cb17-10" data-line-number="10"><span class="kw">rownames</span>(y)[<span class="kw">is.na</span>(top<span class="op">$</span>SYMBOL)] &lt;-<span class="st"> </span>top<span class="op">$</span>ENSEMBL[<span class="kw">is.na</span>(top<span class="op">$</span>SYMBOL)]</a>
<a class="sourceLine" id="cb17-11" data-line-number="11"><span class="kw">colnames</span>(y) &lt;-<span class="st"> </span><span class="ot">NULL</span></a>
<a class="sourceLine" id="cb17-12" data-line-number="12"></a>
<a class="sourceLine" id="cb17-13" data-line-number="13"><span class="kw">library</span>(ComplexHeatmap)</a>
<a class="sourceLine" id="cb17-14" data-line-number="14">smoking &lt;-<span class="st"> </span><span class="kw">as.character</span>(phenotype)</a>
<a class="sourceLine" id="cb17-15" data-line-number="15">smoking[<span class="kw">is.na</span>(smoking)] &lt;-<span class="st"> &quot;Unknown&quot;</span></a>
<a class="sourceLine" id="cb17-16" data-line-number="16">ha &lt;-<span class="st"> </span><span class="kw">HeatmapAnnotation</span>(<span class="dt">df =</span> <span class="kw">data.frame</span>(<span class="dt">Smoking =</span> <span class="kw">factor</span>(smoking)), <span class="dt">col =</span> <span class="kw">list</span>(<span class="dt">Smoking =</span> <span class="kw">c</span>(<span class="st">&quot;current smoker&quot;</span> =<span class="st">  &quot;blue&quot;</span>, <span class="st">&quot;non-smoker&quot;</span> =<span class="st"> &quot;pink&quot;</span>, <span class="st">&quot;Unknown&quot;</span> =<span class="st"> &quot;grey&quot;</span>)))</a>
<a class="sourceLine" id="cb17-17" data-line-number="17"></a>
<a class="sourceLine" id="cb17-18" data-line-number="18"><span class="kw">setEPS</span>()</a>
<a class="sourceLine" id="cb17-19" data-line-number="19"><span class="kw">postscript</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;Smoking_RNAseq_feats.eps&quot;</span>))</a>
<a class="sourceLine" id="cb17-20" data-line-number="20"><span class="kw">Heatmap</span>(y, <span class="dt">top_annotation =</span> ha, <span class="dt">name=</span><span class="st">&quot;log CPM&quot;</span>, <span class="dt">row_names_gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span> <span class="dv">8</span>))</a>
<a class="sourceLine" id="cb17-21" data-line-number="21"><span class="kw">dev.off</span>()</a></code></pre></div>
</div>
<div id="try-out-sparklyr" class="section level2">
<h2>Try out <code>sparklyr</code></h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="co">##setup spark context</span></a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="kw">library</span>(sparklyr)</a>
<a class="sourceLine" id="cb18-3" data-line-number="3">sc &lt;-<span class="st"> </span><span class="kw">spark_connect</span>(<span class="dt">master =</span> <span class="st">&quot;local&quot;</span>, <span class="dt">config =</span> <span class="kw">spark_config</span>(), <span class="dt">version =</span> <span class="st">&quot;1.6.2&quot;</span>)</a>
<a class="sourceLine" id="cb18-4" data-line-number="4"></a>
<a class="sourceLine" id="cb18-5" data-line-number="5"><span class="kw">metadata</span>(cnts)<span class="op">$</span>formula &lt;-<span class="st"> </span><span class="er">~</span><span class="dv">0</span><span class="op">+</span>Sampling_Age</a>
<a class="sourceLine" id="cb18-6" data-line-number="6">phenotype &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">get_all_vars</span>(<span class="kw">metadata</span>(cnts)<span class="op">$</span>formula, <span class="dt">data=</span><span class="kw">colData</span>(cnts)))</a>
<a class="sourceLine" id="cb18-7" data-line-number="7">features &lt;-<span class="st"> </span><span class="kw">assays</span>(cnts)<span class="op">$</span>data</a>
<a class="sourceLine" id="cb18-8" data-line-number="8"></a>
<a class="sourceLine" id="cb18-9" data-line-number="9"><span class="kw">library</span>(HDF5Array)</a>
<a class="sourceLine" id="cb18-10" data-line-number="10">smallMatrix &lt;-<span class="st"> </span><span class="kw">DelayedArray</span>(features)</a>
<a class="sourceLine" id="cb18-11" data-line-number="11"><span class="kw">class</span>(smallMatrix)</a>
<a class="sourceLine" id="cb18-12" data-line-number="12"></a>
<a class="sourceLine" id="cb18-13" data-line-number="13"><span class="kw">writeHDF5Dataset</span>(smallMatrix, <span class="dt">file=</span><span class="st">&quot;test.hdf5&quot;</span>, <span class="dt">name=</span><span class="st">&quot;features&quot;</span>)</a>
<a class="sourceLine" id="cb18-14" data-line-number="14"></a>
<a class="sourceLine" id="cb18-15" data-line-number="15">data &lt;-<span class="st"> </span><span class="kw">t</span>(features)</a>
<a class="sourceLine" id="cb18-16" data-line-number="16">data &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="dt">phenotype=</span>phenotype, data)</a>
<a class="sourceLine" id="cb18-17" data-line-number="17"><span class="kw">rownames</span>(data) &lt;-<span class="st"> </span><span class="kw">colnames</span>(features)</a>
<a class="sourceLine" id="cb18-18" data-line-number="18">data[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a>
<a class="sourceLine" id="cb18-19" data-line-number="19"><span class="kw">dim</span>(data)</a>
<a class="sourceLine" id="cb18-20" data-line-number="20"></a>
<a class="sourceLine" id="cb18-21" data-line-number="21"><span class="kw">library</span>(SparkR, <span class="dt">lib.loc=</span><span class="st">&quot;.cache/spark/spark-1.6.2-bin-hadoop2.6/R/lib/&quot;</span>)</a>
<a class="sourceLine" id="cb18-22" data-line-number="22"><span class="kw">library</span>(rhdf5)</a>
<a class="sourceLine" id="cb18-23" data-line-number="23">data &lt;-<span class="st"> </span><span class="kw">h5read</span>(<span class="st">&quot;test.hdf5&quot;</span>, <span class="dt">name=</span><span class="st">&quot;features&quot;</span>)</a>
<a class="sourceLine" id="cb18-24" data-line-number="24"></a>
<a class="sourceLine" id="cb18-25" data-line-number="25">x &lt;-<span class="st"> </span><span class="kw">as.DataFrame</span>(<span class="kw">data.frame</span>(data))</a>
<a class="sourceLine" id="cb18-26" data-line-number="26"></a>
<a class="sourceLine" id="cb18-27" data-line-number="27"><span class="kw">write.parquet</span>(data, <span class="dt">path=</span><span class="st">&quot;~/test.parquet&quot;</span>)</a>
<a class="sourceLine" id="cb18-28" data-line-number="28"></a>
<a class="sourceLine" id="cb18-29" data-line-number="29">temp_parquet &lt;-<span class="st"> </span><span class="kw">tempfile</span>(<span class="dt">fileext =</span> <span class="st">&quot;.parquet&quot;</span>)</a>
<a class="sourceLine" id="cb18-30" data-line-number="30"></a>
<a class="sourceLine" id="cb18-31" data-line-number="31">hdf5_to &lt;-<span class="st"> </span><span class="cf">function</span>(sc, file) {</a>
<a class="sourceLine" id="cb18-32" data-line-number="32">  <span class="kw">spark_context</span>(sc) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb18-33" data-line-number="33"><span class="st">    </span><span class="kw">invoke</span>(<span class="st">&quot;org.apache.spark.ml.Utils.loadHDF5&quot;</span>, file)</a>
<a class="sourceLine" id="cb18-34" data-line-number="34">}</a>
<a class="sourceLine" id="cb18-35" data-line-number="35"></a>
<a class="sourceLine" id="cb18-36" data-line-number="36"><span class="kw">hdf5_to</span>(sc, <span class="st">&quot;test.hdf5&quot;</span>)</a>
<a class="sourceLine" id="cb18-37" data-line-number="37"></a>
<a class="sourceLine" id="cb18-38" data-line-number="38"></a>
<a class="sourceLine" id="cb18-39" data-line-number="39">partitions &lt;-<span class="st"> </span>data <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb18-40" data-line-number="40"><span class="st">  </span><span class="kw">sdf_partition</span>(<span class="dt">training =</span> <span class="dv">2</span><span class="op">/</span><span class="dv">3</span>, <span class="dt">test =</span> <span class="dv">1</span><span class="op">/</span><span class="dv">3</span>, <span class="dt">seed =</span> <span class="dv">1099</span>)</a>
<a class="sourceLine" id="cb18-41" data-line-number="41"></a>
<a class="sourceLine" id="cb18-42" data-line-number="42">fit &lt;-<span class="st"> </span><span class="kw">ml_linear_regression</span>(<span class="dt">response =</span> <span class="st">&quot;phenotype&quot;</span>, <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;wt&quot;</span>, <span class="st">&quot;cyl&quot;</span>))</a>
<a class="sourceLine" id="cb18-43" data-line-number="43"></a>
<a class="sourceLine" id="cb18-44" data-line-number="44">alpha =<span class="st"> </span><span class="dv">0</span>,</a>
<a class="sourceLine" id="cb18-45" data-line-number="45">lambda =<span class="st"> </span><span class="dv">0</span></a></code></pre></div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
