<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Running a TWAS</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BBMRIomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="data.html">Data</a>
</li>
<li>
  <a href="usage.html">Basic Usage</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Use cases
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">DNA methylation</li>
    <li>
      <a href="EWAS.html">EWAS</a>
    </li>
    <li class="dropdown-header">RNA seq</li>
    <li>
      <a href="TWAS.html">TWAS</a>
    </li>
    <li class="dropdown-header">Metabolites</li>
    <li>
      <a href="MWAS.html">MWAS</a>
    </li>
    <li class="dropdown-header">Genotypes</li>
    <li>
      <a href="HRC.html">imputed genotypes</a>
    </li>
    <li class="dropdown-header">Integrated analyses</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Developers
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="metadatabase.html">Accessing the metadatabase</a>
    </li>
    <li>
      <a href="contribute.html">Contributing</a>
    </li>
    <li class="dropdown-header">Creating Datasets</li>
    <li>
      <a href="create_DNAm_Freeze2_unrelated_datasets.html">DNAm Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_RNAseq_Freeze2_unrelated_datasets.html">RNAseq Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_RNAseq_Freeze2_GoNL_dataset.html">RNAseq Freeze2 GoNL</a>
    </li>
    <li>
      <a href="create_GRCh38_RNAseq_Freeze2_unrelated_datasets.html">GRCh38 RNAseq Freeze2 unrelated</a>
    </li>
    <li>
      <a href="create_GRCh38_RNAseq_Freeze2_GoNL_dataset.html">GRCh38 RNAseq Freeze2 GoNL</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Developers (legacy)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Quality Control</li>
    <li>
      <a href="DNAm_qc.html">DNAm</a>
    </li>
    <li class="dropdown-header">RNAseq</li>
    <li class="dropdown-header">Creating Data Sets</li>
    <li>
      <a href="create_RNA_datasets.html">RNAseq</a>
    </li>
    <li>
      <a href="create_DNAm_datasets.html">DNAm</a>
    </li>
    <li>
      <a href="linkingRP4_RP3.html">Linking RP4 with RP3</a>
    </li>
    <li>
      <a href="freeze2.html">Determining freeze 2</a>
    </li>
    <li class="dropdown-header">Miscellaneous</li>
    <li>
      <a href="metadatabase_legacy.html">Accessing the metadatabase</a>
    </li>
    <li class="dropdown-header">Phenotyping</li>
    <li>
      <a href="phenotyping_rnaseq.html">Using RNAseq</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Running a TWAS</h1>

</div>


<p>A transcriptomewide association study can be run using the same approach as an EWAS.</p>
<div id="collect-data" class="section level2">
<h2>Collect data</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(BBMRIomics)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">bbmri.data</span>(rnaSeqData_ReadCounts_BIOS_Freeze2_unrelated)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">covariates &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;smoking&quot;</span>, <span class="st">&quot;biobank_id&quot;</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">phenotype &lt;-<span class="st"> &quot;sampling_age&quot;</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5">nas &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">as.data.frame</span>(<span class="kw">colData</span>(counts))[,<span class="kw">c</span>(phenotype, covariates)], <span class="dv">1</span>, anyNA)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">table</span>(nas)</a></code></pre></div>
<pre><code>## nas
## FALSE  TRUE 
##  3225   334</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">counts &lt;-<span class="st"> </span>counts[, <span class="op">!</span>nas]</a></code></pre></div>
</div>
<div id="create-design" class="section level2">
<h2>Create design</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">design &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(counts))[,<span class="kw">c</span>(phenotype, covariates)]</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw">str</span>(design)</a></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    3225 obs. of  4 variables:
##  $ sampling_age: int  43 65 33 51 43 77 49 53 27 55 ...
##  $ sex         : chr  &quot;female&quot; &quot;female&quot; &quot;male&quot; &quot;female&quot; ...
##  $ smoking     : chr  &quot;current smoker&quot; &quot;former-smoker&quot; &quot;non-smoker&quot; &quot;current smoker&quot; ...
##  $ biobank_id  : chr  &quot;LL&quot; &quot;LL&quot; &quot;LL&quot; &quot;LL&quot; ...</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">design<span class="op">$</span>smoking &lt;-<span class="st"> </span><span class="kw">factor</span>(design<span class="op">$</span>smoking)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw">str</span>(design)</a></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    3225 obs. of  4 variables:
##  $ sampling_age: int  43 65 33 51 43 77 49 53 27 55 ...
##  $ sex         : chr  &quot;female&quot; &quot;female&quot; &quot;male&quot; &quot;female&quot; ...
##  $ smoking     : Factor w/ 3 levels &quot;current smoker&quot;,..: 1 2 3 1 1 3 3 2 3 2 ...
##  $ biobank_id  : chr  &quot;LL&quot; &quot;LL&quot; &quot;LL&quot; &quot;LL&quot; ...</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">design &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(<span class="op">~</span>., design)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw">str</span>(design)</a></code></pre></div>
<pre><code>##  num [1:3225, 1:10] 1 1 1 1 1 1 1 1 1 1 ...
##  - attr(*, &quot;dimnames&quot;)=List of 2
##   ..$ : chr [1:3225] &quot;BIOS5537445E&quot; &quot;BIOSBFBBBED9&quot; &quot;BIOS3FB6E22E&quot; &quot;BIOSBEB8A07A&quot; ...
##   ..$ : chr [1:10] &quot;(Intercept)&quot; &quot;sampling_age&quot; &quot;sexmale&quot; &quot;smokingformer-smoker&quot; ...
##  - attr(*, &quot;assign&quot;)= int [1:10] 0 1 2 3 3 4 4 4 4 4
##  - attr(*, &quot;contrasts&quot;)=List of 3
##   ..$ sex       : chr &quot;contr.treatment&quot;
##   ..$ smoking   : chr &quot;contr.treatment&quot;
##   ..$ biobank_id: chr &quot;contr.treatment&quot;</code></pre>
</div>
<div id="run-twas" class="section level2">
<h2>Run TWAS</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">data &lt;-<span class="st"> </span><span class="kw">assays</span>(counts)<span class="op">$</span>data</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="kw">library</span>(edgeR)</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">y &lt;-<span class="st"> </span><span class="kw">DGEList</span>(data)</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">sds &lt;-<span class="st"> </span><span class="kw">apply</span>(y, <span class="dv">1</span>, sd)</a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="kw">quantile</span>(sds, <span class="dt">probs=</span><span class="fl">0.1</span>)</a></code></pre></div>
<pre><code>##        10% 
## 0.03520165</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">y &lt;-<span class="st"> </span>y[sds <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.04</span>,]</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">counts &lt;-<span class="st"> </span>counts[sds <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.04</span>, ] <span class="co">##for later use</span></a>
<a class="sourceLine" id="cb12-3" data-line-number="3">y &lt;-<span class="st"> </span><span class="kw">calcNormFactors</span>(y) <span class="co">##takes some time</span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4">data &lt;-<span class="st"> </span><span class="kw">cpm</span>(y, <span class="dt">log=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">library</span>(limma)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">fit &lt;-<span class="st"> </span><span class="kw">lmFit</span>(data, design)</a></code></pre></div>
</div>
<div id="inspect-results" class="section level2">
<h2>Inspect results</h2>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">tstat &lt;-<span class="st"> </span>fit<span class="op">$</span>coef<span class="op">/</span>fit<span class="op">$</span>stdev.unscaled<span class="op">/</span>fit<span class="op">$</span>sigma</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">pval &lt;-<span class="st"> </span><span class="dv">2</span> <span class="op">*</span><span class="st"> </span><span class="kw">pnorm</span>(<span class="op">-</span><span class="kw">abs</span>(tstat[, <span class="dv">2</span>]))</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">padj &lt;-<span class="st"> </span><span class="kw">p.adjust</span>(<span class="kw">sort</span>(pval, <span class="dt">decreasing =</span> <span class="ot">FALSE</span>), <span class="dt">method =</span> <span class="st">&quot;bonf&quot;</span>)</a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="kw">head</span>(padj[padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>])</a></code></pre></div>
<pre><code>## ENSG00000174807 ENSG00000163520 ENSG00000091129 ENSG00000110881 ENSG00000173114 
##   1.619307e-242   1.317583e-178   2.537973e-156   7.460167e-156   1.148739e-133 
## ENSG00000158966 
##   1.031009e-125</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">gp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">data.frame</span>(<span class="dt">pval=</span>pval), <span class="kw">aes</span>(<span class="dt">sample=</span><span class="op">-</span><span class="kw">log10</span>(pval)))</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">stat_qq</span>(<span class="dt">distribution=</span>stats<span class="op">::</span>qexp, <span class="dt">dparams=</span><span class="kw">list</span>(<span class="dt">rate=</span><span class="dv">1</span><span class="op">/</span><span class="kw">log10</span>(<span class="kw">exp</span>(<span class="dv">1</span>))))</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="kw">expression</span>(<span class="kw">paste</span>(<span class="st">&quot;Expected -log&quot;</span>[<span class="dv">10</span>], <span class="kw">plain</span>(P))))</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="kw">expression</span>(<span class="kw">paste</span>(<span class="st">&quot;Observed -log&quot;</span>[<span class="dv">10</span>], <span class="kw">plain</span>(P))))</a>
<a class="sourceLine" id="cb16-5" data-line-number="5">gp &lt;-<span class="st"> </span>gp <span class="op">+</span><span class="st"> </span><span class="kw">geom_abline</span>(<span class="dt">slope=</span><span class="dv">1</span>, <span class="dt">intercept=</span><span class="dv">0</span>)</a>
<a class="sourceLine" id="cb16-6" data-line-number="6">gp</a></code></pre></div>
<p><img src="TWAS_files/figure-html/qqplottwas-1.png" width="672" /></p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw">rowRanges</span>(counts)</a></code></pre></div>
<pre><code>## GRanges object with 49735 ranges and 1 metadata column:
##                   seqnames              ranges strand |         gene_id
##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;
##   ENSG00000000003        X   99883667-99894988      - | ENSG00000000003
##   ENSG00000000005        X   99839799-99854882      + | ENSG00000000005
##   ENSG00000000419       20   49551404-49575092      - | ENSG00000000419
##   ENSG00000000457        1 169821804-169863408      - | ENSG00000000457
##   ENSG00000000460        1 169631245-169823221      + | ENSG00000000460
##               ...      ...                 ...    ... .             ...
##   ENSG00000270177        5 133562101-133563518      + | ENSG00000270177
##   ENSG00000270178        3 179521851-179522154      + | ENSG00000270178
##   ENSG00000270179       11 113239200-113239839      + | ENSG00000270179
##   ENSG00000270182        7   27197963-27198595      + | ENSG00000270182
##   ENSG00000270184       16   85817988-85821223      + | ENSG00000270184
##   -------
##   seqinfo: 25 sequences (1 circular) from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">head</span>(pval)</a></code></pre></div>
<pre><code>## ENSG00000000003 ENSG00000000005 ENSG00000000419 ENSG00000000457 ENSG00000000460 
##    7.703488e-06    3.841097e-01    3.300884e-01    3.824680e-06    6.736994e-01 
## ENSG00000000938 
##    5.524164e-14</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw">tail</span>(pval)</a></code></pre></div>
<pre><code>## ENSG00000270175 ENSG00000270177 ENSG00000270178 ENSG00000270179 ENSG00000270182 
##     0.227680991     0.283357183     0.195873204     0.950020906     0.655623788 
## ENSG00000270184 
##     0.001032878</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">rData &lt;-<span class="st"> </span><span class="kw">rowRanges</span>(counts)</a>
<a class="sourceLine" id="cb23-2" data-line-number="2"><span class="kw">mcols</span>(rData)<span class="op">$</span>pval &lt;-<span class="st"> </span>pval <span class="co">##possible since we subsetted counts already</span></a>
<a class="sourceLine" id="cb23-3" data-line-number="3">rData</a></code></pre></div>
<pre><code>## GRanges object with 49735 ranges and 2 metadata columns:
##                   seqnames              ranges strand |         gene_id
##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;
##   ENSG00000000003        X   99883667-99894988      - | ENSG00000000003
##   ENSG00000000005        X   99839799-99854882      + | ENSG00000000005
##   ENSG00000000419       20   49551404-49575092      - | ENSG00000000419
##   ENSG00000000457        1 169821804-169863408      - | ENSG00000000457
##   ENSG00000000460        1 169631245-169823221      + | ENSG00000000460
##               ...      ...                 ...    ... .             ...
##   ENSG00000270177        5 133562101-133563518      + | ENSG00000270177
##   ENSG00000270178        3 179521851-179522154      + | ENSG00000270178
##   ENSG00000270179       11 113239200-113239839      + | ENSG00000270179
##   ENSG00000270182        7   27197963-27198595      + | ENSG00000270182
##   ENSG00000270184       16   85817988-85821223      + | ENSG00000270184
##                                   pval
##                              &lt;numeric&gt;
##   ENSG00000000003 7.70348751228328e-06
##   ENSG00000000005    0.384109664189747
##   ENSG00000000419     0.33008844650229
##   ENSG00000000457  3.8246797680786e-06
##   ENSG00000000460     0.67369941795068
##               ...                  ...
##   ENSG00000270177    0.283357183163476
##   ENSG00000270178    0.195873204341671
##   ENSG00000270179     0.95002090589992
##   ENSG00000270182    0.655623787505276
##   ENSG00000270184  0.00103287838785457
##   -------
##   seqinfo: 25 sequences (1 circular) from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw">library</span>(GenomeInfoDb)</a>
<a class="sourceLine" id="cb25-2" data-line-number="2"><span class="kw">seqlevels</span>(rData) &lt;-<span class="st"> </span><span class="kw">mapSeqlevels</span>(<span class="kw">seqlevels</span>(rData), <span class="st">&quot;Ensembl&quot;</span>)</a>
<a class="sourceLine" id="cb25-3" data-line-number="3">rData</a></code></pre></div>
<pre><code>## GRanges object with 49735 ranges and 2 metadata columns:
##                   seqnames              ranges strand |         gene_id
##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;
##   ENSG00000000003        X   99883667-99894988      - | ENSG00000000003
##   ENSG00000000005        X   99839799-99854882      + | ENSG00000000005
##   ENSG00000000419       20   49551404-49575092      - | ENSG00000000419
##   ENSG00000000457        1 169821804-169863408      - | ENSG00000000457
##   ENSG00000000460        1 169631245-169823221      + | ENSG00000000460
##               ...      ...                 ...    ... .             ...
##   ENSG00000270177        5 133562101-133563518      + | ENSG00000270177
##   ENSG00000270178        3 179521851-179522154      + | ENSG00000270178
##   ENSG00000270179       11 113239200-113239839      + | ENSG00000270179
##   ENSG00000270182        7   27197963-27198595      + | ENSG00000270182
##   ENSG00000270184       16   85817988-85821223      + | ENSG00000270184
##                                   pval
##                              &lt;numeric&gt;
##   ENSG00000000003 7.70348751228328e-06
##   ENSG00000000005    0.384109664189747
##   ENSG00000000419     0.33008844650229
##   ENSG00000000457  3.8246797680786e-06
##   ENSG00000000460     0.67369941795068
##               ...                  ...
##   ENSG00000270177    0.283357183163476
##   ENSG00000270178    0.195873204341671
##   ENSG00000270179     0.95002090589992
##   ENSG00000270182    0.655623787505276
##   ENSG00000270184  0.00103287838785457
##   -------
##   seqinfo: 25 sequences (1 circular) from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1"><span class="kw">seqlevels</span>(rData) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;X&quot;</span>, <span class="dv">23</span>, <span class="kw">seqlevels</span>(rData))</a>
<a class="sourceLine" id="cb27-2" data-line-number="2"><span class="kw">seqlevels</span>(rData) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;Y&quot;</span>, <span class="dv">24</span>, <span class="kw">seqlevels</span>(rData))</a>
<a class="sourceLine" id="cb27-3" data-line-number="3">pdata &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(rData)</a>
<a class="sourceLine" id="cb27-4" data-line-number="4">pdata<span class="op">$</span>seqnames &lt;-<span class="st"> </span><span class="kw">as.integer</span>(pdata<span class="op">$</span>seqnames)</a>
<a class="sourceLine" id="cb27-5" data-line-number="5"></a>
<a class="sourceLine" id="cb27-6" data-line-number="6"><span class="kw">library</span>(qqman)</a>
<a class="sourceLine" id="cb27-7" data-line-number="7"><span class="kw">manhattan</span>(pdata, <span class="dt">chr=</span><span class="st">&quot;seqnames&quot;</span>, <span class="dt">bp=</span><span class="st">&quot;start&quot;</span>, <span class="dt">p=</span><span class="st">&quot;pval&quot;</span>)</a></code></pre></div>
<pre><code>## Warning in manhattan(pdata, chr = &quot;seqnames&quot;, bp = &quot;start&quot;, p = &quot;pval&quot;): No SNP
## column found. OK unless you&#39;re trying to highlight.</code></pre>
<p><img src="TWAS_files/figure-html/manplottwas-1.png" width="672" /></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">top &lt;-<span class="st"> </span><span class="kw">which.min</span>(rData<span class="op">$</span>pval)</a>
<a class="sourceLine" id="cb29-2" data-line-number="2">y &lt;-<span class="st"> </span>data[top, ]</a>
<a class="sourceLine" id="cb29-3" data-line-number="3">x &lt;-<span class="st"> </span>design[, <span class="dv">2</span>]</a>
<a class="sourceLine" id="cb29-4" data-line-number="4"><span class="kw">plot</span>(x, y, <span class="dt">main =</span> <span class="kw">paste</span>(<span class="st">&quot;Pearson cor.:&quot;</span>, <span class="kw">signif</span>(<span class="kw">cor</span>(x, y), <span class="dv">3</span>)), <span class="dt">xlab =</span> <span class="st">&quot;Age (in yrs)&quot;</span>,</a>
<a class="sourceLine" id="cb29-5" data-line-number="5">     <span class="dt">ylab =</span> <span class="kw">paste</span>(<span class="kw">rownames</span>(data)[top], <span class="st">&quot;Expression (TMM norm. log cpm)&quot;</span>))</a>
<a class="sourceLine" id="cb29-6" data-line-number="6"><span class="kw">abline</span>(<span class="kw">lm</span>(y <span class="op">~</span><span class="st"> </span>x), <span class="dt">col =</span> <span class="dv">2</span>)</a></code></pre></div>
<p><img src="TWAS_files/figure-html/tophittwas-1.png" width="672" /></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
