#' DNA methylation data (beta-values) for the CODAM cohort (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Betas_CODAM_F2_cleaned
#' @usage data(methData_Betas_CODAM_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Betas_CODAM_F2_cleaned)
#' betas
methData_Betas_CODAM_F2_cleaned <- NULL

#' DNA methylation data (M-values) for the CODAM cohort (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Mvalues_CODAM_F2_cleaned
#' @usage data(methData_Mvalues_CODAM_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Mvalues_CODAM_F2_cleaned)
#' mvalues
methData_Mvalues_CODAM_F2_cleaned <- NULL

#' DNA methylation data (beta-values) for the LL cohort (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Betas_LL_F2_cleaned
#' @usage data(methData_Betas_LL_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Betas_LL_F2_cleaned)
#' betas
methData_Betas_LL_F2_cleaned <- NULL

#' DNA methylation data (M-values) for the LL cohort (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Mvalues_LL_F2_cleaned
#' @usage data(methData_Mvalues_LL_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Mvalues_LL_F2_cleaned)
#' mvalues
methData_Mvalues_LL_F2_cleaned <- NULL

#' DNA methylation data (beta-values) for the LLS cohort (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Betas_LLS_F2_cleaned
#' @usage data(methData_Betas_LLS_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Betas_LLS_F2_cleaned)
#' betas
methData_Betas_LLS_F2_cleaned <- NULL

#' DNA methylation data (M-values) for the LLS cohort (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Mvalues_LLS_F2_cleaned
#' @usage data(methData_Mvalues_LLS_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Mvalues_LLS_F2_cleaned)
#' mvalues
methData_Mvalues_LLS_F2_cleaned <- NULL

#' DNA methylation data (beta-values) for the NTR cohort (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Betas_NTR_F2_cleaned
#' @usage data(methData_Betas_NTR_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Betas_NTR_F2_cleaned)
#' betas
methData_Betas_NTR_F2_cleaned <- NULL

#' DNA methylation data (M-values) for the NTR cohort (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Mvalues_NTR_F2_cleaned
#' @usage data(methData_Mvalues_NTR_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Mvalues_NTR_F2_cleaned)
#' mvalues
methData_Mvalues_NTR_F2_cleaned <- NULL

#' DNA methylation data (beta-values) for the RS cohort (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Betas_RS_F2_cleaned
#' @usage data(methData_Betas_RS_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Betas_RS_F2_cleaned)
#' betas
methData_Betas_RS_F2_cleaned <- NULL

#' DNA methylation data (M-values) for the RS cohort (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Mvalues_RS_F2_cleaned
#' @usage data(methData_Mvalues_RS_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Mvalues_RS_F2_cleaned)
#' mvalues
methData_Mvalues_RS_F2_cleaned <- NULL

#' DNA methylation data (beta-values) for the PAN cohort (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Betas_PAN_F2_cleaned
#' @usage data(methData_Betas_PAN_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Betas_PAN_F2_cleaned)
#' betas
methData_Betas_PAN_F2_cleaned <- NULL

#' DNA methylation data (M-values) for the PAN cohort (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Mvalues_PAN_F2_cleaned
#' @usage data(methData_Mvalues_PAN_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Mvalues_PAN_F2_cleaned)
#' mvalues
methData_Mvalues_PAN_F2_cleaned <- NULL

#' DNA methylation data (beta-values) for BIOS (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Betas_BIOS_F2_cleaned
#' @usage data(methData_Betas_BIOS_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Betas_BIOS_F2_cleaned)
#' betas
methData_Betas_BIOS_F2_cleaned <- NULL

#' DNA methylation data (M-values) for BIOS (Outdated)
#'
#' The generation of genome-wide DNA methylation data is described by Bonder et al. (doi:10.1038/ng.3721).
#' Briefly, 500 ng of genomic DNA was bisulfite modified using the EZ DNA Methylation kit (Zymo Research, Irvine, CA, USA)
#' and hybridized on Illumina 450K arrays according to the manufacturer's protocols.
#' The original idat files were generated by the Illumina iScan BeadChip scanner.
#' Subsequently, sample quality control was performed using MethylAid.
#' Probes with a high detection P value (> 0.01), probes with a low bead count (< 3 beads), and probes with a low success rate
#' (missing in > 95% of the samples) were set to missing. Samples containing an excess of missing probes (> 5%) were excluded
#' from the analysis. Subsequently, per cohort, imputation was performed to impute the missing values. Functional normalization,
#' as implemented in the minfi package , was used per cohort.
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name methData_Mvalues_BIOS_F2_cleaned
#' @usage data(methData_Mvalues_BIOS_F2_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(methData_Mvalues_BIOS_F2_cleaned)
#' mvalues
methData_Mvalues_BIOS_F2_cleaned <- NULL

#' RNAseq data for the CODAM cohort  (Outdated)
#'
#' A detailed description of the RNA-seq data processing can be found in Zhernakova et al. (doi:10.1038/ng.3737).
#' Briefly, total RNA from whole blood, depleted of globin transcripts, was sequenced (2x50-bp) using the Illumina
#' HiSeq 2000 platform, and read alignment was performed using STAR (v2.3.0).
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name rnaSeqData_ReadCounts_CODAM_cleaned
#' @usage data(rnaSeqData_ReadCounts_CODAM_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(rnaSeqData_ReadCounts_CODAM_cleaned)
#' counts
rnaSeqData_ReadCounts_CODAM_cleaned <- NULL

#' RNAseq data for the LL cohort (Outdated)
#'
#' A detailed description of the RNA-seq data processing can be found in Zhernakova et al. (doi:10.1038/ng.3737).
#' Briefly, total RNA from whole blood, depleted of globin transcripts, was sequenced (2x50-bp) using the Illumina
#' HiSeq 2000 platform, and read alignment was performed using STAR (v2.3.0).
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name rnaSeqData_ReadCounts_LL_cleaned
#' @usage data(rnaSeqData_ReadCounts_LL_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(rnaSeqData_ReadCounts_LL_cleaned)
#' counts
rnaSeqData_ReadCounts_LL_cleaned <- NULL

#' RNAseq data for the LLS cohort (Outdated)
#'
#' A detailed description of the RNA-seq data processing can be found in Zhernakova et al. (doi:10.1038/ng.3737).
#' Briefly, total RNA from whole blood, depleted of globin transcripts, was sequenced (2x50-bp) using the Illumina
#' HiSeq 2000 platform, and read alignment was performed using STAR (v2.3.0).
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name rnaSeqData_ReadCounts_LLS_cleaned
#' @usage data(rnaSeqData_ReadCounts_LLS_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(rnaSeqData_ReadCounts_LLS_cleaned)
#' counts
rnaSeqData_ReadCounts_LLS_cleaned <- NULL

#' RNAseq data for the NTR cohort (Outdated)
#'
#' A detailed description of the RNA-seq data processing can be found in Zhernakova et al. (doi:10.1038/ng.3737).
#' Briefly, total RNA from whole blood, depleted of globin transcripts, was sequenced (2x50-bp) using the Illumina
#' HiSeq 2000 platform, and read alignment was performed using STAR (v2.3.0).
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name rnaSeqData_ReadCounts_NTR_cleaned
#' @usage data(rnaSeqData_ReadCounts_NTR_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(rnaSeqData_ReadCounts_NTR_cleaned)
#' counts
rnaSeqData_ReadCounts_NTR_cleaned <- NULL

#' RNAseq data for the PAN cohort (Outdated)
#'
#' A detailed description of the RNA-seq data processing can be found in Zhernakova et al. (doi:10.1038/ng.3737).
#' Briefly, total RNA from whole blood, depleted of globin transcripts, was sequenced (2x50-bp) using the Illumina
#' HiSeq 2000 platform, and read alignment was performed using STAR (v2.3.0).
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name rnaSeqData_ReadCounts_PAN_cleaned
#' @usage data(rnaSeqData_ReadCounts_PAN_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(rnaSeqData_ReadCounts_PAN_cleaned)
#' counts
rnaSeqData_ReadCounts_PAN_cleaned <- NULL

#' RNAseq data for the RS cohort (Outdated)
#'
#' A detailed description of the RNA-seq data processing can be found in Zhernakova et al. (doi:10.1038/ng.3737).
#' Briefly, total RNA from whole blood, depleted of globin transcripts, was sequenced (2x50-bp) using the Illumina
#' HiSeq 2000 platform, and read alignment was performed using STAR (v2.3.0).
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name rnaSeqData_ReadCounts_RS_cleaned
#' @usage data(rnaSeqData_ReadCounts_RS_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(rnaSeqData_ReadCounts_RS_cleaned)
#' counts
rnaSeqData_ReadCounts_RS_cleaned <- NULL

#' RNAseq data for BIOS (Outdated)
#'
#' A detailed description of the RNA-seq data processing can be found in Zhernakova et al. (doi:10.1038/ng.3737).
#' Briefly, total RNA from whole blood, depleted of globin transcripts, was sequenced (2x50-bp) using the Illumina
#' HiSeq 2000 platform, and read alignment was performed using STAR (v2.3.0).
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name rnaSeqData_ReadCounts_BIOS_cleaned
#' @usage data(rnaSeqData_ReadCounts_BIOS_cleaned)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(rnaSeqData_ReadCounts_BIOS_cleaned)
#' counts
rnaSeqData_ReadCounts_BIOS_cleaned <- NULL


#' RNAseq data for BIOS GoNL subset (Outdated)
#'
#' A detailed description of the RNA-seq data processing can be found in Zhernakova et al. (doi:10.1038/ng.3737).
#' Briefly, total RNA from whole blood, depleted of globin transcripts, was sequenced (2x50-bp) using the Illumina
#' HiSeq 2000 platform, and read alignment was performed using STAR (v2.3.0).
#'
#' DNA methylation data and RNA-seq data were generated within the Biobank-based Integrative Omics Studies Consortium (http://wiki.bbmri.nl/wiki/BIOS_start-).
#' The data comprise six biobanks: Cohort on Diabetes and Atherosclerosis Maastricht (CODAM, n~180)
#' (doi: 10.1111/j.1365-2362.2010.02418.x), LifeLines (LL, n~700) (doi: 10.1136/bmjopen-2014-006772),
#' the Leiden Longevity Study (LLS, n~600) (doi: 10.1111/j.1532-5415.2009.02381.x),
#' the Rotterdam Study (RS, n~600) (doi: 10.1007/s10654-011-9610-5),
#' the Netherlands Twin Registry (NTR, n~1800) (doi:  10.1017/thg.2012.140) and
#' the Prospective ALS Study Netherlands (PAN, n~180) (doi: 10.1136/jnnp.2011.244939).
#' Sample identity of DNA methylation and gene expression data was confirmed using genotype data.
#'
#' Data were generated by the Human Genotyping facility (HugeF) of ErasmusMC, the Netherlands (www.glimDNA.org).
#'
#' @docType data
#' @name rnaSeqData_ReadCounts_GoNL
#' @usage data(rnaSeqData_ReadCounts_GoNL)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(rnaSeqData_ReadCounts_GoNL)
#' counts
rnaSeqData_ReadCounts_GoNL <- NULL

#' RP4 metabolites
#'
#' add description
#'
#' @docType data
#' @name metabolomics_RP3RP4_overlap
#' @usage data(metabolomics_RP3RP4_overlap)
#'
#' @format An object of class \code{"SummarizedExperiment"}; see \code{\link[SummarizedExperiment]{SummarizedExperiment}}.
#'
#' @keywords datasets
#' @export
#' @examples
#' data(metabolomics_RP3RP4_overlap)
#' metabolomicData
metabolomics_RP3RP4_overlap <- NULL
