<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>RNA</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BBMRIomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="data.html">Data</a>
</li>
<li>
  <a href="usage.html">Basic Usage</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Use cases
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">DNA methylation</li>
    <li>
      <a href="EWAS.html">EWAS</a>
    </li>
    <li class="dropdown-header">RNA seq</li>
    <li>
      <a href="TWAS.html">TWAS</a>
    </li>
    <li class="dropdown-header">Metabolites</li>
    <li>
      <a href="MWAS.html">MWAS</a>
    </li>
    <li class="dropdown-header">Genotypes</li>
    <li>
      <a href="HRC.html">imputed genotypes</a>
    </li>
    <li>
      <a href="GoNL.html">GoNL</a>
    </li>
    <li class="dropdown-header">Integrated analyses</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Apps
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="http://bios-vm.bbmrirp3-lumc.surf-hosted.nl:8008/BBMRIomics/DNAm/CODAM">DNam qc CODAM</a>
    </li>
    <li>
      <a href="http://bios-vm.bbmrirp3-lumc.surf-hosted.nl:8008/BBMRIomics/DNAm/LL">DNam qc LL</a>
    </li>
    <li>
      <a href="http://bios-vm.bbmrirp3-lumc.surf-hosted.nl:8008/BBMRIomics/DNAm/LLS">DNam qc LLS</a>
    </li>
    <li>
      <a href="http://bios-vm.bbmrirp3-lumc.surf-hosted.nl:8008/BBMRIomics/DNAm/NTR">DNam qc NTR</a>
    </li>
    <li>
      <a href="http://bios-vm.bbmrirp3-lumc.surf-hosted.nl:8008/BBMRIomics/DNAm/RS">DNam qc RS</a>
    </li>
    <li>
      <a href="http://bios-vm.bbmrirp3-lumc.surf-hosted.nl:8008/BBMRIomics/DNAm/PAN">DNam qc PAN</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Developers
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Quality Control</li>
    <li>
      <a href="DNAm_qc.html">DNAm</a>
    </li>
    <li>
      <a href="RNA_qc.html">RNAseq</a>
    </li>
    <li class="dropdown-header">Creating Data Sets</li>
    <li>
      <a href="create_RNA_datasets.html">RNAseq</a>
    </li>
    <li>
      <a href="create_DNAm_datasets.html">DNAm</a>
    </li>
    <li>
      <a href="linkingRP4_RP3.html">Linking RP4 with RP3</a>
    </li>
    <li>
      <a href="freeze2.html">Determining freeze 2</a>
    </li>
    <li class="dropdown-header">Miscellaneous</li>
    <li>
      <a href="metadatabase.html">Accessing the metadatabase</a>
    </li>
    <li>
      <a href="contribute.html">Contributing</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">RNA</h1>

</div>


<!-- #END OF FILE. -->
<div id="library-preparation-and-alignment" class="section level1">
<h1>Library preparation and alignment</h1>
<p>Detailed description of the RNASeq data generation and preprocessing can be found in the supplementary material of the paper by Zhernakova et al. <span class="citation">(D. V. Zhernakova et al. 2017)</span></p>
</div>
<div id="feature-quantification-read-counts" class="section level1">
<h1>Feature quantification read counts</h1>
<p>Gene read counts are generated using <a href="http://www-huber.embl.de/HTSeq/doc/count.html">htseq</a> using mode <code>union</code> for handling overlapping exons.</p>
<p>Reference annotation used is from ensembl: <a href="http://ftp.ensembl.org/pub/release-71/gtf/homo_sapiens/Homo_sapiens.GRCh37.71.gtf.gz%22"></a> but only chromosomes 1-22 and X, Y (and MT) are retained.</p>
<div id="prepare-gene-annotation" class="section level2">
<h2>Prepare gene annotation</h2>
<p>For convenience we create a <code>TxDb</code>-package. For more information on using a <code>TxDb</code>-package see <a href="http://bioconductor.org/packages/release/bioc/vignettes/GenomicFeatures/inst/doc/GenomicFeatures.pdf">GenomicFeatures</a>.</p>
<p>*** Note this requires sudo rights to install and use sudo -i to load /etc/profile with global R_SITE_LIB =/opt/… ***</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(GenomicFeatures)
<span class="kw">library</span>(rtracklayer)
gtf &lt;-<span class="st"> &quot;ftp://ftp.ensembl.org/pub/release-71/gtf/homo_sapiens/Homo_sapiens.GRCh37.71.gtf.gz&quot;</span>
txdb &lt;-<span class="st"> </span><span class="kw">makeTxDbFromGFF</span>(gtf, <span class="dt">organism=</span><span class="st">&quot;Homo sapiens&quot;</span>)
##Hack by James MacDonald (https://support.bioconductor.org/p/89855/)
con &lt;-<span class="st"> </span><span class="kw">dbconn</span>(txdb)
## insert a Resource URL entry in the metadata table,
## pointing to the ftp site where the data were procured
DBI::<span class="kw">dbGetQuery</span>(con, <span class="st">&quot;INSERT INTO metadata VALUES (&#39;Resource URL&#39;, &#39;ftp://ftp.ensembl.org/pub/release-71/gtf/homo_sapiens/Homo_sapiens.GRCh37.71.gtf.gz&#39;);&quot;</span>)
<span class="kw">makeTxDbPackage</span>(txdb,
                <span class="dt">version =</span> <span class="st">&quot;0.0.1&quot;</span>,
                <span class="dt">maintainer =</span> <span class="st">&quot;m. van Iterson &lt;m.van_iterson@lumc.nl&gt;&quot;</span>,
                <span class="dt">author =</span> <span class="st">&quot;m. van Iterson&quot;</span>,
                <span class="dt">destDir=</span> <span class="st">&quot;.&quot;</span>,
                <span class="dt">license =</span> <span class="st">&quot;Artistic-2.0&quot;</span>,
                <span class="dt">pkgname =</span> <span class="st">&quot;TxDb.Hsapiens.Ensembl.v71&quot;</span>)
<span class="kw">library</span>(devtools)
<span class="kw">install</span>(<span class="st">&quot;./TxDb.Hsapiens.Ensembl.v71&quot;</span>)</code></pre></div>
<p>Now can use the Ensembl v71 annoation like any other <code>TxDb</code>-package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(TxDb.Hsapiens.Ensembl.v71)
(txdb &lt;-<span class="st"> </span>TxDb.Hsapiens.Ensembl.v71)
<span class="kw">genes</span>(txdb)</code></pre></div>
</div>
<div id="prepare-sample-metadata" class="section level2">
<h2>Prepare sample metadata</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(BBMRIomics)
colData &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="dt">file=</span><span class="kw">file.path</span>(VM_BASE_ANALYSIS, <span class="st">&quot;BBMRIomics/BBMRIomics/data&quot;</span>, <span class="st">&quot;freeze2_rna_identifiers.csv&quot;</span>), <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>, <span class="dt">as.is=</span><span class="ot">TRUE</span>)

##add methylation samplesheet
rdb &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;getStats&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_RDB)
rdb &lt;-<span class="st"> </span><span class="kw">do.call</span>(<span class="st">&#39;cbind&#39;</span>, rdb)
colData &lt;-<span class="st"> </span><span class="kw">merge</span>(colData, rdb, <span class="dt">by.x=</span><span class="st">&quot;run_id&quot;</span>, <span class="dt">by.y=</span><span class="st">&quot;ids&quot;</span>, <span class="dt">all.x=</span><span class="ot">TRUE</span>)

##harmonize phenotypes
<span class="kw">source</span>(<span class="kw">file.path</span>(<span class="kw">path.package</span>(<span class="st">&quot;BBMRIomics&quot;</span>), <span class="st">&quot;scripts/Phenotype_Helpers.R&quot;</span>), <span class="dt">verbose=</span><span class="ot">FALSE</span>)
phenotypes &lt;-<span class="st"> </span><span class="kw">cleanPhenotypes</span>()

##add phenotypes
colData &lt;-<span class="st"> </span><span class="kw">merge</span>(colData, phenotypes, <span class="dt">by=</span><span class="st">&quot;ids&quot;</span>, <span class="dt">all.x=</span><span class="ot">TRUE</span>) 

##add genotype ids
imputation &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;getImputations&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB)
hrc &lt;-<span class="st"> </span><span class="kw">subset</span>(imputation, imputation_reference ==<span class="st"> &quot;HRC&quot;</span>)
colData &lt;-<span class="st"> </span><span class="kw">merge</span>(colData, hrc[, <span class="kw">c</span>(<span class="st">&quot;ids&quot;</span>, <span class="st">&quot;imputation_id&quot;</span>)], <span class="dt">by=</span><span class="st">&quot;ids&quot;</span>, <span class="dt">all.x=</span><span class="ot">TRUE</span>)

##add uuid and gonl_id
ids &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;getIds&quot;</span>, <span class="dt">usrpwd=</span>RP3_MDB_USRPWD, <span class="dt">url=</span>RP3_MDB)
colData &lt;-<span class="st"> </span><span class="kw">merge</span>(colData, ids[, <span class="kw">c</span>(<span class="st">&quot;ids&quot;</span>, <span class="st">&quot;uuid&quot;</span>, <span class="st">&quot;gonl_id&quot;</span>, <span class="st">&quot;biobank_id&quot;</span>)], <span class="dt">by=</span><span class="st">&quot;ids&quot;</span>, <span class="dt">all.x=</span><span class="ot">TRUE</span>)

##ordering
first &lt;-<span class="st"> </span><span class="kw">match</span>(<span class="kw">c</span>(<span class="st">&quot;ids&quot;</span>, <span class="st">&quot;uuid&quot;</span>, <span class="st">&quot;biobank_id&quot;</span>, <span class="st">&quot;gonl_id&quot;</span>, <span class="st">&quot;run_id&quot;</span>), <span class="kw">colnames</span>(colData))

colData &lt;-<span class="st"> </span><span class="kw">cbind</span>(colData[, first], colData[, -first])
<span class="kw">rownames</span>(colData) &lt;-<span class="st"> </span>colData$uuid

##add flowcell maybe other derived lane, index?
colData$flowcell &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;-.*$&quot;</span>, <span class="st">&quot;&quot;</span>, colData$run_id)

<span class="kw">dim</span>(colData)
<span class="kw">head</span>(colData)

<span class="kw">table</span>(!<span class="kw">is.na</span>(colData$gonl_id))
<span class="kw">table</span>(!<span class="kw">is.na</span>(colData$imputation_id))
<span class="kw">table</span>(!(<span class="kw">is.na</span>(colData$imputation_id) &amp;<span class="st"> </span><span class="kw">is.na</span>(colData$gonl_id)))</code></pre></div>
</div>
<div id="prepare-count-table" class="section level2">
<h2>Prepare count table</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(BiocParallel)
<span class="kw">register</span>(<span class="kw">MulticoreParam</span>(<span class="dv">8</span>, <span class="dt">log=</span><span class="ot">TRUE</span>))
files &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="kw">file.path</span>(VM_BASE_DATA, <span class="st">&quot;RNASeq/v2.1.3/&quot;</span>, <span class="st">&quot;gene_read&quot;</span>), <span class="dt">full.names=</span><span class="ot">TRUE</span>, <span class="dt">pattern=</span><span class="st">&quot;gene.read.count.gz$&quot;</span>)
##select runs

files &lt;-<span class="st"> </span>files[<span class="kw">gsub</span>(<span class="st">&quot;.gene.read.count.gz&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">basename</span>(files)) %in%<span class="st"> </span>colData$run_id]

<span class="kw">head</span>(<span class="kw">read.table</span>(files[<span class="dv">1</span>], <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">check.names=</span><span class="ot">FALSE</span>)) ##checking the first few lines
tmp &lt;-<span class="st"> </span><span class="kw">bplapply</span>(files, read.table, <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">check.names=</span><span class="ot">FALSE</span>)
<span class="kw">table</span>(<span class="kw">unlist</span>(<span class="kw">lapply</span>(tmp, nrow))) ##checking all the same length
for(i in <span class="dv">2</span>:<span class="kw">length</span>(tmp)) { ##checking all genes same position
    if(!<span class="kw">all.equal</span>(tmp[[<span class="dv">1</span>]]$gene, tmp[[i]]$gene))
        <span class="kw">message</span>(<span class="st">&quot;not equal:&quot;</span>, i)
}
rownames &lt;-<span class="st"> </span><span class="kw">as.character</span>(tmp[[<span class="dv">1</span>]]$gene)
colnames &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">.gene.read.count.gz&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">basename</span>(files))
counts &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dt">nrow=</span><span class="kw">nrow</span>(tmp[[<span class="dv">1</span>]]), <span class="dt">ncol=</span><span class="kw">length</span>(tmp),
                 <span class="dt">dimnames=</span><span class="kw">list</span>(rownames, colnames))
for(i in <span class="dv">1</span>:<span class="kw">length</span>(tmp)) counts[,i] &lt;-<span class="st"> </span>tmp[[i]][,<span class="dv">2</span>]
counts[<span class="dv">1</span>:<span class="dv">5</span>, <span class="dv">1</span>:<span class="dv">5</span>]
mid &lt;-<span class="st"> </span><span class="kw">match</span>(<span class="kw">colnames</span>(counts), colData$run_id)
<span class="kw">colnames</span>(counts) &lt;-<span class="st"> </span>colData$uuid[mid]</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(TxDb.Hsapiens.Ensembl.v71)
rowRanges &lt;-<span class="st"> </span><span class="kw">genes</span>(TxDb.Hsapiens.Ensembl.v71)
<span class="kw">library</span>(GenomeInfoDb)
rowRanges &lt;-<span class="st"> </span><span class="kw">keepSeqlevels</span>(rowRanges, <span class="kw">c</span>(<span class="dv">1</span>:<span class="dv">22</span>, <span class="st">&quot;X&quot;</span>, <span class="st">&quot;Y&quot;</span>, <span class="st">&quot;MT&quot;</span>))
rowRanges
<span class="kw">head</span>(colData)
counts[<span class="dv">1</span>:<span class="dv">5</span>, <span class="dv">1</span>:<span class="dv">5</span>]
counts &lt;-<span class="st"> </span><span class="kw">makeSE</span>(counts, colData[,-<span class="dv">1</span>], rowRanges, <span class="dt">note=</span><span class="st">&quot;A maximal set of unrelated individuals for which both good quality RNA sequencing and DNA methylation data could be generated.&quot;</span>)
<span class="kw">save</span>(counts, <span class="dt">file =</span> <span class="kw">file.path</span>(VM_BASE_DATA, <span class="st">&quot;RNASeq/v2.1.3/gene_read/&quot;</span>, <span class="kw">paste0</span>(<span class="st">&quot;rnaSeqData_ReadCounts_cleaned.RData&quot;</span>)))
tmp &lt;-<span class="st"> </span>counts
for(biobank in <span class="kw">unique</span>(tmp$biobank_id)) {
    counts &lt;-<span class="st"> </span>tmp[, tmp$biobank_id ==<span class="st"> </span>biobank]
    <span class="kw">save</span>(counts, <span class="dt">file =</span> <span class="kw">file.path</span>(VM_BASE_DATA, <span class="st">&quot;RNASeq/v2.1.3/gene_read/&quot;</span>, <span class="kw">paste0</span>(<span class="st">&quot;rnaSeqData_ReadCounts_&quot;</span>, biobank, <span class="st">&quot;_cleaned.RData&quot;</span>)))
}</code></pre></div>
</div>
<div id="miscellaneous" class="section level2">
<h2>Miscellaneous</h2>
<p>Code to calculate gene-level GC-content.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(AnnotationHub)
ah &lt;-<span class="st"> </span><span class="kw">AnnotationHub</span>()
<span class="kw">query</span>(ah, <span class="kw">c</span>(<span class="st">&quot;ensembl&quot;</span>, <span class="st">&quot;sapiens&quot;</span>, <span class="st">&quot;71&quot;</span>))

gtf &lt;-<span class="st"> </span>rtracklayer::<span class="kw">import</span>(ah[<span class="st">&quot;AH7666&quot;</span>]$sourceurl)

<span class="kw">library</span>(GenomeInfoDb)
gtf &lt;-<span class="st"> </span><span class="kw">keepSeqlevels</span>(gtf, <span class="kw">c</span>(<span class="dv">1</span>:<span class="dv">22</span>, <span class="st">&quot;X&quot;</span>, <span class="st">&quot;Y&quot;</span>, <span class="st">&quot;MT&quot;</span>))
gtf &lt;-<span class="st"> </span>gtf[<span class="kw">order</span>(gtf)] ##now we have `cut and sorted` as on the VM

<span class="kw">seqlevels</span>(gtf) &lt;-<span class="st"> </span><span class="kw">mapSeqlevels</span>(<span class="kw">seqlevels</span>(gtf), <span class="st">&quot;UCSC&quot;</span>)  ##for use in combination with `BSgenome.Hsapiens.UCSC.hg19`
<span class="kw">genome</span>(gtf) &lt;-<span class="st"> &quot;hg19&quot;</span> ##for use in combination with `BSgenome.Hsapiens.UCSC.hg19`

genes &lt;-<span class="st"> </span><span class="kw">split</span>(gtf, <span class="kw">mcols</span>(gtf)$gene_id)

<span class="kw">register</span>(<span class="kw">MulticoreParam</span>(<span class="dv">8</span>))
gc &lt;-<span class="st"> </span><span class="kw">bplapply</span>(genes, function(x) {
    <span class="kw">require</span>(BSgenome.Hsapiens.UCSC.hg19)
    <span class="kw">require</span>(Rsamtools)
    <span class="kw">require</span>(Biostrings)
    seqs &lt;-<span class="st"> </span><span class="kw">getSeq</span>(Hsapiens, <span class="kw">reduce</span>(x))
    alf &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">alphabetFrequency</span>(seqs, <span class="dt">as.prob=</span><span class="ot">TRUE</span>)[, <span class="kw">c</span>(<span class="st">&quot;G&quot;</span>, <span class="st">&quot;C&quot;</span>)], <span class="dt">ncol=</span><span class="dv">2</span>)
    gc &lt;-<span class="st"> </span><span class="kw">rowSums</span>(alf)
    <span class="kw">mean</span>(gc)
})
gc &lt;-<span class="st"> </span><span class="kw">do.call</span>(<span class="st">&quot;rbind&quot;</span>, gc)
<span class="kw">all.equal</span>(<span class="kw">names</span>(genes), <span class="kw">rownames</span>(gc))
<span class="kw">mcols</span>(genes)$gc &lt;-<span class="st"> </span>gc
genes</code></pre></div>
<!-- END OF FILE. -->
<div id="refs" class="references">
<div id="ref-Zhernakova2017">
<p>Zhernakova, D. V., P. Deelen, M. Vermaat, M. van Iterson, M. van Galen, W. Arindrarto, P. van ’t Hof, et al. 2017. “Identification of context-dependent expression quantitative trait loci in whole blood.” <em>Nat. Genet.</em> 49 (1): 139–45.</p>
</div>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
