---
title: "Running an EWAS"
output:
  html_document:
    highlight: tango
    toc: true
    toc_float:
      collapsed: false
---

```{r, child="_setup.Rmd"}
```

There exist several approaches to run an epigenomewide association study, for example:

 1. [missMethyl](https://f1000research.com/articles/5-1281/v1)
 2. [coherent approach](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0600-x)
 
We have implemented our own approach [DNAmArray](https://git.lumc.nl/molepi/DNAmArray_workflow)

## Collect data ##

The following code shows how to extract some covariates and the
phenotype of interest from the
`SummerizedExperiments`-object. Furthermore, we select a subset of the
data for which the covariates and phenotype are complete, i.e. do not
contain missing -values.


<!-- START Code copy--pasted from "./data.Rmd" dd 2017-03-20 17:23 (HD). -->
<!-- Execution of chunk below leads to (an) error(s) dd 2017-03-20 16:08, hence set "eval=FALSE" (HD) -->
```{r, selectdata}
# ```{r, selectdata, eval=FALSE}

#Load 'BBMRIomics' library
library(BBMRIomics)

#Load 'methData_Mvalues_CODAM_F2_cleaned' data set
#
# utils::data(methData_Mvalues_CODAM_F2_cleaned)
#
errmessage <- try(utils::data(methData_Mvalues_CODAM_F2_cleaned))
#
if (errmessage == "methData_Mvalues_CODAM_F2_cleaned") {
  stop("data set 'methData_Mvalues_CODAM_F2_cleaned' not found")
}
# 
#END Code copy--pasted from "./data.Rmd" dd 2017-03-20 17:23 (HD).


covariates <- c("Sex", "Smoking", "Sentrix_Position")
phenotype <- "Sampling_Age"
nas <- apply(as.data.frame(colData(mvalues))[,c(phenotype, covariates)], 1, anyNA)
#
table(nas)

mvalues <- mvalues[, !nas]
```

## Create design ##

Now we can create a design matrix appropriate for use in with
`limma`'s `lmFit`-function. Both `Smoking` and `Sentrix_Position` are
considered categorical variables.

```{r, coldata2design}
# ```{r, coldata2design, eval=FALSE}
design <- as.data.frame(colData(mvalues))[,c(phenotype, covariates)]
str(design)
design$Smoking <- factor(design$Smoking)
design$Sentrix_Position <- factor(design$Sentrix_Position)
str(design)
design <- model.matrix(~., design)
str(design)
```

## Run EWAS ##

Since, we had subsetted the whole `SummerizedExperiments`-object we
can just extract the data without any further subsetting. Runnning the
EWAS is now just one function-call.

> TODO: remove cross-hyb. probes and probe containing SNPS

```{r, filter, eval=FALSE}

files <- c("http://zwdzwd.io/InfiniumAnnotation/current/hm450/hm450.manifest.pop.rda",
           "http://zwdzwd.io/InfiniumAnnotation/current/hm450/hm450.manifest.rda")
tmp <- tempfile()
download.file(files[1], tmp)
load(tmp)

hm450.manifest
hm450.manifest.pop
data <- assays(mvalues)$data


```

```{r, ewas}
# ```{r, ewas, eval=FALSE}
data <- assays(mvalues)$data
library(limma)
fit <- lmFit(data, design)
```

## Inspect results ##

Since, we have over 100 samples we ignore the usual steps of obtaining
regularized test-statistics using the function `eBayes`. The ordinary
t-statistics can be extracted as follows (as described in the vignette
of **limma**):

```{r, ewasstats}
# ```{r, ewasstats, eval=FALSE}
tstat <- fit$coef/fit$stdev.unscaled/fit$sigma
pval <- 2 * pnorm(-abs(tstat[, 2]))
padj <- p.adjust(sort(pval, decreasing = FALSE), method = "bonf")
head(padj[padj < 0.05])
```

After performing multiple-testing correction using Bonferroni's method
we can inspect the results.

```{r, qqplotewas}
# ```{r, qqplotewas, eval=FALSE}
gp <- ggplot(data.frame(pval=pval), aes(sample=-log10(pval)))
gp <- gp + stat_qq(distribution=stats::qexp, dparams=list(rate=1/log10(exp(1))))
gp <- gp + xlab(expression(paste("Expected -log"[10], plain(P))))
gp <- gp + ylab(expression(paste("Observed -log"[10], plain(P))))
gp <- gp + geom_abline(slope=1, intercept=0)
gp
```

```{r, manplotewas}
# ```{r, manplotewas, eval=FALSE}
rowRanges(mvalues)
head(pval)
tail(pval)
rData <- rowRanges(mvalues)
mcols(rData)$pval <- pval
rData

library(GenomeInfoDb)
seqlevels(rData) <- mapSeqlevels(seqlevels(rData), "Ensembl")
rData
seqlevels(rData) <- gsub("X", 23, seqlevels(rData))
seqlevels(rData) <- gsub("Y", 24, seqlevels(rData))
pdata <- as.data.frame(rData)
pdata$seqnames <- as.integer(pdata$seqnames)
library(qqman)
manhattan(pdata, chr="seqnames", bp="start", p="pval")
```


The famous CpG near the ELOVL2 gene.

```{r, tophitewas}
# ```{r, tophitewas, eval=FALSE}
top <- which.min(rData$pval)
y <- data[top, ]
x <- design[, 2]
plot(x, y, main = paste("Pearson cor.:", signif(cor(x, y), 3)), xlab = "Age (in yrs)",
     ylab = paste(rownames(data)[top], "Methylation (on M-value scale)"))
abline(lm(y ~ x), col = 2)
```


<!-- END OF FILE. -->
