---
title: "Working with HRC imputed genotypes"
output:
  html_document:
    highlight: tango
    toc: true
    toc_float:
      collapsed: false
bibliography: BBMRIomics.bib
---

```{r, child="_setup.Rmd"}
```

## Find the vcf-files ##

```{r, vcffiles}
library(BBMRIomics)
dir(VM_BASE_DATA)
dir(VM_BASE_DATA, "HRC_Imputation")
vcfs <- dir(file.path(VM_BASE_DATA, "HRC_Imputation",
                      "CODAM"), pattern = "filtered.dose.vcf.gz$", full.names = TRUE,
            recursive = TRUE)
vcfs
```

## Inspect eQTL single SNP using `VariantAnnotation` ##

Example eQTL from Zhernakova et al. [@Zhernakova2017] (see Figure 2:
Identification of the strongest modifiers of eQTL effects).

```{r, annotation}
library(org.Hs.eg.db)
select(org.Hs.eg.db, keys="NOD2", keytype="SYMBOL", columns="ENSEMBL")
library(SNPlocs.Hsapiens.dbSNP142.GRCh37)
gr <- rsidsToGRanges("rs1981760")
```

```{r, readvcf}
seqlevels(gr) <- gsub("ch", "", seqlevels(gr))
genome(gr) <- "hg19"
from <- vcfs[grepl("chr16", vcfs)]
to <- tempfile()
library(VariantAnnotation)
zipped <- bgzip(from, to)
fl <- indexTabix(zipped, format="vcf")
vcf <- readVcf(fl, genome="hg19", param=gr)
rs1981760 <- geno(vcf)$GT[1,]
```

```{r, linkhrcrna}
data(rnaSeqData_ReadCounts_cleaned)
hrc <- intersect(names(rs1981760), counts$hrc_id)
counts <- counts[,counts$hrc_id %in% hrc]
mid <- match(counts$hrc_id, names(rs1981760))
rs1981760 <- rs1981760[mid]
cnts <- assays(counts)$data
dim(cnts)
NOD2 <- cnts[rownames(cnts) == "ENSG00000167207", ]
boxplot(NOD2~rs1981760)
```

## Inspect Multiple SNPs using `getGenotypes` ##

```{r, getgenotypes}
library(SNPlocs.Hsapiens.dbSNP142.GRCh37)
gr <- rsidsToGRanges(c("rs9302752", "rs1981760"))
data(rnaSeqData_ReadCounts_cleaned)
dosages <- getGenotypes(imputation_id = counts$hrc_id, biobank="ALL", snps=gr, type="HRC", geno="DS", BASE=VM_BASE_DATA)
dosages[1:2, 1:5]
```

```{r, linkhrcrna2}
dim(counts)
dim(dosages)
head(counts$hrc_id)
head(colnames(dosages))
cnts <- assays(counts)$data
library(edgeR)

library(limma)
fit <- lmFit(cnts, dosages[1,]) ##run for 1 SNP
tstat <- fit$coef/fit$stdev.unscaled/fit$sigma
pval <- 2 * pnorm(-abs(tstat[,1]))
head(sort(pval, decreasing=FALSE))
```


## More information ##

see [VariantAnnotation](http://bioconductor.org/packages/VariantAnnotation/).

For example, extracting SNPs overlapping transcripts:

```{r, overlapsnps, eval=FALSE}
library(TxDb.Hsapiens.Ensembl.v71)
txdb <- TxDb.Hsapiens.Ensembl.v71
vcf <- readVcf(vcfs[15], genome="hg19")
rd <- rowRanges(vcf)
loc <- locateVariants(rd, txdb, CodingVariants())
head(loc, 6)
```

## References ##
