---
title: "Accessing the BIOS metadatabase"
output:
  html_document:
    highlight: tango
    toc: true
    toc_float:
      collapsed: false
bibliography: BBMRIomics.bib
---

```{r, child="_setup.Rmd"}
```

The BIOS project has generated RNA-sequencing and DNA methylation data for over
4000 individuals. As part of these data, GoNL imputed genotypes were generated
from existing genotypes and several phenotypes/demographic variables were
collected for the same set of samples. A relational, SQL-based (Postgres)
metadatabase (MDb) was created to store the large-scale multiple-omic data
collected, in a structured way. Metadata and quantifications from the RP4
metabolomics project were also added to this database.

The metadatabase is available on the BIOS-VM.

## MDb contents ##
The MDb contains as much meta-information as possible from all samples and
datatypes: location of (raw) data on srm, md5 checksum verification,
quality control information, links between the different identifiers used
(person_id, dna_id, etc) and phenotype information.
The data has been seperated into a number of entities, as described below:

| Table:                      | Description:                                                                    |
| --------------------------- | ------------------------------------------------------------------------------- |
| person                      | Information about persons (including associated ids)                            |
| relation                    | Relationship information between persons                                        |
| gwas                        | Information about GWAS runs                                                     |
| imputation                  | Information about preformed genotype imputations                                |
| visit                       | Phenotypes and other information related to the collection of samples           |
| dna_sample                  | Information about DNA samples                                                   |
| methylation_450k_run        | Information about Illumina 450k methylation array runs                          |
| methylation_450k_freeze     | Which methylation runs are included in which data freezes (and freeze subsets)  |
| rna_sample                  | Information about RNA samples                                                   |
| rna_run                     | Information about RNAseq runs                                                   |
| rna_merged_run              | Which RNAseq runs are included in merged RNA runs                               |
| rna_freeze                  | Which RNAseq runs are included in which data freezes (and freeze subsets)       |
| nightingale_run             | Information regarding nightingale runs                                          |
| nightingale_quantification  | Metabolomics quantification measurements                                        |

The `getTables` function can be used to retrieve a list of table names as well:
```{r}
getTables()
```

## Available views ##
Views are predefined SQL queries which can be used to extract a subset of the
available information from the database. The names of the available views can
be retrieved using the `getViews`-function:
```{r}
getViews()
```

## Retrieving views and querying the database ##

To retrieve a view the `getSQLview`-function can be used. Note that view names
are not case sensitive.
```{r}
head(getSQLview("getids"))
```

> We can always add views if necessary; please contact Leon Mei.

If you have a Postgres account on the BIOS-VM you can also query the tables
directly using the `runQuery`-function. This function is just a wrapper around
the `dbGetQuery`-function from the `RPostgreSQL`-package, so that package (or 
any other API which interacts with postgres) can also be used directly. 
```{r eval=FALSE}
runQuery("SELECT * FROM visits;")
```