---
title: "RNA "
output:
  html_document:
    highlight: tango
    toc: true
    toc_float:
      collapsed: false
bibliography: BBMRIomics.bib
---

```{r, child="_setup.Rmd"}
```

# Library preparation and alignment #

Detailed description of the RNASeq data generation and preprocessing
can be found in the supplementary material of the paper by Zhernakova
et al. [@Zhernakova2017]

# Feature quantification read counts #

Gene read counts are generated using
[htseq](http://www-huber.embl.de/HTSeq/doc/count.html) using mode
`union` for handling overlapping exons.

Reference annotation used is from ensembl:
[](http://ftp.ensembl.org/pub/release-71/gtf/homo_sapiens/Homo_sapiens.GRCh37.71.gtf.gz")
but only chromosomes 1-22 and X, Y (and MT) are retained.


## Prepare gene annotation ##

For convenience we create a `TxDb`-package. For more information on
using a `TxDb`-package see
[GenomicFeatures](http://bioconductor.org/packages/release/bioc/vignettes/GenomicFeatures/inst/doc/GenomicFeatures.pdf).

*** Note this requires sudo rights to install and use sudo -i to load /etc/profile with global R_SITE_LIB =/opt/... ***

```{r, txdb, eval=FALSE}

library(GenomicFeatures)
library(rtracklayer)
gtf <- "ftp://ftp.ensembl.org/pub/release-71/gtf/homo_sapiens/Homo_sapiens.GRCh37.71.gtf.gz"
txdb <- makeTxDbFromGFF(gtf, organism="Homo sapiens")

##Hack by James MacDonald (https://support.bioconductor.org/p/89855/)
con <- dbconn(txdb)
## insert a Resource URL entry in the metadata table,
## pointing to the ftp site where the data were procured
DBI::dbGetQuery(con, "INSERT INTO metadata VALUES ('Resource URL', 'ftp://ftp.ensembl.org/pub/release-71/gtf/homo_sapiens/Homo_sapiens.GRCh37.71.gtf.gz');")

makeTxDbPackage(txdb,
                version = "0.0.1",
                maintainer = "m. van Iterson <m.van_iterson@lumc.nl>",
                author = "m. van Iterson",
                destDir= ".",                
                license = "Artistic-2.0",
                pkgname = "TxDb.Hsapiens.Ensembl.v71")

library(devtools)
install("./TxDb.Hsapiens.Ensembl.v71")

```

Now can use the Ensembl v71 annoation like any other `TxDb`-package.

```{r, txdbexample, eval=FALSE}
library(TxDb.Hsapiens.Ensembl.v71)
(txdb <- TxDb.Hsapiens.Ensembl.v71)
genes(txdb)
```

## Prepare sample metadata ##

```{r, coldatarna, eval=FALSE}
library(BIOSRutils)
MDB <- "https://metadatabase.bbmrirp3-lumc.surf-hosted.nl:6984/bios/"
RDB <- "https://metadatabase.bbmrirp3-lumc.surf-hosted.nl:6984/rp3_analysis/"

ids <- getView("getIds")

runs <- getView("getRNASeqRuns")
runs <- runs[!duplicated(runs$run_id),] ##drop overlapping freezes
runs <- subset(runs, qc == "passed")
runs <- subset(runs, type != "replicate")

merged <- subset(runs, type == "merged")
merged <- subset(merged, !(run_id %in% c("BD1NYRACXX-5-27_AC1JL5ACXX-6-27", "AC1C40ACXX-6-15_BD2D5MACXX-6-15")))
original <- subset(runs, !(ids %in% merged$ids) & type == "original")

runs <- rbind(merged, original)

mid <- match(runs$ids, ids$ids)
runs$uuid <- ids$uuid[mid]

rdb <- getView("getStats", url=RDB)
rdb <- do.call('cbind', rdb)

runs <- merge(runs, rdb, by.x="run_id", by.y="ids", all.x=TRUE)

relations <- getView("getRelations")
relations <- merge(relations, runs[, c("run_id", "ids")], by="ids") ##add run_id to relations
##renaming relation_id's to uuid's
mid <- match(relations$relation_id, ids$ids)
relations$relation_uuid <- NA
relations$relation_uuid[!is.na(mid)] <- ids$uuid[mid[!is.na(mid)]]

##harmonize phenotypes
phenotypes <- cleanPhenotypes()
colData <- merge(phenotypes, runs, by="ids", all.y=TRUE) ##add phenotypes

colData <- merge(colData, relations[,c("run_id", "gonl_id", "relation_type", "relation_uuid")], by="run_id", all.x=TRUE) ##add relations

colData <- .reduceRelations(colData)

##renaming columns
colnames(colData) <- gsub("^type$", "run_type", colnames(colData))

head(colData)
##ordering
first <- match(c("ids", "uuid", "biobank_id", "run_type", "qc", "relation_type", "relation_uuid", "gonl_id", "run_id"), colnames(colData))

colData <- cbind(colData[, first], colData[, -first])
colData <- colData[, !(colnames(colData) %in% c("ids", "rna_id", "freeze"))]

rownames(colData) <- colData$uuid

imputation <- getView("getImputations")
hrc <- subset(imputation, imputation_reference=="HRC")

uuids <- intersect(colData$uuid, hrc$uuid)
colData <- colData[colData$uuid %in% uuids, ]
mid <- match(colData$uuid, hrc$uuid)
colData$hrc_id <- hrc$imputation_id[mid]
```


## Prepare count table ##

```{r, countdata, eval=FALSE}
library(BiocParallel)
register(MulticoreParam(8, log=TRUE))
files <- list.files(file.path(RP3DATADIR, "RNASeq/v2.1.3/", "gene_read"), full.names=TRUE, pattern="gene.read.count.gz$")
##select runs

files <- files[gsub(".gene.read.count.gz", "", basename(files)) %in% colData$run_id]

head(read.table(files[1], header=TRUE, check.names=FALSE)) ##checking the first few lines
tmp <- bplapply(files, read.table, header=TRUE, check.names=FALSE)
table(unlist(lapply(tmp, nrow))) ##checking all the same length
for(i in 2:length(tmp)) { ##checking all genes same position
    if(!all.equal(tmp[[1]]$gene, tmp[[i]]$gene))
        message("not equal:", i)
}
rownames <- as.character(tmp[[1]]$gene)
colnames <- gsub("\\.gene.read.count.gz", "", basename(files))
counts <- matrix(nrow=nrow(tmp[[1]]), ncol=length(tmp),
                 dimnames=list(rownames, colnames))
for(i in 1:length(tmp)) counts[,i] <- tmp[[i]][,2]
counts[1:5, 1:5]
mid <- match(colnames(counts), colData$run_id)
colnames(counts) <- colData$uuid[mid]
```

```{r, rowdatarna, eval=FALSE}
library(TxDb.Hsapiens.Ensembl.v71)
rowRanges <- genes(TxDb.Hsapiens.Ensembl.v71)
library(GenomeInfoDb)
rowRanges <- keepSeqlevels(rowRanges, c(1:22, "X", "Y", "MT"))
rowRanges
head(colData)
counts[1:5, 1:5]
counts <- makeSE(counts, colData, rowRanges, note="")
save(counts, file = file.path("/virdir/Scratch/RP3_data/RNASeq/v2.1.3/gene_read/", paste0("rnaSeqData_ReadCounts_cleaned.RData")))
```

## Miscellaneous ##

Code to calculate gene-level GC-content.

```{r, gcrna, eval=FALSE}
library(AnnotationHub)
ah <- AnnotationHub()
query(ah, c("ensembl", "sapiens", "71"))

gtf <- rtracklayer::import(ah["AH7666"]$sourceurl)

library(GenomeInfoDb)
gtf <- keepSeqlevels(gtf, c(1:22, "X", "Y", "MT"))
gtf <- gtf[order(gtf)] ##now we have `cut and sorted` as on the VM

seqlevels(gtf) <- mapSeqlevels(seqlevels(gtf), "UCSC")  ##for use in combination with `BSgenome.Hsapiens.UCSC.hg19`
genome(gtf) <- "hg19" ##for use in combination with `BSgenome.Hsapiens.UCSC.hg19`

genes <- split(gtf, mcols(gtf)$gene_id)

register(MulticoreParam(8))
gc <- bplapply(genes, function(x) {
    require(BSgenome.Hsapiens.UCSC.hg19)
    require(Rsamtools)
    require(Biostrings)
    seqs <- getSeq(Hsapiens, reduce(x))
    alf <- as.matrix(alphabetFrequency(seqs, as.prob=TRUE)[, c("G", "C")], ncol=2)
    gc <- rowSums(alf)
    mean(gc)
})
gc <- do.call("rbind", gc)
all.equal(names(genes), rownames(gc))
mcols(genes)$gc <- gc
genes

```
